

<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="./">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>AMBER &mdash; SIRAH Force Field  documentation</title>
      <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=80d5e7a1" />
      <link rel="stylesheet" type="text/css" href="_static/css/theme.css?v=e59714d7" />
      <link rel="stylesheet" type="text/css" href="_static/copybutton.css?v=76b2166b" />
      <link rel="stylesheet" type="text/css" href="_static/custom.css?v=3437f865" />

  
      <script src="_static/jquery.js?v=5d32c60e"></script>
      <script src="_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
      <script src="_static/documentation_options.js?v=5929fcd5"></script>
      <script src="_static/doctools.js?v=9a2dae69"></script>
      <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
      <script src="_static/clipboard.min.js?v=a7894cd8"></script>
      <script src="_static/copybutton.js?v=f281be69"></script>
      <script async="async" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="GROMACS" href="Tutorials%20gromacs.html" />
    <link rel="prev" title="Developers" href="Developers.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="index.html" class="icon icon-home">
            SIRAH Force Field
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Manual</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="About%20SIRAH.html">About SIRAH</a></li>
<li class="toctree-l1"><a class="reference internal" href="Background%20SIRAH.html">Background</a></li>
<li class="toctree-l1"><a class="reference internal" href="Citation.html">Citations</a></li>
<li class="toctree-l1"><a class="reference internal" href="Further%20reading.html">Further Reading</a></li>
<li class="toctree-l1"><a class="reference internal" href="FAQ.html">FAQs</a></li>
<li class="toctree-l1"><a class="reference internal" href="Developers.html">Developers</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Tutorials</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="current reference internal" href="#">AMBER</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#setting-up-sirah">Setting up SIRAH</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#required-software">Required Software</a></li>
<li class="toctree-l3"><a class="reference internal" href="#prior-knowledge">Prior knowledge</a></li>
<li class="toctree-l3"><a class="reference internal" href="#download-and-setting-up-sirah">Download and Setting up SIRAH</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#dna-molecule-in-implicit-solvent">1. DNA molecule in implicit solvent</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#build-cg-representations">1.1. Build CG representations</a></li>
<li class="toctree-l3"><a class="reference internal" href="#prepare-leap-input">1.2. Prepare LEaP input</a></li>
<li class="toctree-l3"><a class="reference internal" href="#run-leap">1.3. Run LEaP</a></li>
<li class="toctree-l3"><a class="reference internal" href="#run-the-simulation">1.4. Run the simulation</a></li>
<li class="toctree-l3"><a class="reference internal" href="#visualizing-the-simulation">1.5. Visualizing the simulation</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#dna-molecule-in-explicit-solvent">2. DNA molecule in explicit solvent</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#id12">2.1. Build CG representations</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id13">2.2. Prepare LEaP input</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id14">2.3. Run LEaP</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id15">2.4. Run the simulation</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id16">2.5. Visualizing the simulation</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#multiscale-simulation-in-implicit-solvent">3. Multiscale simulation in implicit solvent</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#id19">3.1. Build CG representations</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id20">3.2. Prepare LEaP input</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id21">3.3. Run LEaP</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id22">3.4. Run the simulation</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id23">3.5. Visualizing the simulation</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#closed-circular-dna-in-implicit-solvent">4. Closed circular DNA in implicit solvent</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#id29">4.1. Build CG representations</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id30">4.2. Prepare LEaP input</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id31">4.3. Run LEaP</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id32">4.4. Run the simulation</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id33">4.5. Visualizing the simulation</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#proteins-in-explicit-solvent">5. Proteins in explicit solvent</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#id37">5.1. Build CG representations</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id39">5.2. Prepare LEaP input</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id40">5.3. Run LEaP</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id41">5.4. Run the simulation</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id42">5.5. Visualizing the simulation</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#lipid-bilayers-in-explicit-solvent">6. Lipid bilayers in explicit solvent</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#id46">6.1. Build CG representations</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id47">6.2. Prepare LEaP input</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id49">6.3. Run LEaP</a></li>
<li class="toctree-l3"><a class="reference internal" href="#resize-the-box-with-cpptraj">6.4. Resize the box with CPPTRAJ</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id50">6.5. Run the simulation</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id51">6.6. Visualizing the simulation</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#membrane-proteins-in-explicit-solvent">7. Membrane proteins in explicit solvent</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#id56">7.1. Build CG representations</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#embed-the-protein-in-a-lipid-bilayer">7.1.1. Embed the protein in a lipid bilayer</a></li>
<li class="toctree-l4"><a class="reference internal" href="#delete-close-contact-lipid-molecules">7.1.2. Delete close contact lipid molecules</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#id58">7.2. Prepare LEaP input</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id59">7.3. Run LEaP</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id61">7.4. Resize the box with CPPTRAJ</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id62">7.5. Run the simulation</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id63">7.6. Visualizing the simulation</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#glycoprotein-in-explicit-solvent">8. Glycoprotein in explicit solvent</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#id68">8.1. Build CG representations</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id71">8.2. Prepare LEaP input</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id72">8.3. Run LEaP</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id73">8.4. Run the simulation</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id74">8.5. Visualizing the simulation</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#non-enveloped-viral-like-particle">9. Non-enveloped Viral-Like Particle</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#structure-preprocessing">9.1. Structure preprocessing</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#working-with-the-biological-assembly-of-an-entire-vlp">9.1.1 Working with the Biological Assembly of an entire VLP</a></li>
<li class="toctree-l4"><a class="reference internal" href="#calculate-system-charge">9.1.2 Calculate system charge</a></li>
<li class="toctree-l4"><a class="reference internal" href="#packing-the-assimetric-unit-to-build-an-entire-vlp-alternative">9.1.3 Packing the assimetric unit to build an entire VLP (Alternative)</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#id82">9.2. Build CG representations</a></li>
<li class="toctree-l3"><a class="reference internal" href="#wrapping-up-vlp-system-with-packmol">9.3. Wrapping up VLP system with Packmol</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id83">9.4 Prepare LEaP input</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id84">9.5 Run LEaP</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id85">9.6. Run the simulation</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id86">9.7. Visualizing the simulation</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#appendix">Appendix</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#calculating-ionic-concentrations">Calculating ionic concentrations</a></li>
<li class="toctree-l3"><a class="reference internal" href="#available-mapping-files">Available mapping files</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="Tutorials%20gromacs.html">GROMACS</a></li>
<li class="toctree-l1"><a class="reference internal" href="Tutorials%20namd.html">NAMD</a></li>
<li class="toctree-l1"><a class="reference internal" href="Tutorials%20sirahtools.html">SIRAH tools</a></li>
<li class="toctree-l1"><a class="reference internal" href="Tutorials%20analysis.html">Basic analyses VMD</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">SIRAH Force Field</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">AMBER</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/Tutorials amber.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="amber">
<h1>AMBER<a class="headerlink" href="#amber" title="Link to this heading"></a></h1>
<section id="setting-up-sirah">
<span id="download-amber"></span><span id="id1"></span><h2>Setting up SIRAH<a class="headerlink" href="#setting-up-sirah" title="Link to this heading"></a></h2>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>SIRAH Force Field is distributed with <a class="reference external" href="https://ambermd.org/index.php">AMBER</a> and <a class="reference external" href="https://ambermd.org/AmberTools.php">AmberTools</a> official releases.</p>
</div>
<section id="required-software">
<h3>Required Software<a class="headerlink" href="#required-software" title="Link to this heading"></a></h3>
<p><strong>AMBER</strong></p>
<p>Amber and AmberTools (version 16 or later) versions properly installed and running in your computer.</p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p><a class="reference external" href="https://ambermd.org/GetAmber.php#ambertools">Download</a> the latest version of Amber and AmberTools and refer to the <em>How to obtain Amber23</em> section for more details on getting Amber23. In addition, go to the <a class="reference external" href="https://ambermd.org/Installation.php">Install Amber</a> page for specific instructions based on your operating system. If you have Amber already installed, it is unnecessary to install AmberTools separately.</p>
</div>
<p><strong>VMD</strong></p>
<p>The molecular visualization program <a class="reference external" href="https://www.ks.uiuc.edu/Research/vmd/">VMD</a>, version 1.9.3 or later (<a class="reference external" href="https://www.ks.uiuc.edu/Development/Download/download.cgi?PackageName=VMD">freely available download</a>). Check VMD’s <a class="reference external" href="https://www.ks.uiuc.edu/Research/vmd/current/ig/node6.html">Installation guide</a> for more instructions.</p>
</section>
<section id="prior-knowledge">
<h3>Prior knowledge<a class="headerlink" href="#prior-knowledge" title="Link to this heading"></a></h3>
<p>How to perform a standard atomistic molecular dynamics simulations with AMBER and basic usage of
VMD.</p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p><strong>AMBER</strong></p>
<p>If you are not familiar with atomistic molecular dynamics simulations, we strongly recommend you to perform the basic usage tutorials of Amber and AmberTools: <a class="reference external" href="https://ambermd.org/tutorials/BuildingSystems.php">1. Building Systems</a>, <a class="reference external" href="https://ambermd.org/tutorials/Relaxation.php">3. Creating Stable Systems and Running MD</a> and <a class="reference external" href="https://ambermd.org/tutorials/Introductory.php">5. Case Studies</a>. For more details and documentation, check <a class="reference external" href="https://ambermd.org/doc12/Amber23.pdf">Amber23 Manual</a>.</p>
<p><strong>VMD</strong></p>
<p>If you are not familiar with VMD, we strongly recommend you to perform the basic usage tutorial of VMD (<a class="reference external" href="https://www.ks.uiuc.edu/Training/Tutorials/vmd/tutorial-html/index.html">Using VMD</a>). For more details and documentation, check <a class="reference external" href="https://www.ks.uiuc.edu/Research/vmd/current/ug/ug.html">VMD User Guide</a> and the official <a class="reference external" href="https://www.ks.uiuc.edu/Research/vmd/current/docs.html#tutorials">VMD documentation</a>.</p>
</div>
</section>
<section id="download-and-setting-up-sirah">
<h3>Download and Setting up SIRAH<a class="headerlink" href="#download-and-setting-up-sirah" title="Link to this heading"></a></h3>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Please report bugs, errors or enhancement requests through <a class="reference external" href="https://github.com/SIRAHFF/documentation/issues">Issue Tracker</a> or if you have a question about SIRAH open a <a class="reference external" href="https://github.com/SIRAHFF/documentation/discussions">New Discussion</a>.</p>
</div>
<p>Download <a class="reference external" href="https://github.com/SIRAHFF/documentation/releases/tag/AMBER">SIRAH for Amber</a> and uncompress it into your working directory.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>tar<span class="w"> </span>-xzvf<span class="w"> </span>sirah_x2.3_24-07.amber.tar.gz
</pre></div>
</div>
<p>If you are using a different version of SIRAH, type:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>tar<span class="w"> </span>-xzvf<span class="w"> </span>sirah_<span class="o">[</span>version<span class="o">]</span>.amber.tgz
</pre></div>
</div>
<p>You will get the folder <code class="docutils literal notranslate"><span class="pre">sirah_[version].amber/</span></code> containing the force field definition, SIRAH Tools in the
<code class="docutils literal notranslate"><span class="pre">sirah_[version].amber/tools/</span></code> folder, molecular structures to build up systems in <code class="docutils literal notranslate"><span class="pre">sirah_[version].amber/PDB/</span></code> and the required material to perform the tutorials in <code class="docutils literal notranslate"><span class="pre">sirah_[version].amber/tutorial/X.X/</span></code></p>
<div class="admonition caution">
<p class="admonition-title">Caution</p>
<p>Remember to change <strong>X.X</strong> to the number that corresponds to the tutorial you are going to do.</p>
</div>
<p>Make a new folder for this tutorial in your working directory:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>mkdir<span class="w"> </span>tutorialX.X<span class="p">;</span><span class="w"> </span><span class="nb">cd</span><span class="w"> </span>tutorialX.X
</pre></div>
</div>
<p>Create the following symbolic link in the folder tutorialX.X:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>ln<span class="w"> </span>-s<span class="w"> </span>../sirah_x2.3_24-07.amber<span class="w"> </span>sirah.amber
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>SIRAH Force Field is also distributed with <a class="reference external" href="https://ambermd.org/index.php">AMBER</a> and <a class="reference external" href="https://ambermd.org/AmberTools.php">AmberTools</a> official releases. To use the native AMBER version of SIRAH, create a symbolic link located in $AMBERHOME:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>ln<span class="w"> </span>-s<span class="w"> </span><span class="nv">$AMBERHOME</span>/dat/SIRAH<span class="w"> </span>sirah.amber
</pre></div>
</div>
<p>Check the <a class="reference external" href="https://ambermd.org/doc12/Amber23.pdf">AMBER Manual</a> section <strong>3.11.2</strong> for more details.</p>
<p>However, if you want the latest parameters and implementations strongly advise you to use the developers version of SIRAH from GitHub.</p>
</div>
</section>
</section>
<section id="dna-molecule-in-implicit-solvent">
<span id="tutorial-1"></span><h2>1. DNA molecule in implicit solvent<a class="headerlink" href="#dna-molecule-in-implicit-solvent" title="Link to this heading"></a></h2>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Please report bugs, errors or enhancement requests through <a class="reference external" href="https://github.com/SIRAHFF/documentation/issues">Issue Tracker</a> or if you have a question about SIRAH open a <a class="reference external" href="https://github.com/SIRAHFF/documentation/discussions">New Discussion</a>.</p>
</div>
<p>This tutorial shows how to perform a Coarse Grained (CG) simulation of a double stranded DNA using the Generalized Born model for implicit solvent (GB) and the SIRAH force field. The main references
for this tutorial are: <a class="reference external" href="https://pubs.acs.org/doi/abs/10.1021/ct900653p">Dans et al.</a> (latest parameters are those reported <a class="reference external" href="https://pubs.acs.org/doi/abs/10.1021/ct100379f">here</a>) and <a class="reference external" href="https://academic.oup.com/bioinformatics/article/32/10/1568/1743152">Machado &amp; Pantano</a>. We strongly advise you to read these articles before starting the tutorial.</p>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>Check the <a class="reference internal" href="#download-amber"><span class="std std-ref">Setting up SIRAH</span></a> section for download and set up details before starting this tutorial.
Since this is <strong>Tutorial 1</strong>, remember to replace <code class="docutils literal notranslate"><span class="pre">X.X</span></code> and the files corresponding to this tutorial can be found in: <code class="docutils literal notranslate"><span class="pre">sirah_[version].amber/tutorial/1/</span></code></p>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>If you are not familiar with DNA stuff, we strongly recommend you to first perform the <a class="reference external" href="http://ambermd.org/tutorials/basic/tutorial1">AMBER
tutorial on DNA</a>.</p>
</div>
<section id="build-cg-representations">
<h3>1.1. Build CG representations<a class="headerlink" href="#build-cg-representations" title="Link to this heading"></a></h3>
<p>Map the atomistic structure of a 20-mer DNA to its CG representation:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>./sirah.amber/tools/CGCONV/cgconv.pl<span class="w"> </span>-i<span class="w"> </span>./sirah.amber/tutorial/1/dna.pdb<span class="w">  </span>-o<span class="w"> </span>dna_cg.pdb
</pre></div>
</div>
<p>The input file <code class="docutils literal notranslate"><span class="pre">-i</span></code> dna.pdb contains all the heavy atoms composing the DNA molecule, while the  output <code class="docutils literal notranslate"><span class="pre">-o</span></code> dna_cg.pdb preserves a few of them.
Please check both PDB structures using VMD:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>vmd<span class="w"> </span>-m<span class="w"> </span>./sirah.amber/tutorial/1/dna.pdb<span class="w"> </span>dna_cg.pdb
</pre></div>
</div>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>This is the basic usage of the script <strong>cgconv.pl</strong>, you can learn other capabilities from its help by typing:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>./sirah.amber/tools/CGCONV/cgconv.pl<span class="w"> </span>-h
</pre></div>
</div>
</div>
<p>From now on it is just normal AMBER stuff!</p>
</section>
<section id="prepare-leap-input">
<h3>1.2. Prepare LEaP input<a class="headerlink" href="#prepare-leap-input" title="Link to this heading"></a></h3>
<p>Use a text editor to create the file <code class="docutils literal notranslate"><span class="pre">gensystem.leap</span></code> including the following lines:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="gp"># </span>Load<span class="w"> </span>SIRAH<span class="w"> </span>force<span class="w"> </span>field
<span class="go">addPath ./sirah.amber</span>
<span class="go">source leaprc.sirah</span>

<span class="gp"># </span>Load<span class="w"> </span>model
<span class="go">dna = loadpdb dna_cg.pdb</span>

<span class="gp"># </span>Save<span class="w"> </span>Parms
<span class="go">saveAmberParmNetcdf dna dna_cg.prmtop dna_cg.ncrst</span>

<span class="gp"># </span>EXIT
<span class="go">quit</span>
</pre></div>
</div>
</section>
<section id="run-leap">
<h3>1.3. Run LEaP<a class="headerlink" href="#run-leap" title="Link to this heading"></a></h3>
<p>Run the LEAP application to generate the molecular topology and initial coordinate files:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>tleap<span class="w"> </span>-f<span class="w"> </span>gensystem.leap
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Warning messages about long, triangular or square bonds in <code class="docutils literal notranslate"><span class="pre">leap.log</span></code> file are fine and
expected due to the CG topology.</p>
</div>
<p>This should create a topology file <code class="docutils literal notranslate"><span class="pre">dna_cg.prmtop</span></code> and a coordinate file <code class="docutils literal notranslate"><span class="pre">dna_cg.ncrst</span></code>.</p>
<p>Use VMD to check how the CG model looks like:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>vmd<span class="w"> </span>dna_cg.prmtop<span class="w"> </span>dna_cg.ncrst<span class="w"> </span>-e<span class="w"> </span>./sirah.amber/tools/sirah_vmdtk.tcl
</pre></div>
</div>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>VMD assigns default radius to unknown atom types, the script <code class="docutils literal notranslate"><span class="pre">sirah_vmdtk.tcl</span></code> sets the right
ones, according to the CG representation. It also provides a kit of useful selection macros, coloring methods and backmapping utilities.
Use the command <code class="docutils literal notranslate"><span class="pre">sirah_help</span></code> in the Tcl/Tk console of VMD to access the manual pages. To learn about SIRAH Tools’ capabilities, you can also go to the <a class="reference internal" href="Tutorials%20sirahtools.html#sirah-tools"><span class="std std-ref">SIRAH Tools tutorial</span></a>.</p>
</div>
</section>
<section id="run-the-simulation">
<h3>1.4. Run the simulation<a class="headerlink" href="#run-the-simulation" title="Link to this heading"></a></h3>
<p>Make a new folder for the run:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>mkdir<span class="w"> </span>-p<span class="w"> </span>run<span class="p">;</span><span class="w"> </span><span class="nb">cd</span><span class="w"> </span>run
</pre></div>
</div>
<p>The folder <code class="docutils literal notranslate"><span class="pre">sirah.amber/tutorial/1/PMEMD.GPU/</span></code> contains typical input files for energy minimization
(<code class="docutils literal notranslate"><span class="pre">em_GB.in</span></code>), equilibration (<code class="docutils literal notranslate"><span class="pre">eq_GB.in</span></code>) and production (<code class="docutils literal notranslate"><span class="pre">md_GB.in</span></code>) runs. Please check carefully the input
flags therein.</p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<blockquote>
<div><p><strong>Some commonly used flags in AMBER</strong></p>
</div></blockquote>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">-i</span></code>: Input file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-o</span></code>: Output file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-p</span></code>: Parameter/topology file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-c</span></code>: Coordinate file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-r</span></code>: Restart file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-x</span></code>: Trajectory file.</p></li>
</ul>
</div>
<div class="admonition caution">
<p class="admonition-title">Caution</p>
<p>These input files are executed by the <strong>GPU</strong> implementation of <code class="docutils literal notranslate"><span class="pre">pmemd.cuda</span></code>. Other available modules are <code class="docutils literal notranslate"><span class="pre">sander</span></code>  or <code class="docutils literal notranslate"><span class="pre">pmemd</span></code>, which are both <strong>CPU</strong> implementations of Amber.</p>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>You can find example input files for CPU versions of <code class="docutils literal notranslate"><span class="pre">sander</span></code> and <code class="docutils literal notranslate"><span class="pre">pmemd</span></code> at folders <code class="docutils literal notranslate"><span class="pre">SANDER/</span></code> and  <code class="docutils literal notranslate"><span class="pre">PMEMD.CPU/</span></code>, within <code class="docutils literal notranslate"><span class="pre">sirah.amber/tutorial/1/</span></code></p>
</div>
<p><strong>Energy Minimization:</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/1/PMEMD.GPU/em_GB.in<span class="w"> </span>-p<span class="w"> </span>../dna_cg.prmtop<span class="w"> </span>-c<span class="w"> </span>../dna_cg.ncrst<span class="w"> </span>-o<span class="w"> </span>dna_cg_em.out<span class="w"> </span>-r<span class="w"> </span>dna_cg_em.ncrst<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>In the course of long MD simulations the capping residues may eventually separate, this effect is
called helix fraying. To avoid such behavior is necessary to set Watson-Crick restraints for the capping base pairs of this CG DNA at the end of <code class="docutils literal notranslate"><span class="pre">eq_GB.in</span></code> and <code class="docutils literal notranslate"><span class="pre">md_GB.in</span></code> files. Check the files lines that start with <em>&amp;rst</em>.</p>
</div>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>If you are using <code class="docutils literal notranslate"><span class="pre">sander</span></code>, to avoid the helix fraying, you must create a symbolic link to the file <code class="docutils literal notranslate"><span class="pre">dna_cg.RST</span></code>, which
contains the definition of Watson-Crick restraints for the capping base pairs of this CG DNA:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>ln<span class="w"> </span>-s<span class="w"> </span>../sirah.amber/tutorial/1/SANDER/dna_cg.RST
</pre></div>
</div>
<p>The file <code class="docutils literal notranslate"><span class="pre">dna_cg.RST</span></code> can only be read by <code class="docutils literal notranslate"><span class="pre">sander</span></code>, <code class="docutils literal notranslate"><span class="pre">pmemd</span></code> reads a different restrain format.</p>
</div>
<p><strong>Equilibration:</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/1/PMEMD.GPU/eq_GB.in<span class="w"> </span>-p<span class="w"> </span>../dna_cg.prmtop<span class="w"> </span>-c<span class="w"> </span>dna_cg_em.ncrst<span class="w"> </span>-o<span class="w"> </span>dna_cg_eq.out<span class="w"> </span>-r<span class="w"> </span>dna_cg_eq.ncrst<span class="w"> </span>-x<span class="w"> </span>dna_cg_eq.nc<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
<p><strong>Production (100ns):</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/1/PMEMD.GPU/md_GB.in<span class="w"> </span>-p<span class="w"> </span>../dna_cg.prmtop<span class="w"> </span>-c<span class="w"> </span>dna_cg_eq.ncrst<span class="w"> </span>-o<span class="w"> </span>dna_cg_md.out<span class="w"> </span>-r<span class="w"> </span>dna_cg_md.ncrst<span class="w">  </span>-x<span class="w"> </span>dna_cg_md.nc<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
</section>
<section id="visualizing-the-simulation">
<h3>1.5. Visualizing the simulation<a class="headerlink" href="#visualizing-the-simulation" title="Link to this heading"></a></h3>
<p>Now you can load, visualize and analize the trajectory in VMD:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">vmd</span> <span class="o">../</span><span class="n">dna_cg</span><span class="o">.</span><span class="n">prmtop</span> <span class="o">../</span><span class="n">dna_cg</span><span class="o">.</span><span class="n">ncrst</span> <span class="n">dna_cg_md</span><span class="o">.</span><span class="n">nc</span> <span class="o">-</span><span class="n">e</span> <span class="o">../</span><span class="n">sirah</span><span class="o">.</span><span class="n">amber</span><span class="o">/</span><span class="n">tools</span><span class="o">/</span><span class="n">sirah_vmdtk</span><span class="o">.</span><span class="n">tcl</span>
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The file <code class="docutils literal notranslate"><span class="pre">sirah_vmdtk.tcl</span></code> is a Tcl script that is part of SIRAH Tools and contains the macros to properly visualize the coarse-grained structures in VMD. Use the command <code class="docutils literal notranslate"><span class="pre">sirah-help</span></code> in the Tcl/Tk console of VMD to access the manual pages. To learn about SIRAH Tools’ capabilities, you can also go to the <a class="reference internal" href="Tutorials%20sirahtools.html#sirah-tools"><span class="std std-ref">SIRAH Tools tutorial</span></a>.</p>
</div>
</section>
</section>
<section id="dna-molecule-in-explicit-solvent">
<span id="tutorial-2"></span><h2>2. DNA molecule in explicit solvent<a class="headerlink" href="#dna-molecule-in-explicit-solvent" title="Link to this heading"></a></h2>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Please report bugs, errors or enhancement requests through <a class="reference external" href="https://github.com/SIRAHFF/documentation/issues">Issue Tracker</a> or if you have a question about SIRAH open a <a class="reference external" href="https://github.com/SIRAHFF/documentation/discussions">New Discussion</a>.</p>
</div>
<p>This tutorial shows how to use the SIRAH force field to perform a coarse grained (CG) simulation of a
double stranded DNA embedded in explicit solvent (called WatFour, WT4). The main references for
this tutorial are: <a class="reference external" href="https://pubs.acs.org/doi/abs/10.1021/ct900653p">Dans et al.</a> (latest parameters are those reported <a class="reference external" href="https://pubs.acs.org/doi/abs/10.1021/ct100379f">here</a>) and <a class="reference external" href="https://academic.oup.com/bioinformatics/article/32/10/1568/1743152">Machado &amp; Pantano</a>.
We strongly advise you to read these articles before starting the tutorial.</p>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>Check the <a class="reference internal" href="#download-amber"><span class="std std-ref">Setting up SIRAH</span></a> section for download and set up details before starting this tutorial.
Since this is <strong>Tutorial 2</strong>, remember to replace <code class="docutils literal notranslate"><span class="pre">X.X</span></code> and the files corresponding to this tutorial can be found in: <code class="docutils literal notranslate"><span class="pre">sirah_[version].amber/tutorial/2/</span></code></p>
</div>
<section id="id12">
<h3>2.1. Build CG representations<a class="headerlink" href="#id12" title="Link to this heading"></a></h3>
<p>Map the atomistic structure of a 20-mer DNA to its CG representation:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>./sirah.amber/tools/CGCONV/cgconv.pl<span class="w"> </span>-i<span class="w"> </span>./sirah.amber/tutorial/2/dna.pdb<span class="w"> </span>-o<span class="w"> </span>dna_cg.pdb
</pre></div>
</div>
<p>The input file <code class="docutils literal notranslate"><span class="pre">-i</span></code> dna.pdb contains all the heavy atoms composing the DNA molecule, while the  output <code class="docutils literal notranslate"><span class="pre">-o</span></code> dna_cg.pdb preserves a few of them.
Please check both PDB structures using VMD:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>vmd<span class="w"> </span>-m<span class="w"> </span>./sirah.amber/tutorial/2/dna.pdb<span class="w"> </span>dna_cg.pdb
</pre></div>
</div>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>This is the basic usage of the script <strong>cgconv.pl</strong>, you can learn other capabilities from its help by typing:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>./sirah.amber/tools/CGCONV/cgconv.pl<span class="w"> </span>-h
</pre></div>
</div>
</div>
<p>From now on it is just normal AMBER stuff!</p>
</section>
<section id="id13">
<h3>2.2. Prepare LEaP input<a class="headerlink" href="#id13" title="Link to this heading"></a></h3>
<p>Use a text editor to create the file <code class="docutils literal notranslate"><span class="pre">gensystem.leap</span></code> including the following lines:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="gp"># </span>Load<span class="w"> </span>SIRAH<span class="w"> </span>force<span class="w"> </span>field
<span class="go">addPath ./sirah.amber</span>
<span class="go">source leaprc.sirah</span>

<span class="gp"># </span>Load<span class="w"> </span>model
<span class="go">dna = loadpdb dna_cg.pdb</span>

<span class="gp"># </span>Info<span class="w"> </span>on<span class="w"> </span>system<span class="w"> </span>charge
<span class="go">charge dna</span>

<span class="gp"># </span>Add<span class="w"> </span>solvent,<span class="w"> </span>counterions<span class="w"> </span>and<span class="w"> </span><span class="m">0</span>.15M<span class="w"> </span>NaCl
<span class="gp"># </span>Tuned<span class="w"> </span>solute-solvent<span class="w"> </span>closeness<span class="w"> </span><span class="k">for</span><span class="w"> </span>best<span class="w"> </span>hydration
<span class="go">solvateoct dna WT4BOX 20 0.7</span>
<span class="go">addionsrand dna NaW 88 ClW 50</span>

<span class="gp"># </span>Save<span class="w"> </span>Parms
<span class="go">saveAmberParmNetcdf dna dna_cg.prmtop dna_cg.ncrst</span>

<span class="gp"># </span>EXIT
<span class="go">quit</span>
</pre></div>
</div>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<p>The available electrolyte species in SIRAH force field are: <code class="docutils literal notranslate"><span class="pre">Na⁺</span></code> (NaW), <code class="docutils literal notranslate"><span class="pre">K⁺</span></code> (KW) and <code class="docutils literal notranslate"><span class="pre">Cl⁻</span></code> (ClW) which represent solvated ions in solution. One ion pair (e.g., NaW-ClW) each 34 WT4 molecules results in a salt concentration of ~0.15M (see <a class="reference internal" href="#appendix"><span class="std std-ref">Appendix</span></a> for details). Counterions were added according to <a class="reference external" href="https://pubs.acs.org/doi/10.1021/acs.jctc.9b00953">Machado et al.</a>.</p>
</div>
</section>
<section id="id14">
<h3>2.3. Run LEaP<a class="headerlink" href="#id14" title="Link to this heading"></a></h3>
<p>Run the LEaP application to generate the molecular topology and initial coordinate files:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>tleap<span class="w"> </span>-f<span class="w"> </span>gensystem.leap
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Warning messages about long, triangular or square bonds in <code class="docutils literal notranslate"><span class="pre">leap.log</span></code> file are fine and
expected due to the CG topology of some residues.</p>
</div>
<p>This should create a topology file <code class="docutils literal notranslate"><span class="pre">dna_cg.prmtop</span></code> and a coordinate file <code class="docutils literal notranslate"><span class="pre">dna_cg.ncrst</span></code>.</p>
<p>Use VMD to check how the CG model looks like:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>vmd<span class="w"> </span>dna_cg.prmtop<span class="w"> </span>dna_cg.ncrst<span class="w"> </span>-e<span class="w"> </span>./sirah.amber/tools/sirah_vmdtk.tcl
</pre></div>
</div>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>VMD assigns default radius to unknown atom types, the script <code class="docutils literal notranslate"><span class="pre">sirah_vmdtk.tcl</span></code> sets the right
ones, according to the CG representation. It also provides a kit of useful selection macros, coloring methods and backmapping utilities.
Use the command <code class="docutils literal notranslate"><span class="pre">sirah_help</span></code> in the Tcl/Tk console of VMD to access the manual pages. To learn about SIRAH Tools’ capabilities, you can also go to the <a class="reference internal" href="Tutorials%20sirahtools.html#sirah-tools"><span class="std std-ref">SIRAH Tools tutorial</span></a>.</p>
</div>
</section>
<section id="id15">
<h3>2.4. Run the simulation<a class="headerlink" href="#id15" title="Link to this heading"></a></h3>
<p>Make a new folder for the run:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>mkdir<span class="w"> </span>-p<span class="w"> </span>run<span class="p">;</span><span class="w"> </span><span class="nb">cd</span><span class="w"> </span>run
</pre></div>
</div>
<p>The folder <code class="docutils literal notranslate"><span class="pre">sirah.amber/tutorial/2/PMEMD/</span></code> contains typical input files for energy minimization
(<code class="docutils literal notranslate"><span class="pre">em_WT4.in</span></code>), equilibration (<code class="docutils literal notranslate"><span class="pre">eq_WT4.in</span></code>) and production (<code class="docutils literal notranslate"><span class="pre">md_WT4.in</span></code>) runs. Please check carefully the input flags therein, in particular the definition of flag <em>chngmask=0</em> at <em>&amp;ewald</em> section is <strong>mandatory</strong>.</p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<blockquote>
<div><p><strong>Some commonly used flags in AMBER</strong></p>
</div></blockquote>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">-i</span></code>: Input file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-o</span></code>: Output file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-p</span></code>: Parameter/topology file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-c</span></code>: Coordinate file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-r</span></code>: Restart file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-x</span></code>: Trajectory file.</p></li>
</ul>
</div>
<div class="admonition caution">
<p class="admonition-title">Caution</p>
<p>These input files are executed by the <strong>GPU</strong> implementation of <code class="docutils literal notranslate"><span class="pre">pmemd.cuda</span></code>. Other available modules are <code class="docutils literal notranslate"><span class="pre">sander</span></code>  or <code class="docutils literal notranslate"><span class="pre">pmemd</span></code>, which are both <strong>CPU</strong> implementations of Amber.</p>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>You can find example input files for CPU versions of <code class="docutils literal notranslate"><span class="pre">sander</span></code> and <code class="docutils literal notranslate"><span class="pre">pmemd</span></code> at folders <code class="docutils literal notranslate"><span class="pre">SANDER/</span></code> and  <code class="docutils literal notranslate"><span class="pre">PMEMD/</span></code>, within <code class="docutils literal notranslate"><span class="pre">sirah.amber/tutorial/2/</span></code></p>
</div>
<p><strong>Energy Minimization:</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/2/PMEMD/em_WT4.in<span class="w"> </span>-p<span class="w"> </span>../dna_cg.prmtop<span class="w"> </span>-c<span class="w"> </span>../dna_cg.ncrst<span class="w"> </span>-o<span class="w"> </span>dna_cg_em.out<span class="w"> </span>-r<span class="w"> </span>dna_cg_em.ncrst<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>In the course of long MD simulations the capping residues may eventually separate, this effect is
called helix fraying. To avoid such behavior is necessary to set Watson-Crick restraints for the capping base pairs of this CG DNA at the end of <code class="docutils literal notranslate"><span class="pre">eq_GB.in</span></code> and <code class="docutils literal notranslate"><span class="pre">md_GB.in</span></code> files. Check the files lines that start with <em>&amp;rst</em>.</p>
</div>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>If you are using <code class="docutils literal notranslate"><span class="pre">sander</span></code>, to avoid the helix fraying, you must create a symbolic link to the file <code class="docutils literal notranslate"><span class="pre">dna_cg.RST</span></code>, which
contains the definition of Watson-Crick restraints for the capping base pairs of this CG DNA:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>ln<span class="w"> </span>-s<span class="w"> </span>../sirah.amber/tutorial/1/SANDER/dna_cg.RST
</pre></div>
</div>
<p>The file <code class="docutils literal notranslate"><span class="pre">dna_cg.RST</span></code> can only be read by <code class="docutils literal notranslate"><span class="pre">sander</span></code>, <code class="docutils literal notranslate"><span class="pre">pmemd</span></code> reads a different restrain format.</p>
</div>
<p><strong>Equilibration (NPT):</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/2/PMEMD/eq_WT4.in<span class="w"> </span>-p<span class="w"> </span>../dna_cg.prmtop<span class="w"> </span>-c<span class="w"> </span>dna_cg_em.ncrst<span class="w"> </span>-ref<span class="w"> </span>dna_cg_em.ncrst<span class="w"> </span>-o<span class="w"> </span>dna_cg_eq.out<span class="w"> </span>-r<span class="w"> </span>dna_cg_eq.ncrst<span class="w"> </span>-x<span class="w"> </span>dna_cg_eq.nc<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
<p><strong>Production (100ns):</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/2/PMEMD/md_WT4.in<span class="w"> </span>-p<span class="w"> </span>../dna_cg.prmtop<span class="w"> </span>-c<span class="w"> </span>dna_cg_eq.ncrst<span class="w"> </span>-o<span class="w"> </span>dna_cg_md.out<span class="w"> </span>-r<span class="w"> </span>dna_cg_md.ncrst<span class="w"> </span>-x<span class="w"> </span>dna_cg_md.nc<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
</section>
<section id="id16">
<h3>2.5. Visualizing the simulation<a class="headerlink" href="#id16" title="Link to this heading"></a></h3>
<p>That’s it! Now you can analyze the trajectory.
First, process the output trajectory to account for the Periodic Boundary Conditions (PBC):</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;autoimage\ngo\nquit\n&quot;</span><span class="w"> </span><span class="p">|</span><span class="w"> </span>cpptraj<span class="w"> </span>-p<span class="w"> </span>../dna_cg.prmtop<span class="w"> </span>-y<span class="w"> </span>dna_cg_md.nc<span class="w"> </span>-x<span class="w"> </span>dna_cg_md_pbc.nc<span class="w"> </span>--interactive
</pre></div>
</div>
<p>Then, load the processed trajectory in VMD:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">vmd</span> <span class="o">../</span><span class="n">dna_cg</span><span class="o">.</span><span class="n">prmtop</span> <span class="o">../</span><span class="n">dna_cg</span><span class="o">.</span><span class="n">ncrst</span> <span class="n">dna_cg_md</span><span class="o">.</span><span class="n">nc</span> <span class="o">-</span><span class="n">e</span> <span class="o">../</span><span class="n">sirah</span><span class="o">.</span><span class="n">amber</span><span class="o">/</span><span class="n">tools</span><span class="o">/</span><span class="n">sirah_vmdtk</span><span class="o">.</span><span class="n">tcl</span>
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The file <code class="docutils literal notranslate"><span class="pre">sirah_vmdtk.tcl</span></code> is a Tcl script that is part of SIRAH Tools and contains the macros to properly visualize the coarse-grained structures in VMD. Use the command <code class="docutils literal notranslate"><span class="pre">sirah-help</span></code> in the Tcl/Tk console of VMD to access the manual pages. To learn about SIRAH Tools’ capabilities, you can also go to the <a class="reference internal" href="Tutorials%20sirahtools.html#sirah-tools"><span class="std std-ref">SIRAH Tools tutorial</span></a>.</p>
</div>
</section>
</section>
<section id="multiscale-simulation-in-implicit-solvent">
<span id="tutorial-3"></span><h2>3. Multiscale simulation in implicit solvent<a class="headerlink" href="#multiscale-simulation-in-implicit-solvent" title="Link to this heading"></a></h2>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Please report bugs, errors or enhancement requests through <a class="reference external" href="https://github.com/SIRAHFF/documentation/issues">Issue Tracker</a> or if you have a question about SIRAH open a <a class="reference external" href="https://github.com/SIRAHFF/documentation/discussions">New Discussion</a>.</p>
</div>
<p>This tutorial shows how to apply a multiscale representation of DNA, which is available in the SIRAH
force field, to study protein-DNA interactions. In this approach, the protein and its binding region are
treated with an atomistic force field, while the contextual DNA is represented at coarse-grained (CG)
level. The simulated system is composed of the human TATA binding protein (TBP PDB: <a class="reference external" href="https://www.rcsb.org/structure/1c9b">1C9B</a>) bounded to the TATA box at the promoter region (-64 to +13) of the Human Immunodeficiency Virus type 1 (HIV-1, GenBank: <a class="reference external" href="https://www.ncbi.nlm.nih.gov/nuccore/K03455.1">K03455</a>, <strong>Figure 1</strong>). Solvent effects are considered implicitly using the Generalized Born model (GB).</p>
<p>The main references for this tutorial are: <a class="reference external" href="https://pubs.rsc.org/en/Content/ArticleLanding/2011/CP/c1cp21248f">Machado et al.</a> and <a class="reference external" href="https://academic.oup.com/bioinformatics/article/32/10/1568/1743152">Machado &amp; Pantano</a>. We strongly advise you to read these articles before starting the tutorial.</p>
<figure class="align-center" id="id89">
<a class="reference internal image-reference" href="_images/Tuto3.png"><img alt="map to buried treasure" src="_images/Tuto3.png" style="width: 100%;" />
</a>
<figcaption>
<p><span class="caption-text"><strong>Figure 1.</strong> Promoter region of HIV-1. The TATA box is highlighted in orange, while base pairs at 0.7nm of TBP are colored in yellow.</span><a class="headerlink" href="#id89" title="Link to this image"></a></p>
</figcaption>
</figure>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>Check the <a class="reference internal" href="#download-amber"><span class="std std-ref">Setting up SIRAH</span></a> section for download and set up details before starting this tutorial.
Since this is <strong>Tutorial 3</strong>, remember to replace <code class="docutils literal notranslate"><span class="pre">X.X</span></code> and the files corresponding to this tutorial can be found in: <code class="docutils literal notranslate"><span class="pre">sirah_[version].amber/tutorial/3/</span></code></p>
</div>
<section id="id19">
<h3>3.1. Build CG representations<a class="headerlink" href="#id19" title="Link to this heading"></a></h3>
<p>Map only the atomistic structure of nucleotides outside the TATA box of the HIV-1 promoter to CG:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>./sirah.amber/tools/CGCONV/cgconv.pl<span class="w"> </span>-i<span class="w"> </span>./sirah.amber/tutorial/3/tbphiv.pdb<span class="w"> </span>-R<span class="w"> </span><span class="m">1</span>-32,45-110,123-154<span class="w"> </span>-o<span class="w"> </span>dna_hyb.pdb
</pre></div>
</div>
<p>The input file <code class="docutils literal notranslate"><span class="pre">-i</span></code> tbphiv.pdb contains all the heavy atoms composing the protein-DNA system. Mapped
residues are selected through option <code class="docutils literal notranslate"><span class="pre">-R</span></code>. The selection considers a “buffer region” of two base pairs at each
side of the TATA box (<strong>Figure 1</strong>). The atomistic/CG interface must always be a step of B form DNA. The
resulting coordinates are saved in the output <code class="docutils literal notranslate"><span class="pre">-o</span></code> dna_hyb.pdb.</p>
<p>Please check both PDB structures in VMD:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>vmd<span class="w"> </span>-m<span class="w"> </span>./sirah.amber/tutorial/3/tbphiv.pdb<span class="w"> </span>dna_hyb.pdb
</pre></div>
</div>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>This is the basic usage of the script <strong>cgconv.pl</strong>, you can learn other capabilities from its help by typing:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>./sirah.amber/tools/CGCONV/cgconv.pl<span class="w"> </span>-h
</pre></div>
</div>
</div>
<p>From now on it is just normal AMBER stuff!</p>
</section>
<section id="id20">
<h3>3.2. Prepare LEaP input<a class="headerlink" href="#id20" title="Link to this heading"></a></h3>
<p>Use a text editor to create the file <code class="docutils literal notranslate"><span class="pre">gensystem.leap</span></code> including the following lines:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="gp"># </span>Load<span class="w"> </span>AMBER<span class="w"> </span>force<span class="w"> </span>field<span class="w"> </span><span class="o">(</span>parm14SB/parmbsc0<span class="o">)</span>
<span class="go">source oldff/leaprc.ff14SB</span>

<span class="gp"># </span>Load<span class="w"> </span>SIRAH<span class="w"> </span>force<span class="w"> </span>field
<span class="go">addPath ./sirah.amber</span>
<span class="go">source leaprc.sirah</span>

<span class="gp"># </span>Load<span class="w"> </span>model
<span class="go">dna = loadpdb dna_hyb.pdb</span>

<span class="gp"># </span>Save<span class="w"> </span>Parms
<span class="go">saveAmberParmNetcdf dna dna_hyb.prmtop dna_hyb.ncrst</span>

<span class="gp"># </span>EXIT
<span class="go">quit</span>
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<ul class="simple">
<li><p>According to AMBER version 10, 11 or 12 (14) the source file for parm99SB/bsc0 is leaprc.ff99bsc0, leaprc.ff10 or leaprc.ff12SB respectively.</p></li>
<li><p>The newest force field version for DNA, <a class="reference external" href="https://www.nature.com/articles/nmeth.3658">bsc1</a>, works equally well with SIRAH.</p></li>
</ul>
</div>
</section>
<section id="id21">
<h3>3.3. Run LEaP<a class="headerlink" href="#id21" title="Link to this heading"></a></h3>
<p>Run the LEaP application to generate the molecular topology and initial coordinate files:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>tleap<span class="w"> </span>-f<span class="w"> </span>gensystem.leap
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Warning messages about long, triangular or square bonds in <code class="docutils literal notranslate"><span class="pre">leap.log</span></code> file are fine and
expected due to the CG topology of some residues.</p>
</div>
<p>This should create a topology file <code class="docutils literal notranslate"><span class="pre">dna_hyb.prmtop</span></code> and a coordinate file <code class="docutils literal notranslate"><span class="pre">dna_hyb.ncrst</span></code>.</p>
<p>Use VMD to check how the multiscale model looks like:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>vmd<span class="w"> </span>dna_hyb.prmtop<span class="w"> </span>dna_hyb.ncrst<span class="w"> </span>-e<span class="w"> </span>./sirah.amber/tools/sirah_vmdtk.tcl
</pre></div>
</div>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>VMD assigns default radius to unknown atom types, the script <code class="docutils literal notranslate"><span class="pre">sirah_vmdtk.tcl</span></code> sets the right
ones, according to the CG representation. It also provides a kit of useful selection macros, coloring methods and backmapping utilities.
Use the command <code class="docutils literal notranslate"><span class="pre">sirah_help</span></code> in the Tcl/Tk console of VMD to access the manual pages. To learn about SIRAH Tools’ capabilities, you can also go to the <a class="reference internal" href="Tutorials%20sirahtools.html#sirah-tools"><span class="std std-ref">SIRAH Tools tutorial</span></a>.</p>
</div>
</section>
<section id="id22">
<h3>3.4. Run the simulation<a class="headerlink" href="#id22" title="Link to this heading"></a></h3>
<p>Make a new folder for the run:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>mkdir<span class="w"> </span>-p<span class="w"> </span>run<span class="p">;</span><span class="w"> </span><span class="nb">cd</span><span class="w"> </span>run
</pre></div>
</div>
<p>The folder <code class="docutils literal notranslate"><span class="pre">sirah.amber/tutorial/3/PMEMD.GPU/</span></code> contains typical input files for energy minimization
(<code class="docutils literal notranslate"><span class="pre">em_HYB.in</span></code>), equilibration (<code class="docutils literal notranslate"><span class="pre">eq_HYB.in</span></code>) and production (<code class="docutils literal notranslate"><span class="pre">md_HYB.in</span></code>) runs. Please check carefully the input flags therein.</p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<blockquote>
<div><p><strong>Some commonly used flags in AMBER</strong></p>
</div></blockquote>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">-i</span></code>: Input file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-o</span></code>: Output file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-p</span></code>: Parameter/topology file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-c</span></code>: Coordinate file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-r</span></code>: Restart file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-x</span></code>: Trajectory file.</p></li>
</ul>
</div>
<div class="admonition caution">
<p class="admonition-title">Caution</p>
<p>These input files are executed by the <strong>GPU</strong> implementation of <code class="docutils literal notranslate"><span class="pre">pmemd.cuda</span></code>. Other available modules are <code class="docutils literal notranslate"><span class="pre">sander</span></code>  or <code class="docutils literal notranslate"><span class="pre">pmemd</span></code>, which are both <strong>CPU</strong> implementations of Amber.</p>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>You can find example input files for CPU versions of <code class="docutils literal notranslate"><span class="pre">sander</span></code> and <code class="docutils literal notranslate"><span class="pre">pmemd</span></code> at folders <code class="docutils literal notranslate"><span class="pre">SANDER/</span></code> and  <code class="docutils literal notranslate"><span class="pre">PMEMD.CPU/</span></code>, within <code class="docutils literal notranslate"><span class="pre">sirah.amber/tutorial/3/</span></code></p>
<p>However, this simulation is time consuming owing to the system’s size. A parallel or CUDA implementation of Amber is advised.</p>
</div>
<p><strong>Energy Minimization:</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/3/PMEMD.GPU/em_HYB.in<span class="w"> </span>-p<span class="w"> </span>../dna_hyb.prmtop<span class="w"> </span>-c<span class="w"> </span>../dna_hyb.ncrst<span class="w"> </span>-o<span class="w"> </span>dna_hyb_em.out<span class="w"> </span>-r<span class="w"> </span>dna_hyb_em.ncrst<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>In the course of long MD simulations the capping residues may eventually separate, this effect is
called helix fraying. To avoid such behavior is necessary to set Watson-Crick restraints for the capping base pairs of this CG DNA at the end of <code class="docutils literal notranslate"><span class="pre">eq_GB.in</span></code> and <code class="docutils literal notranslate"><span class="pre">md_GB.in</span></code> files. Check the files lines that start with <em>&amp;rst</em>.</p>
</div>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>If you are using <code class="docutils literal notranslate"><span class="pre">sander</span></code>, to avoid the helix fraying, you must create a symbolic link to the file <code class="docutils literal notranslate"><span class="pre">dna_cg.RST</span></code>, which
contains the definition of Watson-Crick restraints for the capping base pairs of this CG DNA:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>ln<span class="w"> </span>-s<span class="w"> </span>../sirah.amber/tutorial/3/SANDER/dna_hyb.RST
</pre></div>
</div>
<p>The file <code class="docutils literal notranslate"><span class="pre">dna_cg.RST</span></code> can only be read by <code class="docutils literal notranslate"><span class="pre">sander</span></code>, <code class="docutils literal notranslate"><span class="pre">pmemd</span></code> reads a different restrain format.</p>
</div>
<p><strong>Equilibration:</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/3/PMEMD.GPU/eq_HYB.in<span class="w"> </span>-p<span class="w"> </span>../dna_hyb.prmtop<span class="w"> </span>-c<span class="w"> </span>dna_hyb_em.ncrst<span class="w"> </span>-o<span class="w"> </span>dna_hyb_eq.out<span class="w"> </span>-r<span class="w"> </span>dna_hyb_eq.ncrst<span class="w"> </span>-x<span class="w"> </span>dna_hyb_eq.nc<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
<p><strong>Production (10ns):</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/3/PMEMD.GPU/md_HYB.in<span class="w"> </span>-p<span class="w"> </span>../dna_hyb.prmtop<span class="w"> </span>-c<span class="w"> </span>dna_hyb_eq.ncrst<span class="w"> </span>-o<span class="w"> </span>dna_hyb_md.out<span class="w"> </span>-r<span class="w"> </span>dna_hyb_md.ncrst<span class="w"> </span>-x<span class="w"> </span>dna_hyb_md.nc<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
</section>
<section id="id23">
<h3>3.5. Visualizing the simulation<a class="headerlink" href="#id23" title="Link to this heading"></a></h3>
<p>That’s it! Now you can load, visualize and analize the trajectory file in VMD:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">vmd</span> <span class="o">../</span><span class="n">dna_hyb</span><span class="o">.</span><span class="n">prmtop</span> <span class="o">../</span><span class="n">dna_hyb</span><span class="o">.</span><span class="n">ncrst</span> <span class="n">dna_hyb_md</span><span class="o">.</span><span class="n">nc</span> <span class="o">-</span><span class="n">e</span> <span class="o">../</span><span class="n">sirah</span><span class="o">.</span><span class="n">amber</span><span class="o">/</span><span class="n">tools</span><span class="o">/</span><span class="n">sirah_vmdtk</span><span class="o">.</span><span class="n">tcl</span>
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The file <code class="docutils literal notranslate"><span class="pre">sirah_vmdtk.tcl</span></code> is a Tcl script that is part of SIRAH Tools and contains the macros to properly visualize the coarse-grained structures in VMD. Use the command <code class="docutils literal notranslate"><span class="pre">sirah-help</span></code> in the Tcl/Tk console of VMD to access the manual pages. To learn about SIRAH Tools’ capabilities, you can also go to the <a class="reference internal" href="Tutorials%20sirahtools.html#sirah-tools"><span class="std std-ref">SIRAH Tools tutorial</span></a>.</p>
</div>
</section>
</section>
<section id="closed-circular-dna-in-implicit-solvent">
<span id="tutorial-4"></span><h2>4. Closed circular DNA in implicit solvent<a class="headerlink" href="#closed-circular-dna-in-implicit-solvent" title="Link to this heading"></a></h2>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Please report bugs, errors or enhancement requests through <a class="reference external" href="https://github.com/SIRAHFF/documentation/issues">Issue Tracker</a> or if you have a question about SIRAH open a <a class="reference external" href="https://github.com/SIRAHFF/documentation/discussions">New Discussion</a>.</p>
</div>
<p>This tutorial shows how to use the SIRAH force field to perform a coarse grained (CG) simulation of a
closed circular DNA using the Generalized Born model (GB) for implicit solvent.</p>
<p>The main references for this tutorial are: <a class="reference external" href="https://pubs.acs.org/doi/abs/10.1021/ct900653p">Dans et al.</a> (latest parameters are those reported <a class="reference external" href="https://pubs.acs.org/doi/abs/10.1021/ct100379f">here</a>) and <a class="reference external" href="https://academic.oup.com/bioinformatics/article/32/10/1568/1743152">Machado &amp; Pantano</a>. We strongly advise you to read these articles before starting the tutorial.</p>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>Check the <a class="reference internal" href="#download-amber"><span class="std std-ref">Setting up SIRAH</span></a> section for download and set up details before starting this tutorial.
Since this is <strong>Tutorial 4</strong>, remember to replace <code class="docutils literal notranslate"><span class="pre">X.X</span></code> and the files corresponding to this tutorial can be found in: <code class="docutils literal notranslate"><span class="pre">sirah_[version].amber/tutorial/4/</span></code></p>
</div>
<section id="id29">
<h3>4.1. Build CG representations<a class="headerlink" href="#id29" title="Link to this heading"></a></h3>
<p>Map the atomistic structure of the closed circular DNA to its CG representation:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>./sirah.amber/tools/CGCONV/cgconv.pl<span class="w"> </span>-i<span class="w"> </span>./sirah.amber/tutorial/4/ccdna.pdb<span class="w"> </span><span class="p">|</span><span class="w"> </span>sed<span class="w"> </span>-e<span class="w"> </span><span class="s1">&#39;s/DCX A   1/CW5 A   1/&#39;</span><span class="w">  </span>-e<span class="w"> </span><span class="s1">&#39;s/DCX B   1/CW5 B   1/&#39;</span><span class="w"> </span>-e<span class="w"> </span><span class="s1">&#39;s/DGX A 100/GW3 A 100/&#39;</span><span class="w"> </span>-e<span class="w"> </span><span class="s1">&#39;s/DGX B 100/GW3 B 100/&#39;</span><span class="w"> </span>&gt;<span class="w"> </span>ccdna_cg.pdb
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The 5’ and 3’ end residues must be renamed to AW5, TW5, GW5 or CW5 and AW3, TW3, GW3 or CW3 to represent the corresponding Adenine, Thymine, Guanine or Cytosine extremes in a closed circular DNA.</p>
</div>
<p>The input file <code class="docutils literal notranslate"><span class="pre">-i</span></code> ccdna.pdb contains all the heavy atoms composing the DNA molecule, while the output <code class="docutils literal notranslate"><span class="pre">-o</span></code> ccdna_cg.pdb preserves a few of them.</p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>This is the basic usage of the script <strong>cgconv.pl</strong>, you can learn other capabilities from its help by typing:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>./sirah.amber/tools/CGCONV/cgconv.pl<span class="w"> </span>-h
</pre></div>
</div>
</div>
<p>Please check both PDB structures using VMD:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>vmd<span class="w"> </span>-m<span class="w"> </span>./sirah.amber/tutorial/4/ccdna.pdb<span class="w"> </span>ccdna_cg.pdb
</pre></div>
</div>
<p>From now on it is just normal AMBER stuff!</p>
</section>
<section id="id30">
<h3>4.2. Prepare LEaP input<a class="headerlink" href="#id30" title="Link to this heading"></a></h3>
<p>Use a text editor to create the file <code class="docutils literal notranslate"><span class="pre">gensystem.leap</span></code> including the following lines:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="gp"># </span>Load<span class="w"> </span>SIRAH<span class="w"> </span>force<span class="w"> </span>field
<span class="go">addPath ./sirah.amber</span>
<span class="go">source leaprc.sirah</span>

<span class="gp"># </span>Load<span class="w"> </span>model
<span class="go">dna = loadpdb ccdna_cg.pdb</span>

<span class="gp"># </span>Make<span class="w"> </span>a<span class="w"> </span>covalently<span class="w"> </span>closed<span class="w"> </span>circular<span class="w"> </span>DNA
<span class="go">bond dna.1.PX dna.100.C1X</span>
<span class="go">bond dna.101.PX dna.200.C1X</span>

<span class="gp"># </span>Save<span class="w"> </span>Parms
<span class="go">saveAmberParmNetcdf dna ccdna_cg.prmtop ccdna_cg.ncrst</span>

<span class="gp"># </span>EXIT
<span class="go">quit</span>
</pre></div>
</div>
</section>
<section id="id31">
<h3>4.3. Run LEaP<a class="headerlink" href="#id31" title="Link to this heading"></a></h3>
<p>Run the LEaP application to generate the molecular topology and initial coordinate files:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>tleap<span class="w"> </span>-f<span class="w"> </span>gensystem.leap
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Warning messages about long, triangular or square bonds in <code class="docutils literal notranslate"><span class="pre">leap.log</span></code> file are fine and
expected due to the CG topology of some residues.</p>
</div>
<p>This should create a topology file <code class="docutils literal notranslate"><span class="pre">ccdna_cg.prmtop</span></code> and a coordinate file <code class="docutils literal notranslate"><span class="pre">ccdna_cg.ncrst</span></code>.</p>
<p>Use VMD to check how the CG model looks like:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">vmd</span> <span class="n">ccdna_cg</span><span class="o">.</span><span class="n">prmtop</span> <span class="n">ccdna_cg</span><span class="o">.</span><span class="n">ncrst</span> <span class="o">-</span><span class="n">e</span> <span class="o">./</span><span class="n">sirah</span><span class="o">.</span><span class="n">amber</span><span class="o">/</span><span class="n">tools</span><span class="o">/</span><span class="n">sirah_vmdtk</span><span class="o">.</span><span class="n">tcl</span>
</pre></div>
</div>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>VMD assigns default radius to unknown atom types, the script <code class="docutils literal notranslate"><span class="pre">sirah_vmdtk.tcl</span></code> sets the right
ones, according to the CG representation. It also provides a kit of useful selection macros, coloring methods and backmapping utilities.
Use the command <code class="docutils literal notranslate"><span class="pre">sirah_help</span></code> in the Tcl/Tk console of VMD to access the manual pages. To learn about SIRAH Tools’ capabilities, you can also go to the <a class="reference internal" href="Tutorials%20sirahtools.html#sirah-tools"><span class="std std-ref">SIRAH Tools tutorial</span></a>.</p>
</div>
</section>
<section id="id32">
<h3>4.4. Run the simulation<a class="headerlink" href="#id32" title="Link to this heading"></a></h3>
<p>Make a new folder for the run:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>mkdir<span class="w"> </span>-p<span class="w"> </span>run<span class="p">;</span><span class="w"> </span><span class="nb">cd</span><span class="w"> </span>run
</pre></div>
</div>
<p>The folder <code class="docutils literal notranslate"><span class="pre">sirah.amber/tutorial/4/GPU</span></code> contains typical input files for energy minimization (<code class="docutils literal notranslate"><span class="pre">em_GBSA.in</span></code>), equilibration (<code class="docutils literal notranslate"><span class="pre">eq_GBSA.in</span></code>) and production (<code class="docutils literal notranslate"><span class="pre">md_GBSA.in</span></code>) runs. Please check carefully the input flags therein.</p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<blockquote>
<div><p><strong>Some commonly used flags in AMBER</strong></p>
</div></blockquote>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">-i</span></code>: Input file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-o</span></code>: Output file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-p</span></code>: Parameter/topology file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-c</span></code>: Coordinate file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-r</span></code>: Restart file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-x</span></code>: Trajectory file.</p></li>
</ul>
</div>
<div class="admonition caution">
<p class="admonition-title">Caution</p>
<p>These input files are executed by the <strong>GPU</strong> implementation of <code class="docutils literal notranslate"><span class="pre">pmemd.cuda</span></code>. Other available modules are <code class="docutils literal notranslate"><span class="pre">sander</span></code>  or <code class="docutils literal notranslate"><span class="pre">pmemd</span></code>, which are both <strong>CPU</strong> implementations of Amber.</p>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>You can find example input files for CPU versions of <code class="docutils literal notranslate"><span class="pre">sander</span></code> and <code class="docutils literal notranslate"><span class="pre">pmemd</span></code> at folder <code class="docutils literal notranslate"><span class="pre">CPU/</span></code>, within <code class="docutils literal notranslate"><span class="pre">sirah.amber/tutorial/4/</span></code></p>
</div>
<p><strong>Energy Minimization:</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/4/GPU/em_GBSA.in<span class="w"> </span>-p<span class="w"> </span>../ccdna_cg.prmtop<span class="w"> </span>-c<span class="w"> </span>../ccdna_cg.ncrst<span class="w"> </span>-o<span class="w"> </span>ccdna_cg_em.out<span class="w"> </span>-r<span class="w"> </span>ccdna_cg_em.ncrst<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
<p><strong>Equilibration:</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/4/GPU/eq_GBSA.in<span class="w"> </span>-p<span class="w"> </span>../ccdna_cg.prmtop<span class="w"> </span>-c<span class="w"> </span>ccdna_cg_em.ncrst<span class="w"> </span>-o<span class="w"> </span>ccdna_cg_eq.out<span class="w"> </span>-r<span class="w"> </span>ccdna_cg_eq.ncrst<span class="w"> </span>-x<span class="w"> </span>ccdna_cg_eq.nc<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
<p><strong>Production (100ns):</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/4/GPU/md_GBSA.in<span class="w"> </span>-p<span class="w"> </span>../ccdna_cg.prmtop<span class="w"> </span>-c<span class="w"> </span>ccdna_cg_eq.ncrst<span class="w"> </span>-o<span class="w"> </span>ccdna_cg_md.out<span class="w"> </span>-r<span class="w"> </span>ccdna_cg_md.ncrst<span class="w"> </span>-x<span class="w"> </span>ccdna_cg_md.nc<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
</section>
<section id="id33">
<h3>4.5. Visualizing the simulation<a class="headerlink" href="#id33" title="Link to this heading"></a></h3>
<p>That’s it! Now you can load, visualize and analize the trajectory file in VMD:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">vmd</span> <span class="o">../</span><span class="n">ccdna_cg</span><span class="o">.</span><span class="n">prmtop</span> <span class="o">../</span><span class="n">ccdna_cg</span><span class="o">.</span><span class="n">ncrst</span> <span class="n">ccdna_cg_md</span><span class="o">.</span><span class="n">nc</span> <span class="o">-</span><span class="n">e</span> <span class="o">../</span><span class="n">sirah</span><span class="o">.</span><span class="n">amber</span><span class="o">/</span><span class="n">tools</span><span class="o">/</span><span class="n">sirah_vmdtk</span><span class="o">.</span><span class="n">tcl</span>
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The file <code class="docutils literal notranslate"><span class="pre">sirah_vmdtk.tcl</span></code> is a Tcl script that is part of SIRAH Tools and contains the macros to properly visualize the coarse-grained structures in VMD. Use the command <code class="docutils literal notranslate"><span class="pre">sirah-help</span></code> in the Tcl/Tk console of VMD to access the manual pages. To learn about SIRAH Tools’ capabilities, you can also go to the <a class="reference internal" href="Tutorials%20sirahtools.html#sirah-tools"><span class="std std-ref">SIRAH Tools tutorial</span></a>.</p>
</div>
</section>
</section>
<section id="proteins-in-explicit-solvent">
<span id="tutorial-5"></span><h2>5. Proteins in explicit solvent<a class="headerlink" href="#proteins-in-explicit-solvent" title="Link to this heading"></a></h2>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Please report bugs, errors or enhancement requests through <a class="reference external" href="https://github.com/SIRAHFF/documentation/issues">Issue Tracker</a> or if you have a question about SIRAH open a <a class="reference external" href="https://github.com/SIRAHFF/documentation/discussions">New Discussion</a>.</p>
</div>
<p>This tutorial shows how to use the SIRAH force field to perform a coarse grained (CG) simulation of a
protein in explicit solvent (called WatFour, WT4). The main references for
this tutorial are: <a class="reference external" href="https://pubs.acs.org/doi/abs/10.1021/ct100379f">Darré et al.</a>, <a class="reference external" href="https://doi.org/10.1021/acs.jctc.9b00006">Machado et al.</a> and <a class="reference external" href="https://academic.oup.com/bioinformatics/article/32/10/1568/1743152">Machado &amp; Pantano</a>. We strongly advise you to read these articles before starting the tutorial.</p>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>Check the <a class="reference internal" href="#download-amber"><span class="std std-ref">Setting up SIRAH</span></a> section for download and set up details before starting this tutorial.
Since this is <strong>Tutorial 5</strong>, remember to replace <code class="docutils literal notranslate"><span class="pre">X.X</span></code> and the files corresponding to this tutorial can be found in: <code class="docutils literal notranslate"><span class="pre">sirah_[version].amber/tutorial/5/</span></code></p>
</div>
<section id="id37">
<h3>5.1. Build CG representations<a class="headerlink" href="#id37" title="Link to this heading"></a></h3>
<div class="admonition caution">
<p class="admonition-title">Caution</p>
<p>The mapping to CG requires the correct protonation state of each residue at a given pH. We recommend using the <a class="reference external" href="https://www.charmm-gui.org/">CHARMM-GUI server</a> and use the <strong>PDB Reader &amp; Manipulator</strong> to prepare your system. An account is required to access any of the CHARMM-GUI Input Generator modules, and it can take up to 24 hours to obtain one.</p>
<p>Other option is the <a class="reference external" href="https://server.poissonboltzmann.org/pdb2pqr">PDB2PQR server</a> and choosing the output naming scheme of AMBER for best compatibility. This server was utilized to generate the <em>PQR</em> file featured in this tutorial. Be aware that modified residues lacking parameters such as: MSE (seleno MET), TPO (phosphorylated TYR), SEP (phosphorylated SER) or others are deleted from the PQR file by the server. In that case, mutate the residues to their unmodified form before submitting the structure to the server.</p>
</div>
<p>Map the protonated atomistic structure of protein <a class="reference external" href="https://www.rcsb.org/structure/1CRN">1CRN</a> to its CG representation:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>./sirah.amber/tools/CGCONV/cgconv.pl<span class="w"> </span>-i<span class="w"> </span>./sirah.amber/tutorial/5/1CRN.pqr<span class="w"> </span>-o<span class="w"> </span>1CRN_cg.pdb
</pre></div>
</div>
<p>The input file <code class="docutils literal notranslate"><span class="pre">-i</span></code> 1CRN.pqr contains the atomistic representation of <a class="reference external" href="https://www.rcsb.org/structure/1CRN">1CRN</a> structure at pH <strong>7.0</strong>, while the output <code class="docutils literal notranslate"><span class="pre">-o</span></code> 1CRN_cg.pdb is its SIRAH CG representation.</p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>This is the basic usage of the script <strong>cgconv.pl</strong>, you can learn other capabilities from its help by typing:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>./sirah.amber/tools/CGCONV/cgconv.pl<span class="w"> </span>-h
</pre></div>
</div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p><strong>Pay attention to residue names when mapping structures from other atomistic force fields or experimental structures.</strong> Although we provide compatibility for naming schemes in PDB, GMX, GROMOS, CHARMM and OPLS, there might always be some ambiguity in the residue naming, specially regarding protonation states, that may lead to a wrong mapping. For example, SIRAH Tools always maps the residue name “HIS” to a Histidine protonated at the epsilon nitrogen (<span class="math notranslate nohighlight">\(N_{\epsilon}\)</span>) regardless the actual proton placement. Similarly, protonated Glutamic and Aspartic acid residues must be named “GLH” and “ASH”, otherwise they will be treated as negative charged residues. In addition, protonated and disulfide bonded Cysteines must be named “CYS” and “CYX” respectively. These kind of situations need to be carefully checked by the users. In all cases the residues preserve their identity when mapping and back-mapping the structures. Hence, the total charge of the protein should be the same at atomistic and SIRAH levels. You can check the following mapping file to be sure of the compatibility: <code class="docutils literal notranslate"><span class="pre">sirah.amber/tools/CGCONV/maps/sirah_prot.map</span></code>.</p>
</div>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>By default, charged termini are used, but it is possible to set them neutral by renaming the residues from <strong>s</strong>[code] to <strong>a</strong>[code] (Nt-acetylated) or <strong>m</strong>[code] (Ct-amidated) after mapping to CG, where [code] is the root residue name in SIRAH. For example, to set a neutral N-terminal Histidine protonated at epsilon nitrogen (<span class="math notranslate nohighlight">\(N_{\epsilon}\)</span>) rename it from “sHe” to “aHe”.</p>
</div>
<p>Please check both PDB and PQR structures using VMD:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>vmd<span class="w"> </span>-m<span class="w"> </span>sirah.amber/tutorial/5/1CRN.pqr<span class="w"> </span>1CRN_cg.pdb
</pre></div>
</div>
<p>From now on it is just normal Amber stuff!</p>
</section>
<section id="id39">
<h3>5.2. Prepare LEaP input<a class="headerlink" href="#id39" title="Link to this heading"></a></h3>
<p>Use a text editor to create the file <code class="docutils literal notranslate"><span class="pre">gensystem.leap</span></code> including the following lines:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="gp"># </span>Load<span class="w"> </span>SIRAH<span class="w"> </span>force<span class="w"> </span>field
<span class="go">addPath ./sirah.amber</span>
<span class="go">source leaprc.sirah</span>

<span class="gp"># </span>Load<span class="w"> </span>model
<span class="go">protein = loadpdb 1CRN_cg.pdb</span>

<span class="gp"># </span>Info<span class="w"> </span>on<span class="w"> </span>system<span class="w"> </span>charge
<span class="go">charge protein</span>

<span class="gp"># </span>Set<span class="w"> </span>S-S<span class="w"> </span>bridges
<span class="go">bond protein.3.BSG protein.40.BSG</span>
<span class="go">bond protein.4.BSG protein.32.BSG</span>
<span class="go">bond protein.16.BSG protein.26.BSG</span>

<span class="gp"># </span>Add<span class="w"> </span>solvent,<span class="w"> </span>counterions<span class="w"> </span>and<span class="w"> </span><span class="m">0</span>.15M<span class="w"> </span>NaCl
<span class="gp"># </span>Tuned<span class="w"> </span>solute-solvent<span class="w"> </span>closeness<span class="w"> </span><span class="k">for</span><span class="w"> </span>best<span class="w"> </span>hydration
<span class="go">solvateOct protein WT4BOX 20 0.7</span>
<span class="go">addIonsRand protein NaW 22 ClW 22</span>

<span class="gp"># </span>Save<span class="w"> </span>Parms
<span class="go">saveAmberParmNetcdf protein 1CRN_cg.prmtop 1CRN_cg.ncrst</span>

<span class="gp"># </span>EXIT
<span class="go">quit</span>
</pre></div>
</div>
<div class="admonition caution">
<p class="admonition-title">Caution</p>
<p>Each disulfide bond must be defined explicitly in LEaP using the command bond, e.g.: “<em>bond unit.ri.BSG unit.rj.BSG</em>”. Where <em>ri</em> and <em>rj</em> correspond to the residue index in the topology file starting from 1, which may differ from the biological sequence in the PDB file. You can try the command <em>pdb4amber</em> to get those indexes from the atomistic structure, but be aware that it may not work if the Cysteine residues are too far away:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pdb4amber<span class="w"> </span>-i<span class="w"> </span>sirah.amber/tutorial/5/1CRN.pqr<span class="w"> </span>-o<span class="w"> </span>1CRN_aa.pdb<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span>cat<span class="w"> </span>1CRN_aa_sslink
</pre></div>
</div>
</div>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<p>The available electrolyte species in SIRAH force field are: <code class="docutils literal notranslate"><span class="pre">Na⁺</span></code> (NaW), <code class="docutils literal notranslate"><span class="pre">K⁺</span></code> (KW) and <code class="docutils literal notranslate"><span class="pre">Cl⁻</span></code> (ClW) which represent solvated ions in solution. One ion pair (e.g., NaW-ClW) each 34 WT4 molecules results in a salt concentration of ~0.15M (see <a class="reference internal" href="#appendix"><span class="std std-ref">Appendix</span></a> for details). Counterions were added according to <a class="reference external" href="https://pubs.acs.org/doi/10.1021/acs.jctc.9b00953">Machado et al.</a>.</p>
</div>
</section>
<section id="id40">
<h3>5.3. Run LEaP<a class="headerlink" href="#id40" title="Link to this heading"></a></h3>
<p>Run the LEaP application to generate the molecular topology and initial coordinate files:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>tleap<span class="w"> </span>-f<span class="w"> </span>gensystem.leap
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Warning messages about long, triangular or square bonds in <code class="docutils literal notranslate"><span class="pre">leap.log</span></code> file are fine and expected due to the CG topology of some residues.</p>
</div>
<p>This should create a topology file <code class="docutils literal notranslate"><span class="pre">1CRN_cg.prmtop</span></code> and a coordinate file <code class="docutils literal notranslate"><span class="pre">1CRN_cg.ncrst</span></code>.</p>
<p>Use VMD to check how the CG model looks like and particularly the presence of disulfide bonds:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>vmd<span class="w"> </span>1CRN_cg.prmtop<span class="w"> </span>1CRN_cg.ncrst<span class="w"> </span>-e<span class="w"> </span>./sirah.amber/tools/sirah_vmdtk.tcl
</pre></div>
</div>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>VMD assigns default radius to unknown atom types, the script <code class="docutils literal notranslate"><span class="pre">sirah_vmdtk.tcl</span></code> sets the right
ones, according to the CG representation. It also provides a kit of useful selection macros, coloring methods and backmapping utilities.
Use the command <code class="docutils literal notranslate"><span class="pre">sirah_help</span></code> in the Tcl/Tk console of VMD to access the manual pages. To learn about SIRAH Tools’ capabilities, you can also go to the <a class="reference internal" href="Tutorials%20sirahtools.html#sirah-tools"><span class="std std-ref">SIRAH Tools tutorial</span></a>.</p>
</div>
</section>
<section id="id41">
<h3>5.4. Run the simulation<a class="headerlink" href="#id41" title="Link to this heading"></a></h3>
<p>Make a new folder for the run:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>mkdir<span class="w"> </span>-p<span class="w"> </span>run<span class="p">;</span><span class="w"> </span><span class="nb">cd</span><span class="w"> </span>run
</pre></div>
</div>
<p>The folder <code class="docutils literal notranslate"><span class="pre">sirah.amber/tutorial/5/</span></code> contains typical input files for energy minimization
(<code class="docutils literal notranslate"><span class="pre">em1_WT4.in</span></code> and <code class="docutils literal notranslate"><span class="pre">em2_WT4.in</span></code>), equilibration (<code class="docutils literal notranslate"><span class="pre">eq1_WT4.in</span></code> and <code class="docutils literal notranslate"><span class="pre">eq2_WT4.in</span></code>) and production (<code class="docutils literal notranslate"><span class="pre">md_WT4.in</span></code>) runs. Please check carefully the
input flags therein, in particular the definition of flag <em>chngmask=0</em> at <em>&amp;ewald</em> section is <strong>mandatory</strong>.</p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<blockquote>
<div><p><strong>Some commonly used flags in Amber</strong></p>
</div></blockquote>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">-i</span></code>: Input file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-o</span></code>: Output file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-p</span></code>: Parameter/topology file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-c</span></code>: Coordinate file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-r</span></code>: Restart file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-x</span></code>: Trajectory file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-ref</span></code>: Reference file</p></li>
</ul>
</div>
<div class="admonition caution">
<p class="admonition-title">Caution</p>
<p>These input files are executed by the <strong>GPU</strong> implementation of <code class="docutils literal notranslate"><span class="pre">pmemd.cuda</span></code>. Other available modules are <code class="docutils literal notranslate"><span class="pre">sander</span></code> or <code class="docutils literal notranslate"><span class="pre">pmemd</span></code>, which are both <strong>CPU</strong> implementations of Amber.</p>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The same input files can be used to run on CPU with the modules <code class="docutils literal notranslate"><span class="pre">pmemd</span></code> or <code class="docutils literal notranslate"><span class="pre">sander</span></code>.</p>
</div>
<p><strong>Energy Minimization of side chains and solvent by restraining the backbone:</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/5/em1_WT4.in<span class="w"> </span>-p<span class="w"> </span>../1CRN_cg.prmtop<span class="w"> </span>-c<span class="w"> </span>../1CRN_cg.ncrst<span class="w"> </span>-ref<span class="w"> </span>../1CRN_cg.ncrst<span class="w"> </span>-o<span class="w"> </span>1CRN_cg_em1.out<span class="w"> </span>-r<span class="w"> </span>1CRN_cg_em1.ncrst<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
<p><strong>Energy Minimization of whole system:</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/5/em2_WT4.in<span class="w"> </span>-p<span class="w"> </span>../1CRN_cg.prmtop<span class="w"> </span>-c<span class="w"> </span>1CRN_cg_em1.ncrst<span class="w"> </span>-o<span class="w"> </span>1CRN_cg_em2.out<span class="w"> </span>-r<span class="w"> </span>1CRN_cg_em2.ncrst<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
<p><strong>Solvent Equilibration (NPT):</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/5/eq1_WT4.in<span class="w"> </span>-p<span class="w"> </span>../1CRN_cg.prmtop<span class="w"> </span>-c<span class="w"> </span>1CRN_cg_em2.ncrst<span class="w"> </span>-ref<span class="w"> </span>1CRN_cg_em2.ncrst<span class="w"> </span>-o<span class="w"> </span>1CRN_cg_eq1.out<span class="w"> </span>-r<span class="w"> </span>1CRN_cg_eq1.ncrst<span class="w"> </span>-x<span class="w"> </span>1CRN_cg_eq1.nc<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
<div class="admonition caution">
<p class="admonition-title">Caution</p>
<p>Option <strong>restraintmask=:’1-46’</strong> in input file <code class="docutils literal notranslate"><span class="pre">eq1_WT4.in</span></code> must be set specifically for each system to embrace all protein’s residues.</p>
</div>
<p><strong>Soft equilibration to improve side chain solvation (NPT):</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/5/eq2_WT4.in<span class="w"> </span>-p<span class="w"> </span>../1CRN_cg.prmtop<span class="w"> </span>-c<span class="w"> </span>1CRN_cg_eq1.ncrst<span class="w"> </span>-ref<span class="w"> </span>1CRN_cg_eq1.ncrst<span class="w"> </span>-o<span class="w"> </span>1CRN_cg_eq2.out<span class="w"> </span>-r<span class="w"> </span>1CRN_cg_eq2.ncrst<span class="w"> </span>-x<span class="w"> </span>1CRN_cg_eq2.nc<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
<p><strong>Production (1000ns):</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/5/md_WT4.in<span class="w"> </span>-p<span class="w"> </span>../1CRN_cg.prmtop<span class="w"> </span>-c<span class="w"> </span>1CRN_cg_eq2.ncrst<span class="w"> </span>-o<span class="w"> </span>1CRN_cg_md.out<span class="w"> </span>-r<span class="w"> </span>1CRN_cg_md.ncrst<span class="w"> </span>-x<span class="w"> </span>1CRN_cg_md.nc<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
</section>
<section id="id42">
<h3>5.5. Visualizing the simulation<a class="headerlink" href="#id42" title="Link to this heading"></a></h3>
<p>That’s it! Now you can analyze the trajectory.
Process the output trajectory to account for the Periodic Boundary Conditions (PBC):</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;autoimage\ngo\nquit\n&quot;</span><span class="w"> </span><span class="p">|</span><span class="w"> </span>cpptraj<span class="w"> </span>-p<span class="w"> </span>../1CRN_cg.prmtop<span class="w"> </span>-y<span class="w"> </span>1CRN_cg_md.nc<span class="w"> </span>-x<span class="w"> </span>1CRN_cg_md_pbc.nc<span class="w"> </span>--interactive
</pre></div>
</div>
<p>Load the processed trajectory in VMD:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">vmd</span> <span class="o">../</span><span class="mi">1</span><span class="n">CRN_cg</span><span class="o">.</span><span class="n">prmtop</span> <span class="o">../</span><span class="mi">1</span><span class="n">CRN_cg</span><span class="o">.</span><span class="n">ncrst</span> <span class="mi">1</span><span class="n">CRN_cg_md</span><span class="o">.</span><span class="n">nc</span> <span class="o">-</span><span class="n">e</span> <span class="o">../</span><span class="n">sirah</span><span class="o">.</span><span class="n">amber</span><span class="o">/</span><span class="n">tools</span><span class="o">/</span><span class="n">sirah_vmdtk</span><span class="o">.</span><span class="n">tcl</span>
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The file <code class="docutils literal notranslate"><span class="pre">sirah_vmdtk.tcl</span></code> is a Tcl script that is part of SIRAH Tools and contains the macros to properly visualize the coarse-grained structures in VMD. Use the command <code class="docutils literal notranslate"><span class="pre">sirah-help</span></code> in the Tcl/Tk console of VMD to access the manual pages. To learn about SIRAH Tools’ capabilities, you can also go to the <a class="reference internal" href="Tutorials%20sirahtools.html#sirah-tools"><span class="std std-ref">SIRAH Tools tutorial</span></a>.</p>
</div>
</section>
</section>
<section id="lipid-bilayers-in-explicit-solvent">
<span id="tutorial-6"></span><h2>6. Lipid bilayers in explicit solvent<a class="headerlink" href="#lipid-bilayers-in-explicit-solvent" title="Link to this heading"></a></h2>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Please report bugs, errors or enhancement requests through <a class="reference external" href="https://github.com/SIRAHFF/documentation/issues">Issue Tracker</a> or if you have a question about SIRAH open a <a class="reference external" href="https://github.com/SIRAHFF/documentation/discussions">New Discussion</a>.</p>
</div>
<p>This tutorial shows how to use the SIRAH force field to perform a coarse grained (CG) simulation of a
DMPC bilayer in explicit solvent (called WatFour, WT4). The main references for
this tutorial are: <a class="reference external" href="https://doi.org/10.1021/acs.jctc.9b00435">Barrera et al.</a> and <a class="reference external" href="https://academic.oup.com/bioinformatics/article/32/10/1568/1743152">Machado &amp; Pantano</a>.
We strongly advise you to read these articles before starting the tutorial. You may also find interesting <a class="reference external" href="https://pubs.aip.org/books/monograph/137/chapter-abstract/58880922/Simulating-Transmembrane-Proteins-with-the-Coarse?redirectedFrom=fulltext">this book chapter</a>.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>If you are not familiar with membrane stuff we strongly recommend you to first perform the <a class="reference external" href="http://ambermd.org/tutorials/advanced/tutorial16/">Amber tutorial on lipids</a>.</p>
</div>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>Check the <a class="reference internal" href="#download-amber"><span class="std std-ref">Setting up SIRAH</span></a> section for download and set up details before starting this tutorial.
Since this is <strong>Tutorial 6</strong>, remember to replace <code class="docutils literal notranslate"><span class="pre">X.X</span></code> and the files corresponding to this tutorial can be found in: <code class="docutils literal notranslate"><span class="pre">sirah_[version].amber/tutorial/6/</span></code></p>
</div>
<section id="id46">
<h3>6.1. Build CG representations<a class="headerlink" href="#id46" title="Link to this heading"></a></h3>
<p>Map the atomistic structure of the preassembled DMPC bilayer to its CG representation:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>./sirah.amber/tools/CGCONV/cgconv.pl<span class="w"> </span>-i<span class="w"> </span>./sirah.amber/tutorial/6/DMPC64.pdb<span class="w"> </span>-o<span class="w"> </span>DMPC64_cg.pdb<span class="w"> </span>-a<span class="w"> </span>sirah.amber/tools/CGCONV/maps/tieleman_lipid.map
</pre></div>
</div>
<p>The input file <code class="docutils literal notranslate"><span class="pre">-i</span></code> DMPC64.pdb contains the atomistic representation of the DMPC bilayer, while the output <code class="docutils literal notranslate"><span class="pre">-o</span></code> DMPC64_cg.pdb is its SIRAH CG representation. The flag <code class="docutils literal notranslate"><span class="pre">-a</span></code> is a mapping scheme for lipids.</p>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>By default, no mapping is applied to lipids, as there is no standard naming convention for them. So users are requested to append a MAP file from the list in <a class="reference internal" href="#table"><span class="std std-ref">Table 1</span></a>, by setting the flag <code class="docutils literal notranslate"><span class="pre">-a</span></code> in <code class="docutils literal notranslate"><span class="pre">cgconv.pl</span></code>. We recommend using <a class="reference external" href="https://m3g.github.io/packmol/">PACKMOL</a> for building the system. Reference building-block structures are provided at folder <code class="docutils literal notranslate"><span class="pre">sirah.amber/PDB/</span></code>, which agree with the mapping scheme in <code class="docutils literal notranslate"><span class="pre">sirah.amber/tools/CGCONV/maps/tieleman_lipid.map</span></code>. The provided DMPC bilayer contains 64 lipid molecules per leaflet distributed in a 6.4 * 6.4 nm surface, taking into account an approximate area per lipid of 0.64 nm<sup>2</sup> at 333 K. The starting configuration was created with the input file <code class="docutils literal notranslate"><span class="pre">sirah.amber/tutorial/6/DMPC_bilayer.pkm</span></code>. See <a class="reference internal" href="FAQ.html"><span class="doc">FAQs</span></a> for cautions on mapping lipids to SIRAH and tips on using fragment-based topologies. If you do not find your issue please start a discussion in our <a class="reference external" href="https://github.com/SIRAHFF/documentation/discussions">github discussion page F&amp;Q</a>.</p>
</div>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>This an advanced usage of the script <strong>cgconv.pl</strong>, you can learn other capabilities from its help by typing:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>./sirah.amber/tools/CGCONV/cgconv.pl<span class="w"> </span>-h
</pre></div>
</div>
</div>
<p>The input file <code class="docutils literal notranslate"><span class="pre">DMPC64.pdb</span></code> contains all the heavy atoms composing the lipids, while the output <code class="docutils literal notranslate"><span class="pre">DMPC64_cg.pdb</span></code> preserves a few of them. Please check both PDB structures using VMD:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>vmd<span class="w"> </span>-m<span class="w"> </span>sirah.amber/tutorial/6/DMPC64.pdb<span class="w"> </span>DMPC64_cg.pdb
</pre></div>
</div>
<p>From now on it is just normal Amber stuff!</p>
</section>
<section id="id47">
<h3>6.2. Prepare LEaP input<a class="headerlink" href="#id47" title="Link to this heading"></a></h3>
<p>Use a text editor to create the file <code class="docutils literal notranslate"><span class="pre">gensystem.leap</span></code> including the following lines:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="gp"># </span>Load<span class="w"> </span>SIRAH<span class="w"> </span>force<span class="w"> </span>field
<span class="go">addPath ./sirah.amber</span>
<span class="go">source leaprc.sirah</span>

<span class="gp"># </span>Load<span class="w"> </span>model
<span class="go">Lipid = loadpdb DMPC64_cg.pdb</span>

<span class="gp"># </span>Add<span class="w"> </span>solvent,<span class="w"> </span>counterions<span class="w"> </span>and<span class="w"> </span><span class="m">0</span>.15M<span class="w"> </span>NaCl
<span class="gp"># </span>Tuned<span class="w"> </span>solute-solvent<span class="w"> </span>closeness<span class="w"> </span><span class="k">for</span><span class="w"> </span>best<span class="w"> </span>hydration
<span class="go">solvateBox Lipid WT4BOX {0 0 40} 0.7</span>
<span class="go">addIonsRand Lipid NaW 33 ClW 33</span>

<span class="gp"># </span>Save<span class="w"> </span>Parms
<span class="go">saveAmberParmNetcdf Lipid DMPC64_cg.prmtop DMPC64_cg.ncrst</span>

<span class="gp"># </span>EXIT
<span class="go">quit</span>
</pre></div>
</div>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<p>The available electrolyte species in SIRAH force field are: <code class="docutils literal notranslate"><span class="pre">Na⁺</span></code> (NaW), <code class="docutils literal notranslate"><span class="pre">K⁺</span></code> (KW) and <code class="docutils literal notranslate"><span class="pre">Cl⁻</span></code> (ClW) which represent solvated ions in solution. One ion pair (e.g., NaW-ClW) each 34 WT4 molecules results in a salt concentration of ~0.15M (see <a class="reference internal" href="#appendix"><span class="std std-ref">Appendix</span></a> for details). Counterions were added according to <a class="reference external" href="https://pubs.acs.org/doi/10.1021/acs.jctc.9b00953">Machado et al.</a>.</p>
</div>
</section>
<section id="id49">
<h3>6.3. Run LEaP<a class="headerlink" href="#id49" title="Link to this heading"></a></h3>
<p>Run the LEaP application to generate the molecular topology and initial coordinate files:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>tleap<span class="w"> </span>-f<span class="w"> </span>gensystem.leap
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Warning messages about long, triangular or square bonds in <code class="docutils literal notranslate"><span class="pre">leap.log</span></code> file are fine and expected due to the CG topology.</p>
</div>
<p>This should create a topology file <code class="docutils literal notranslate"><span class="pre">DMPC64_cg.prmtop</span></code> and a coordinate file <code class="docutils literal notranslate"><span class="pre">DMPC64_cg.ncrst</span></code>.</p>
<p>Use VMD to check how the CG model looks:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>vmd<span class="w"> </span>DMPC64_cg.prmtop<span class="w"> </span>DMPC64_cg.ncrst<span class="w"> </span>-e<span class="w"> </span>./sirah.amber/tools/sirah_vmdtk.tcl
</pre></div>
</div>
<p>By selecting +X, +Y and +Z periodic images from the <em>Periodic</em> tab in the <em>Graphical Representations</em> window you will see small vacuum slices at box boundaries. In the following step we will fix this issue by reducing the box dimensions a few angstroms. See <a class="reference internal" href="FAQ.html"><span class="doc">FAQs</span></a> for issues on membrane systems in Amber.</p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>VMD assigns default radius to unknown atom types, the script <code class="docutils literal notranslate"><span class="pre">sirah_vmdtk.tcl</span></code> sets the right
ones, according to the CG representation. It also provides a kit of useful selection macros, coloring methods and backmapping utilities.
Use the command <code class="docutils literal notranslate"><span class="pre">sirah_help</span></code> in the Tcl/Tk console of VMD to access the manual pages. To learn about SIRAH Tools’ capabilities, you can also go to the <a class="reference internal" href="Tutorials%20sirahtools.html#sirah-tools"><span class="std std-ref">SIRAH Tools tutorial</span></a>.</p>
</div>
</section>
<section id="resize-the-box-with-cpptraj">
<h3>6.4. Resize the box with CPPTRAJ<a class="headerlink" href="#resize-the-box-with-cpptraj" title="Link to this heading"></a></h3>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>As PACKMOL does not consider periodicity while building up the system, increasing the XY box sides a few Angstroms may be required to avoid bad contacts between images.</p>
</div>
<p>Use a text editor to create the file <code class="docutils literal notranslate"><span class="pre">resize_box.cpptraj</span></code> including the following lines:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="gp"># </span>New<span class="w"> </span>box<span class="w"> </span>dimensions
<span class="go">box x 66 y 66 z 132</span>

<span class="gp"># </span>Amber<span class="w"> </span>NetCDF<span class="w"> </span>Restart<span class="w"> </span>generation
<span class="go">trajout DMPC64_cg_nb.ncrst</span>

<span class="gp"># </span>Do<span class="w"> </span>it!
<span class="go">go</span>

<span class="gp"># </span>EXIT
<span class="go">quit</span>
</pre></div>
</div>
<p>Run the CPPTRAJ application to adjust the size of the simulation box:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>cpptraj<span class="w"> </span>-p<span class="w"> </span>DMPC64_cg.prmtop<span class="w"> </span>-y<span class="w"> </span>DMPC64_cg.ncrst<span class="w"> </span>-i<span class="w"> </span>resize_box.cpptraj
</pre></div>
</div>
<p>Once again, use VMD to check the PBC images in the new box of the system:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>vmd<span class="w"> </span>DMPC64_cg.prmtop<span class="w"> </span>DMPC64_cg_nb.ncrst<span class="w"> </span>-e<span class="w"> </span>./sirah.amber/tools/sirah_vmdtk.tcl
</pre></div>
</div>
</section>
<section id="id50">
<h3>6.5. Run the simulation<a class="headerlink" href="#id50" title="Link to this heading"></a></h3>
<p>Make a new folder for the run:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>mkdir<span class="w"> </span>-p<span class="w"> </span>run<span class="p">;</span><span class="w"> </span><span class="nb">cd</span><span class="w"> </span>run
</pre></div>
</div>
<p>The folder <code class="docutils literal notranslate"><span class="pre">sirah.amber/tutorial/6/</span></code> contains typical input files for energy minimization
(<code class="docutils literal notranslate"><span class="pre">em_Lipid.in</span></code>), heating (<code class="docutils literal notranslate"><span class="pre">heat_Lipid.in</span></code>), equilibration (<code class="docutils literal notranslate"><span class="pre">eq_Lipid.in</span></code>) and production (<code class="docutils literal notranslate"><span class="pre">md_Lipid.in</span></code>) runs. Please check carefully the input flags.</p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<blockquote>
<div><p><strong>Some commonly used flags in AMBER</strong></p>
</div></blockquote>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">-i</span></code>: Input file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-o</span></code>: Output file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-p</span></code>: Parameter/topology file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-c</span></code>: Coordinate file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-r</span></code>: Restart file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-x</span></code>: Trajectory file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-ref</span></code>: Reference file</p></li>
</ul>
</div>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>These input files are executed by the <strong>GPU</strong> implementation of <code class="docutils literal notranslate"><span class="pre">pmemd.cuda</span></code>. Other available modules are <code class="docutils literal notranslate"><span class="pre">sander</span></code> or <code class="docutils literal notranslate"><span class="pre">pmemd</span></code>, which are both <strong>CPU</strong> implementations of Amber.</p>
<p>However, this simulation is time consuming owing to the system’s size. A parallel or CUDA implementation of Amber is advised.</p>
</div>
<p><strong>Energy Minimization:</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/6/em_Lipid.in<span class="w"> </span>-p<span class="w"> </span>../DMPC64_cg.prmtop<span class="w"> </span>-c<span class="w"> </span>../DMPC64_cg_nb.ncrst<span class="w"> </span>-o<span class="w"> </span>DMPC64_cg_em.out<span class="w"> </span>-r<span class="w"> </span>DMPC64_cg_em.ncrst<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
<p><strong>Heating:</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/6/heat_Lipid.in<span class="w"> </span>-p<span class="w"> </span>../DMPC64_cg.prmtop<span class="w"> </span>-c<span class="w"> </span>DMPC64_cg_em.ncrst<span class="w"> </span>-o<span class="w"> </span>DMPC64_cg_eq_0.out<span class="w"> </span>-r<span class="w"> </span>DMPC64_cg_eq_0.ncrst<span class="w"> </span>-x<span class="w"> </span>DMPC64_cg_eq_0.nc<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>To avoid “<em>skinnb errors</em>” on GPU due to large box size fluctuations, the system must be equilibrated by several “short” runs using a large <em>skinnb</em> value. The number and length of the runs may vary according to the characteristic stabilization times of the system. For more information visit the <a class="reference external" href="http://ambermd.org/tutorials/advanced/tutorial16/">Amber tutorial on lipids</a>.</p>
</div>
<p><strong>Periodic box equilibration in GPU code (500 ps x 9):</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="k">for</span><span class="w"> </span>i<span class="w"> </span><span class="k">in</span><span class="w"> </span><span class="k">$(</span>seq<span class="w"> </span><span class="m">1</span><span class="w"> </span><span class="m">9</span><span class="k">)</span>
<span class="k">do</span>
<span class="w">        </span><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;running equilibration </span><span class="nv">$i</span><span class="s2">&quot;</span>
<span class="w">        </span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span><span class="se">\</span>
<span class="w">        </span>-i<span class="w"> </span>../sirah.amber/tutorial/6/eq_Lipid.in<span class="w"> </span><span class="se">\</span>
<span class="w">        </span>-p<span class="w"> </span>../DMPC64_cg.prmtop<span class="w"> </span><span class="se">\</span>
<span class="w">        </span>-c<span class="w"> </span>DMPC64_cg_eq_<span class="k">$((</span><span class="nv">$i</span><span class="w"> </span><span class="o">-</span><span class="m">1</span><span class="k">))</span>.ncrst<span class="w"> </span><span class="se">\</span>
<span class="w">        </span>-o<span class="w"> </span>DMPC64_cg_eq_<span class="nv">$i</span>.out<span class="w"> </span><span class="se">\</span>
<span class="w">        </span>-r<span class="w"> </span>DMPC64_cg_eq_<span class="nv">$i</span>.ncrst<span class="w"> </span><span class="se">\</span>
<span class="w">        </span>-x<span class="w"> </span>DMPC64_cg_eq_<span class="nv">$i</span>.nc
<span class="k">done</span><span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
<p><strong>Production (1000ns):</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/6/md_Lipid.in<span class="w"> </span>-p<span class="w"> </span>../DMPC64_cg.prmtop<span class="w"> </span>-c<span class="w"> </span>DMPC64_cg_eq_9.ncrst<span class="w"> </span>-o<span class="w"> </span>DMPC64_cg_md.out<span class="w"> </span>-r<span class="w"> </span>DMPC64_cg_md.ncrst<span class="w"> </span>-x<span class="w"> </span>DMPC64_cg_md.nc<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
</section>
<section id="id51">
<h3>6.6. Visualizing the simulation<a class="headerlink" href="#id51" title="Link to this heading"></a></h3>
<p>That’s it! Now you can analyze the trajectory.</p>
<p>Process the output trajectory to account for the Periodic Boundary Conditions (PBC):</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;autoimage\ngo\nquit\n&quot;</span><span class="w"> </span><span class="p">|</span><span class="w"> </span>cpptraj<span class="w"> </span>-p<span class="w"> </span>../DMPC64_cg.prmtop<span class="w"> </span>-y<span class="w"> </span>DMPC64_cg_md.nc<span class="w"> </span>-x<span class="w"> </span>DMPC64_cg_md_pbc.nc<span class="w"> </span>--interactive
</pre></div>
</div>
<p>Now you can check the simulation using VMD:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">vmd</span> <span class="o">../</span><span class="n">DMPC64_cg</span><span class="o">.</span><span class="n">prmtop</span> <span class="n">DMPC64_cg_md_pbc</span><span class="o">.</span><span class="n">nc</span> <span class="o">-</span><span class="n">e</span> <span class="o">../</span><span class="n">sirah</span><span class="o">.</span><span class="n">amber</span><span class="o">/</span><span class="n">tools</span><span class="o">/</span><span class="n">sirah_vmdtk</span><span class="o">.</span><span class="n">tcl</span>
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The file <code class="docutils literal notranslate"><span class="pre">sirah_vmdtk.tcl</span></code> is a Tcl script that is part of SIRAH Tools and contains the macros to properly visualize the coarse-grained structures in VMD. Use the command <code class="docutils literal notranslate"><span class="pre">sirah-help</span></code> in the Tcl/Tk console of VMD to access the manual pages. To learn about SIRAH Tools’ capabilities, you can also go to the <a class="reference internal" href="Tutorials%20sirahtools.html#sirah-tools"><span class="std std-ref">SIRAH Tools tutorial</span></a>.</p>
</div>
<p>You can also use CPPTRAJ to calculate the area per lipid:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>cpptraj<span class="w"> </span>-p<span class="w"> </span>../DMPC64_cg.prmtop<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/6/area_lipid.cpptraj
</pre></div>
</div>
<p>Use Grace to plot the results:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>xmgrace<span class="w"> </span>apl_DMPC64_310K.dat
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>To calculate the area per lipid, divide the membrane’s area by the DMPC molecules per leaflet:</p>
<blockquote>
<div><div class="math notranslate nohighlight">
\[\frac{Area}{Lipid} = \frac{Box(x) * Box(y)}{64}\]</div>
</div></blockquote>
</div>
<p>And density profiles and bilayer thickness:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>cpptraj<span class="w"> </span>-p<span class="w"> </span>../DMPC64_cg.prmtop<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/6/dens_profile.cpptraj
</pre></div>
</div>
<p>Use Grace to plot the results:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>xmgrace<span class="w"> </span>-nxy<span class="w"> </span>dens_profile_DMPC64_310K.dat
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The thickness of the bilayer is the distance between the two peaks corresponding to the position of phosphate beads (BFO) along the z-axis.</p>
</div>
</section>
</section>
<section id="membrane-proteins-in-explicit-solvent">
<span id="tutorial-7"></span><h2>7. Membrane proteins in explicit solvent<a class="headerlink" href="#membrane-proteins-in-explicit-solvent" title="Link to this heading"></a></h2>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Please report bugs, errors or enhancement requests through <a class="reference external" href="https://github.com/SIRAHFF/documentation/issues">Issue Tracker</a> or if you have a question about SIRAH open a <a class="reference external" href="https://github.com/SIRAHFF/documentation/discussions">New Discussion</a>.</p>
</div>
<p>This tutorial shows how to use the SIRAH force field to perform a coarse grained (CG) simulation of a protein embedded in a lipid bilayer using explicit solvent (called WatFour, WT4). The main references for
this tutorial are: <a class="reference external" href="https://doi.org/10.1021/acs.jctc.9b00006">Machado et al.</a>, <a class="reference external" href="https://doi.org/10.1021/acs.jctc.9b00435">Barrera et al.</a>, and <a class="reference external" href="https://academic.oup.com/bioinformatics/article/32/10/1568/1743152">Machado &amp; Pantano</a>.
We strongly advise you to read these articles before starting the tutorial.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>We strongly advise you to read and complete <a class="reference internal" href="#tutorial-5"><span class="std std-ref">Tutorial 5</span></a> and <a class="reference internal" href="#tutorial-6"><span class="std std-ref">Tutorial 6</span></a> before starting.</p>
</div>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>Check the <a class="reference internal" href="#download-amber"><span class="std std-ref">Setting up SIRAH</span></a> section for download and set up details before starting this tutorial.
Since this is <strong>Tutorial 7</strong>, remember to replace <code class="docutils literal notranslate"><span class="pre">X.X</span></code> and the files corresponding to this tutorial can be found in: <code class="docutils literal notranslate"><span class="pre">sirah_[version].amber/tutorial/7/</span></code></p>
</div>
<section id="id56">
<h3>7.1. Build CG representations<a class="headerlink" href="#id56" title="Link to this heading"></a></h3>
<div class="admonition caution">
<p class="admonition-title">Caution</p>
<p>The mapping to CG requires the correct protonation state of each residue at a given pH. We recommend using the <a class="reference external" href="https://www.charmm-gui.org/">CHARMM-GUI server</a> and use the <strong>PDB Reader &amp; Manipulator</strong> to prepare your system. An account is required to access any of the CHARMM-GUI Input Generator modules, and it can take up to 24 hours to obtain one.</p>
<p>See <a class="reference internal" href="#tutorial-5"><span class="std std-ref">Tutorial 5</span></a> for cautions while preparing and mapping atomistic proteins to SIRAH.</p>
</div>
<p>Map the atomistic structure of protein <a class="reference external" href="https://www.rcsb.org/structure/2KYV">2KYV</a> to its CG representation:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>./sirah.amber/tools/CGCONV/cgconv.pl<span class="w"> </span>-i<span class="w"> </span>sirah.amber/tutorial/7/2kyv.pqr<span class="w"> </span>-o<span class="w"> </span>2kyv_cg.pdb
</pre></div>
</div>
<p>The input file <code class="docutils literal notranslate"><span class="pre">-i</span></code> 2kyv.pqr contains the atomistic representation of <a class="reference external" href="https://www.rcsb.org/structure/2KYV">2KYV</a> structure at pH <strong>7.0</strong>, while the output <code class="docutils literal notranslate"><span class="pre">-o</span></code> 2kyv_cg.pdb is its SIRAH CG representation.</p>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>If you already have an atomistic protein within a membrane, then you can simply map the entire system to SIRAH (this is highly recommended) and skip the step 2, however clipping the membrane patch may be required to set a correct solvation box (see bellow). By default, no mapping is applied to lipids, as there is no standard naming convention for them. So users are requested to append a MAP file from the list in <a class="reference internal" href="#table"><span class="std std-ref">Table 1</span></a>, by setting the flag <code class="docutils literal notranslate"><span class="pre">-a</span></code> in <code class="docutils literal notranslate"><span class="pre">cgconv.pl</span></code>. We recommend using <a class="reference external" href="https://m3g.github.io/packmol/">PACKMOL</a> for building the system. Reference building-block structures are provided at folder <code class="docutils literal notranslate"><span class="pre">sirah.amber/PDB/</span></code>, which agree with the mapping scheme <code class="docutils literal notranslate"><span class="pre">sirah.amber/tools/CGCONV/maps/tieleman_lipid.map</span></code>. See <a class="reference internal" href="FAQ.html"><span class="doc">FAQs</span></a> for cautions on mapping lipids to SIRAH and tips on using fragment-based topologies.</p>
</div>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>This an advanced usage of the script <strong>cgconv.pl</strong>, you can learn other capabilities from its help by typing:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>./sirah.amber/tools/CGCONV/cgconv.pl<span class="w"> </span>-h
</pre></div>
</div>
</div>
<section id="embed-the-protein-in-a-lipid-bilayer">
<h4>7.1.1. Embed the protein in a lipid bilayer<a class="headerlink" href="#embed-the-protein-in-a-lipid-bilayer" title="Link to this heading"></a></h4>
<p>We will show one possible way to do it by starting from a pre-equilibrated CG membrane patch.</p>
<p>In <code class="docutils literal notranslate"><span class="pre">sirah_[version].amber/tutorial/7/</span></code> you will find a pre-stabilized CG DMPC bilayer patch, concatenate it with the previously generated CG representation of the phospholamban (PLN) pentamer (PDB code: <a class="reference external" href="https://www.rcsb.org/structure/2KYV">2KYV</a>):</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>head<span class="w"> </span>-qn<span class="w"> </span>-1<span class="w"> </span>2kyv_cg.pdb<span class="w"> </span>./sirah.amber/tutorial/7/DMPC_cg.pdb<span class="w"> </span>&gt;<span class="w"> </span>2kyv_DMPC_cg_init.pdb
</pre></div>
</div>
<p>Luckily, we already oriented the protein inside the membrane. For setting up your own system you can go to <a class="reference external" href="https://opm.phar.umich.edu/">Orientations of Proteins in Membranes (OPM) database</a> and (if your structure is available) use the dummy atoms provided there to make them match with your membrane model (see <strong>Figure 1</strong>).</p>
<figure class="align-center" id="id90">
<a class="reference internal image-reference" href="_images/Tuto7.png"><img alt="_images/Tuto7.png" src="_images/Tuto7.png" style="width: 100%;" />
</a>
<figcaption>
<p><span class="caption-text"><strong>Figure 1.</strong> Protein oriented inside the membrane from the OPM database with dummy atoms represented as orange spheres.</span><a class="headerlink" href="#id90" title="Link to this image"></a></p>
</figcaption>
</figure>
</section>
<section id="delete-close-contact-lipid-molecules">
<h4>7.1.2. Delete close contact lipid molecules<a class="headerlink" href="#delete-close-contact-lipid-molecules" title="Link to this heading"></a></h4>
<p>We need to use VMD to to delete lipid molecules in close contact with the protein. For a proper treatment and visualization of the system in VMD you must first generate the molecular topology and initial coordinate files.</p>
<p>Use a text editor to create the file <code class="docutils literal notranslate"><span class="pre">geninit.leap</span></code> including the following lines:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="gp"># </span>Load<span class="w"> </span>SIRAH<span class="w"> </span>force<span class="w"> </span>field
<span class="go">addPath ./sirah.amber</span>
<span class="go">source leaprc.sirah</span>

<span class="gp"># </span>Load<span class="w"> </span>model
<span class="go">ProtMem = loadpdb 2kyv_DMPC_cg_init.pdb</span>

<span class="gp"># </span>Save<span class="w"> </span>Parms
<span class="go">saveAmberParmNetcdf ProtMem 2kyv_DMPC_cg_init.prmtop 2kyv_DMPC_cg_init.ncrst</span>

<span class="gp"># </span>EXIT
<span class="go">quit</span>
</pre></div>
</div>
<p>Run the LEaP application to generate the molecular topology and initial coordinate files:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>tleap<span class="w"> </span>-f<span class="w"> </span>geninit.leap
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Warning messages about long, triangular or square bonds in <code class="docutils literal notranslate"><span class="pre">leap.log</span></code> file are fine and expected due to the CG topology.</p>
</div>
<p>Now, open the files on VMD:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>vmd<span class="w"> </span>2kyv_DMPC_cg_init.prmtop<span class="w"> </span>2kyv_DMPC_cg_init.ncrst<span class="w"> </span>-e<span class="w"> </span>sirah.amber/tools/sirah_vmdtk.tcl
</pre></div>
</div>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>VMD assigns default radius to unknown atom types, the script <code class="docutils literal notranslate"><span class="pre">sirah_vmdtk.tcl</span></code> sets the right
ones, according to the CG representation. It also provides a kit of useful selection macros, coloring methods and backmapping utilities.
Use the command <code class="docutils literal notranslate"><span class="pre">sirah_help</span></code> in the Tcl/Tk console of VMD to access the manual pages. To learn about SIRAH Tools’ capabilities, you can also go to the <a class="reference internal" href="Tutorials%20sirahtools.html#sirah-tools"><span class="std std-ref">SIRAH Tools tutorial</span></a>.</p>
</div>
<p>In the VMD main window, select <em>Graphics</em> &gt; <em>Representations</em>. In the <em>Selected Atoms</em> box, type:</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>not (same residue as (sirah_membrane within 3.5 of sirah_protein) or (sirah_membrane and x &lt; 5 or x &gt; 142 or y &lt; 2 or y &gt; 140))
</pre></div>
</div>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>In the first part of the selection, lipid molecules in close contact with the protein are removed. The second one is made to “trim” the membrane patch, deleting lipids with acyl chains located in the periodic boundary images. This is frequent when using pre-equilibrated membrane patches and is necessary to avoid clashes in the following steps.</p>
</div>
<p>To save the refined protein-membrane system, in the VMD main window click on <code class="docutils literal notranslate"><span class="pre">2kyv_DMPC_cg_init.prmtop</span></code>, then select <em>File</em> &gt; <em>Save Coordinates</em>. In the <em>Selected atoms</em> option choose the selection you have just created and Save as <code class="docutils literal notranslate"><span class="pre">2kyv_DMPC_cg.pdb</span></code>.</p>
<p>From now on it is just normal AMBER stuff!</p>
</section>
</section>
<section id="id58">
<h3>7.2. Prepare LEaP input<a class="headerlink" href="#id58" title="Link to this heading"></a></h3>
<p>Use a text editor to create the file <code class="docutils literal notranslate"><span class="pre">gensystem.leap</span></code> including the following lines:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="gp"># </span>Load<span class="w"> </span>SIRAH<span class="w"> </span>force<span class="w"> </span>field
<span class="go">addPath ./sirah.amber</span>
<span class="go">source leaprc.sirah</span>

<span class="gp"># </span>Load<span class="w"> </span>model
<span class="go">ProtMem = loadpdb 2kyv_DMPC_cg.pdb</span>

<span class="gp"># </span>Info<span class="w"> </span>on<span class="w"> </span>system<span class="w"> </span>charge
<span class="go">charge ProtMem</span>

<span class="gp"># </span>Prevent<span class="w"> </span>adding<span class="w"> </span>solvent<span class="w"> </span>molecules<span class="w"> </span>beyond<span class="w"> </span>the<span class="w"> </span>membrane<span class="w"> </span>boundaries
<span class="go">setbox ProtMem centers 0</span>

<span class="gp"># </span>Add<span class="w"> </span>solvent,<span class="w"> </span>counterions<span class="w"> </span>and<span class="w"> </span><span class="m">0</span>.15M<span class="w"> </span>NaCl
<span class="gp"># </span>Tuned<span class="w"> </span>solute-solvent<span class="w"> </span>closeness<span class="w"> </span><span class="k">for</span><span class="w"> </span>best<span class="w"> </span>hydration
<span class="go">solvateBox ProtMem WT4BOX {0,0,27} 0.7</span>
<span class="go">addIonsRand ProtMem NaW 109 ClW 124</span>

<span class="gp"># </span>Save<span class="w"> </span>Parms
<span class="go">saveAmberParmNetcdf ProtMem 2kyv_DMPC_cg.prmtop 2kyv_DMPC_cg.ncrst</span>

<span class="gp"># </span>EXIT
<span class="go">quit</span>
</pre></div>
</div>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<p>The available electrolyte species in SIRAH force field are: <code class="docutils literal notranslate"><span class="pre">Na⁺</span></code> (NaW), <code class="docutils literal notranslate"><span class="pre">K⁺</span></code> (KW) and <code class="docutils literal notranslate"><span class="pre">Cl⁻</span></code> (ClW) which represent solvated ions in solution. One ion pair (e.g., NaW-ClW) each 34 WT4 molecules results in a salt concentration of ~0.15M (see <a class="reference internal" href="#appendix"><span class="std std-ref">Appendix</span></a> for details). Counterions were added according to <a class="reference external" href="https://pubs.acs.org/doi/10.1021/acs.jctc.9b00953">Machado et al.</a>.</p>
</div>
</section>
<section id="id59">
<h3>7.3. Run LEaP<a class="headerlink" href="#id59" title="Link to this heading"></a></h3>
<p>Run the LEaP application to generate the molecular topology and initial coordinate files:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>tleap<span class="w"> </span>-f<span class="w"> </span>gensystem.leap
</pre></div>
</div>
<div class="admonition caution">
<p class="admonition-title">Caution</p>
<p>Warning messages about long, triangular or square bonds in <code class="docutils literal notranslate"><span class="pre">leap.log</span></code> file are fine and expected due to the CG topology.</p>
</div>
<p>This should create a topology file <code class="docutils literal notranslate"><span class="pre">2kyv_DMPC_cg.prmtop</span></code> and a coordinate file <code class="docutils literal notranslate"><span class="pre">2kyv_DMPC_cg.ncrst</span></code>.</p>
<p>Use VMD to check how the CG model looks:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>vmd<span class="w"> </span>2kyv_DMPC_cg.prmtop<span class="w"> </span>2kyv_DMPC_cg.ncrst<span class="w"> </span>-e<span class="w"> </span>./sirah.amber/tools/sirah_vmdtk.tcl
</pre></div>
</div>
<p>By selecting +X, +Y and +Z periodic images from the <em>Periodic</em> tab in the <em>Graphical Representations</em> window you will see unwanted water near the
hydrophobic region of the membrane and small vacuum slices at box boundaries. In the following step we will fix these issues by deleting those water molecules and reducing the box dimensions a few angstroms. See <a class="reference internal" href="FAQ.html"><span class="doc">FAQs</span></a> for issues on membrane systems in Amber. If you do not find your issue please start a discussion in our <a class="reference external" href="https://github.com/SIRAHFF/documentation/discussions">github discussion page F&amp;Q</a>.</p>
</section>
<section id="id61">
<h3>7.4. Resize the box with CPPTRAJ<a class="headerlink" href="#id61" title="Link to this heading"></a></h3>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>As PACKMOL does not consider periodicity while building up the system, increasing the XY box sides a few Angstroms may be required to avoid bad contacts between images.</p>
</div>
<p>Use a text editor to create the file <code class="docutils literal notranslate"><span class="pre">resize_box.cpptraj</span></code> including the following lines:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="gp"># </span>Set<span class="w"> </span>reference<span class="w"> </span>coordinate<span class="w"> </span>file
<span class="go">reference 2kyv_DMPC_cg.ncrst</span>

<span class="gp"># </span>Remove<span class="w"> </span>water<span class="w"> </span>using<span class="w"> </span>distance-based<span class="w"> </span>masks
<span class="go">strip :WT4&amp;(@BCT1,BCT2,BC13,BC23&lt;:12.0) parmout 2kyv_DMPC_cg.prmtop</span>

<span class="gp"># </span>New<span class="w"> </span>box<span class="w"> </span>dimensions
<span class="go">box x 128 y 128 z 114</span>

<span class="gp"># </span>Amber<span class="w"> </span>NetCDF<span class="w"> </span>Restart<span class="w"> </span>generation
<span class="go">trajout 2kyv_DMPC_cg_nb.ncrst</span>

<span class="gp"># </span>Do<span class="w"> </span>it!
<span class="go"> go</span>

<span class="gp"># </span>EXIT
<span class="go"> quit</span>
</pre></div>
</div>
<div class="admonition caution">
<p class="admonition-title">Caution</p>
<p>This is a critical step when preparing membrane systems to simulate with Amber. In this case, the new box dimensions were set after some trial and error tests to allow for limited overlap between periodic box images. An excessive overlap may lead to important atom clashes an eventual system explosion during minimization/simulation, while insufficient overlap may impact the membrane cohesivity at PBC boundaries leading to pore formations or other issues.</p>
</div>
<p>Run the CPPTRAJ application application to adjust the size of the simulation box:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>cpptraj<span class="w"> </span>-p<span class="w"> </span>2kyv_DMPC_cg.prmtop<span class="w"> </span>-y<span class="w"> </span>2kyv_DMPC_cg.ncrst<span class="w"> </span>-i<span class="w"> </span>resize_box.cpptraj
</pre></div>
</div>
<p>Once again, use VMD to check the PBC images in the new box of the system:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>vmd<span class="w"> </span>2kyv_DMPC_cg.prmtop<span class="w"> </span>2kyv_DMPC_cg_nb.ncrst<span class="w"> </span>-e<span class="w"> </span>./sirah.amber/tools/sirah_vmdtk.tcl
</pre></div>
</div>
</section>
<section id="id62">
<h3>7.5. Run the simulation<a class="headerlink" href="#id62" title="Link to this heading"></a></h3>
<p>Make a new folder for the run:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>mkdir<span class="w"> </span>-p<span class="w"> </span>run<span class="p">;</span><span class="w"> </span><span class="nb">cd</span><span class="w"> </span>run
</pre></div>
</div>
<p>The folder <code class="docutils literal notranslate"><span class="pre">sirah.amber/tutorial/7/</span></code> contains typical input files for energy minimization
(<code class="docutils literal notranslate"><span class="pre">em1_Prot-Lip.in</span></code> and <code class="docutils literal notranslate"><span class="pre">em2_Prot-Lip.in</span></code>), heating (<code class="docutils literal notranslate"><span class="pre">heat_Prot-Lip.in</span></code>), equilibration (<code class="docutils literal notranslate"><span class="pre">eq_Prot-Lip.in</span></code>) and production (<code class="docutils literal notranslate"><span class="pre">md_Prot-Lip.in</span></code>) runs. Please check carefully the input flags.</p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<blockquote>
<div><p><strong>Some commonly used flags in Amber</strong></p>
</div></blockquote>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">-i</span></code>: Input file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-o</span></code>: Output file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-p</span></code>: Parameter/topology file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-c</span></code>: Coordinate file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-r</span></code>: Restart file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-x</span></code>: Trajectory file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-ref</span></code>: Reference file</p></li>
</ul>
</div>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>These input files are executed by the <strong>GPU</strong> implementation of <code class="docutils literal notranslate"><span class="pre">pmemd.cuda</span></code>. Other available modules are <code class="docutils literal notranslate"><span class="pre">sander</span></code> or <code class="docutils literal notranslate"><span class="pre">pmemd</span></code>, which are both <strong>CPU</strong> implementations of Amber.</p>
<p>However, this simulation is time consuming owing to the system’s size. A parallel or CUDA implementation of Amber is advised.</p>
</div>
<p><strong>Energy Minimization of side chains by restraining the backbone:</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/7/em1_Prot-Lip.in<span class="w"> </span>-p<span class="w"> </span>../2kyv_DMPC_cg.prmtop<span class="w"> </span>-c<span class="w"> </span>../2kyv_DMPC_cg_nb.ncrst<span class="w"> </span>-ref<span class="w"> </span>../2kyv_DMPC_cg_nb.ncrst<span class="w"> </span>-o<span class="w"> </span>2kyv_DMPC_cg_em_1.out<span class="w"> </span>-r<span class="w"> </span>2kyv_DMPC_cg_em_1.ncrst<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
<p><strong>Energy Minimization of of the whole system:</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/7/em2_Prot-Lip.in<span class="w"> </span>-p<span class="w"> </span>../2kyv_DMPC_cg.prmtop<span class="w"> </span>-c<span class="w"> </span>2kyv_DMPC_cg_em_1.ncrst<span class="w"> </span>-ref<span class="w"> </span>2kyv_DMPC_cg_em_1.ncrst<span class="w"> </span>-o<span class="w"> </span>2kyv_DMPC_cg_em_2.out<span class="w"> </span>-r<span class="w"> </span>2kyv_DMPC_cg_em_2.ncrst<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
<p><strong>Heating:</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/7/heat_Prot-Lip.in<span class="w"> </span>-p<span class="w"> </span>../2kyv_DMPC_cg.prmtop<span class="w"> </span>-c<span class="w"> </span>2kyv_DMPC_cg_em_2.ncrst<span class="w"> </span>-ref<span class="w"> </span>2kyv_DMPC_cg_em_2.ncrst<span class="w"> </span>-o<span class="w"> </span>2kyv_DMPC_cg_eq_0.out<span class="w"> </span>-r<span class="w"> </span>2kyv_DMPC_cg_eq_0.ncrst<span class="w"> </span>-x<span class="w"> </span>2kyv_DMPC_cg_eq_0.nc<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>To avoid “<em>skinnb errors</em>” on GPU due to large box size fluctuations, the system must be equilibrated by several “short” runs using a large <em>skinnb</em> value. The number and length of the runs may vary according to the characteristic stabilization times of the system. For more information visit the <a class="reference external" href="http://ambermd.org/tutorials/advanced/tutorial16/">Amber tutorial on lipids</a>.</p>
</div>
<p><strong>Periodic box equilibration in GPU code (500 ps x 9):</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="k">for</span><span class="w"> </span>i<span class="w"> </span><span class="k">in</span><span class="w"> </span><span class="k">$(</span>seq<span class="w"> </span><span class="m">1</span><span class="w"> </span><span class="m">9</span><span class="k">)</span>
<span class="k">do</span>
<span class="w">        </span><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;running equilibration </span><span class="nv">$i</span><span class="s2">&quot;</span>
<span class="w">        </span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span><span class="se">\</span>
<span class="w">        </span>-i<span class="w"> </span>../sirah.amber/tutorial/7/heat_Prot-Lip.in<span class="w"> </span><span class="se">\</span>
<span class="w">        </span>-p<span class="w"> </span>../2kyv_DMPC_cg.prmtop<span class="w"> </span><span class="se">\</span>
<span class="w">        </span>-c<span class="w"> </span>2kyv_DMPC_cg_eq_<span class="k">$((</span><span class="nv">$i</span><span class="w"> </span><span class="o">-</span><span class="m">1</span><span class="k">))</span>.ncrst<span class="w"> </span><span class="se">\</span>
<span class="w">        </span>-ref<span class="w"> </span>2kyv_DMPC_cg_eq_<span class="k">$((</span><span class="nv">$i</span><span class="w"> </span><span class="o">-</span><span class="m">1</span><span class="k">))</span>.ncrst<span class="w"> </span><span class="se">\</span>
<span class="w">        </span>-o<span class="w"> </span>2kyv_DMPC_cg_eq_<span class="nv">$i</span>.out<span class="w"> </span><span class="se">\</span>
<span class="w">        </span>-r<span class="w"> </span>2kyv_DMPC_cg_eq_<span class="nv">$i</span>.ncrst<span class="w"> </span><span class="se">\</span>
<span class="w">        </span>-x<span class="w"> </span>2kyv_DMPC_cg_eq_<span class="nv">$i</span>.nc
<span class="k">done</span><span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
<p><strong>Production (1000ns):</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/7/md_Prot-Lip.in<span class="w"> </span>-p<span class="w"> </span>../2kyv_DMPC_cg.prmtop<span class="w"> </span>-c<span class="w"> </span>2kyv_DMPC_cg_eq_9.ncrst<span class="w"> </span>-o<span class="w"> </span>2kyv_DMPC_cg_md.out<span class="w"> </span>-r<span class="w"> </span>2kyv_DMPC_cg_md.ncrst<span class="w"> </span>-x<span class="w"> </span>2kyv_DMPC_cg_md.nc<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
</section>
<section id="id63">
<h3>7.6. Visualizing the simulation<a class="headerlink" href="#id63" title="Link to this heading"></a></h3>
<p>That’s it! Now you can analyze the trajectory.</p>
<p>Process the output trajectory to account for the Periodic Boundary Conditions (PBC):</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;autoimage\ngo\nquit\n&quot;</span><span class="w"> </span><span class="p">|</span><span class="w"> </span>cpptraj<span class="w"> </span>-p<span class="w"> </span>../2kyv_DMPC_cg.prmtop<span class="w"> </span>-y<span class="w"> </span>2kyv_DMPC_cg_md.nc<span class="w"> </span>-x<span class="w"> </span>2kyv_DMPC_cg_md_pbc.nc<span class="w"> </span>--interactive
</pre></div>
</div>
<p>Now you can check the simulation using VMD:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">vmd</span> <span class="o">../</span><span class="mi">2</span><span class="n">kyv_DMPC_cg</span><span class="o">.</span><span class="n">prmtop</span> <span class="mi">2</span><span class="n">kyv_DMPC_cg_md_pbc</span><span class="o">.</span><span class="n">nc</span> <span class="o">-</span><span class="n">e</span> <span class="o">../</span><span class="n">sirah</span><span class="o">.</span><span class="n">amber</span><span class="o">/</span><span class="n">tools</span><span class="o">/</span><span class="n">sirah_vmdtk</span><span class="o">.</span><span class="n">tcl</span>
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The file <code class="docutils literal notranslate"><span class="pre">sirah_vmdtk.tcl</span></code> is a Tcl script that is part of SIRAH Tools and contains the macros to properly visualize the coarse-grained structures in VMD. Use the command <code class="docutils literal notranslate"><span class="pre">sirah-help</span></code> in the Tcl/Tk console of VMD to access the manual pages. To learn about SIRAH Tools’ capabilities, you can also go to the <a class="reference internal" href="Tutorials%20sirahtools.html#sirah-tools"><span class="std std-ref">SIRAH Tools tutorial</span></a>.</p>
</div>
</section>
</section>
<section id="glycoprotein-in-explicit-solvent">
<span id="tutorial-8"></span><h2>8. Glycoprotein in explicit solvent<a class="headerlink" href="#glycoprotein-in-explicit-solvent" title="Link to this heading"></a></h2>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Please report bugs, errors or enhancement requests through <a class="reference external" href="https://github.com/SIRAHFF/documentation/issues">Issue Tracker</a> or if you have a question about SIRAH open a <a class="reference external" href="https://github.com/SIRAHFF/documentation/discussions">New Discussion</a>.</p>
</div>
<p>This tutorial shows how to use the SIRAH force field to perform a coarse grained (CG) simulation of a
glycoprotein in explicit solvent (called WatFour, WT4). The main references for
this tutorial are: <a class="reference external" href="https://pubs.acs.org/doi/abs/10.1021/ct100379f">Darré et al.</a>, <a class="reference external" href="https://doi.org/10.1021/acs.jctc.9b00006">Machado et al.</a> and <a class="reference external" href="https://academic.oup.com/bioinformatics/article/32/10/1568/1743152">Machado &amp; Pantano</a>. We strongly advise you to read these articles before starting the tutorial.</p>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>Check the <a class="reference internal" href="#download-amber"><span class="std std-ref">Setting up SIRAH</span></a> section for download and set up details before starting this tutorial.
Since this is <strong>Tutorial 8</strong>, remember to replace <code class="docutils literal notranslate"><span class="pre">X.X</span></code> and the files corresponding to this tutorial can be found in: <code class="docutils literal notranslate"><span class="pre">sirah_[version].amber/tutorial/8/</span></code></p>
</div>
<section id="id68">
<h3>8.1. Build CG representations<a class="headerlink" href="#id68" title="Link to this heading"></a></h3>
<div class="admonition caution">
<p class="admonition-title">Caution</p>
<p>The mapping to CG requires the correct protonation state of each residue at a given pH. We recommend using the <a class="reference external" href="https://www.charmm-gui.org/">CHARMM-GUI server</a> with the <strong>Glycan Reader &amp; Modeler</strong> to prepare your system, choosing the output naming scheme of AMBER for best compatibility. An account is required to access any of the CHARMM-GUI Input Generator modules, and it can take up to 24 hours to obtain one.</p>
</div>
<p>This example use the structure of glycoprotein <a class="reference external" href="https://www.rcsb.org/structure/1GYA">1GYA</a>, using CHARMM-GUI server are obtained the files parm7 and rst7, these files are converted to pdb format with the naming scheme of AMBER/GLYCAM using AmberTool as follow:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>ambpdb<span class="w"> </span>-p<span class="w"> </span>./sirah.amber/tutorial/8/step3_input.parm7<span class="w"> </span>-c<span class="w"> </span>./sirah.amber/tutorial/8/step3_input.rst7<span class="w"> </span>&gt;<span class="w"> </span>1GYA_glycam.pdb
</pre></div>
</div>
<p>From the file 1GYA_glycam.pdb generated delete the solvent, rename to 1GYA_glycam_NoW.pdb and then map the protonated atomistic structure to its CG representation:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>./sirah.amber/tools/CGCONV/cgconv.pl<span class="w"> </span>-i<span class="w"> </span>./sirah.amber/tutorial/8/1GYA_glycam_NoW.pdb<span class="w"> </span>-o<span class="w"> </span>1GYA_cg.pdb
</pre></div>
</div>
<p>The input file <code class="docutils literal notranslate"><span class="pre">-i</span></code> 1GYA_glycam_NoW.pdb contains the atomistic representation of <a class="reference external" href="https://www.rcsb.org/structure/1GYA">1GYA</a> structure at pH <strong>7.0</strong>, while the output <code class="docutils literal notranslate"><span class="pre">-o</span></code> 1GYA_cg.pdb is its SIRAH CG representation.</p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>This is the basic usage of the script <strong>cgconv.pl</strong>, you can learn other capabilities from its help by typing:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>./sirah.amber/tools/CGCONV/cgconv.pl<span class="w"> </span>-h
</pre></div>
</div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p><strong>Pay attention to residue names when mapping structures from other atomistic force fields or experimental structures.</strong> Although we provide compatibility for naming schemes in PDB, GMX, GROMOS, CHARMM and OPLS, there might always be some ambiguity in the residue naming, specially regarding protonation states, that may lead to a wrong mapping. For example, SIRAH Tools always maps the residue name “HIS” to a Histidine protonated at the epsilon nitrogen (<span class="math notranslate nohighlight">\(N_{\epsilon}\)</span>) regardless the actual proton placement. Similarly, protonated Glutamic and Aspartic acid residues must be named “GLH” and “ASH”, otherwise they will be treated as negative charged residues. In addition, protonated and disulfide bonded Cysteines must be named “CYS” and “CYX” respectively. These kind of situations need to be carefully checked by the users. In all cases the residues preserve their identity when mapping and back-mapping the structures. Hence, the total charge of the protein should be the same at atomistic and SIRAH levels. You can check the following mapping file to be sure of the compatibility: <code class="docutils literal notranslate"><span class="pre">sirah.amber/tools/CGCONV/maps/sirah_prot.map</span></code> and <code class="docutils literal notranslate"><span class="pre">sirah_glycans.map</span></code>.</p>
</div>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>By default, charged termini are used. However, it is possible to set them neutral by renaming the residues from <strong>s</strong>[code] to <strong>a</strong>[code] (Nt-acetylated) or <strong>m</strong>[code] (Ct-amidated) after mapping to CG, where [code] is the root residue name in SIRAH. For example, to set a neutral N-terminal Histidine protonated at epsilon nitrogen (<span class="math notranslate nohighlight">\(N_{\epsilon}\)</span>) rename it from “sHe” to “aHe”.</p>
</div>
<p>Please check both PDBs structures using VMD:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>vmd<span class="w"> </span>-m<span class="w"> </span>sirah.amber/tutorial/8/1GYA_glycam_NoW.pdb<span class="w"> </span>1GYA_cg.pdb
</pre></div>
</div>
<p>From now on it is just normal Amber stuff!</p>
</section>
<section id="id71">
<h3>8.2. Prepare LEaP input<a class="headerlink" href="#id71" title="Link to this heading"></a></h3>
<p>Use a text editor to create the file <code class="docutils literal notranslate"><span class="pre">gensystem.leap</span></code> including the following lines:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="gp"># </span>Load<span class="w"> </span>SIRAH<span class="w"> </span>force<span class="w"> </span>field
<span class="go">addPath ./sirah.amber</span>
<span class="go">source leaprc.sirah</span>

<span class="gp"># </span>Load<span class="w"> </span>model
<span class="go">glycoprot = loadpdb 1GYA_cg.pdb</span>

<span class="go">charge glycoprot</span>

<span class="gp"># </span>N-Glycosilation
<span class="go">bond glycoprot.111.GO2  glycoprot.110.GO2</span>
<span class="go">bond glycoprot.110.GO2  glycoprot.109.GO6</span>

<span class="go">bond glycoprot.112.GO2  glycoprot.109.GO3</span>
<span class="go">bond glycoprot.109.GO2  glycoprot.108.GO6</span>

<span class="go">bond glycoprot.114.GO2 glycoprot.113.GO2</span>
<span class="go">bond glycoprot.113.GO2 glycoprot.108.GO3</span>

<span class="go">bond glycoprot.108.GO2  glycoprot.107.GO4</span>
<span class="go">bond glycoprot.107.GNac glycoprot.106.GO4</span>
<span class="gp"># </span>ASN<span class="w"> </span><span class="m">65</span>
<span class="go">bond glycoprot.106.GNac glycoprot.65.BND</span>


<span class="gp"># </span>Add<span class="w"> </span>solvent,<span class="w"> </span>counterions<span class="w"> </span>and<span class="w"> </span><span class="m">0</span>.15M<span class="w"> </span>NaCl
<span class="gp"># </span>Tuned<span class="w"> </span>solute-solvent<span class="w"> </span>closeness<span class="w"> </span><span class="k">for</span><span class="w"> </span>best<span class="w"> </span>hydration
<span class="go">solvateOct glycoprot WT4BOX 20 0.7</span>
<span class="go">addIonsRand glycoprot NaW 40 ClW 41</span>

<span class="gp"># </span>Save<span class="w"> </span>topology
<span class="go">saveAmberParmNetcdf glycoprot 1GYA_cg.prmtop 1GYA_cg.ncrst</span>

<span class="gp"># </span>EXIT
<span class="go">quit</span>
</pre></div>
</div>
<div class="admonition caution">
<p class="admonition-title">Caution</p>
<p>Each glycosidic bond must be defined explicitly in LEaP using the command bond, e.g.: “<em>bond unit.ri.beadi unit.rj.beadj</em>”. Where <em>ri</em> and <em>rj</em> correspond to the residue index in the topology file starting from 1, which may differ from the biological sequence in the PDB file. And <em>beadi</em> and <em>beadj</em> are the names of the beads involved in the glycosidic bond of the corresponding residues.</p>
<p>The above also applies to each disulfide bond, e.g.: “<em>bond unit.ri.BSG unit.rj.BSG</em>”. You can try the command <em>pdb4amber</em> to get those indexes from the atomistic structure, but be aware that it may not work if the Cysteine residues are too far away (in this case result in an empty file):</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pdb4amber<span class="w"> </span>-i<span class="w"> </span>sirah.amber/tutorial/8/1GYA_glycam_NoW.pdb<span class="w"> </span>-o<span class="w"> </span>1GYA_aa.pdb<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span>cat<span class="w"> </span>1GYA_aa_sslink
</pre></div>
</div>
</div>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<p>The available electrolyte species in SIRAH force field are: <code class="docutils literal notranslate"><span class="pre">Na⁺</span></code> (NaW), <code class="docutils literal notranslate"><span class="pre">K⁺</span></code> (KW) and <code class="docutils literal notranslate"><span class="pre">Cl⁻</span></code> (ClW) which represent solvated ions in solution. One ion pair (e.g., NaW-ClW) each 34 WT4 molecules results in a salt concentration of ~0.15M (see <a class="reference internal" href="#appendix"><span class="std std-ref">Appendix</span></a> for details). Counterions were added according to <a class="reference external" href="https://pubs.acs.org/doi/10.1021/acs.jctc.9b00953">Machado et al.</a>.</p>
</div>
</section>
<section id="id72">
<h3>8.3. Run LEaP<a class="headerlink" href="#id72" title="Link to this heading"></a></h3>
<p>Run the LEaP application to generate the molecular topology and initial coordinate files:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>tleap<span class="w"> </span>-f<span class="w"> </span>gensystem.leap
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Warning messages about long, triangular or square bonds in <code class="docutils literal notranslate"><span class="pre">leap.log</span></code> file are fine and expected due to the CG topology of some residues.</p>
</div>
<p>This should create a topology file <code class="docutils literal notranslate"><span class="pre">1GYA_cg.prmtop</span></code> and a coordinate file <code class="docutils literal notranslate"><span class="pre">1GYA_cg.ncrst</span></code>.</p>
<p>Use VMD to check how the CG model looks like and particularly the presence of glycosidic bonds:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>vmd<span class="w"> </span>1GYA_cg.prmtop<span class="w"> </span>1GYA_cg.ncrst<span class="w"> </span>-e<span class="w"> </span>./sirah.amber/tools/sirah_vmdtk.tcl
</pre></div>
</div>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>VMD assigns default radius to unknown atom types, the script <code class="docutils literal notranslate"><span class="pre">sirah_vmdtk.tcl</span></code> sets the right
ones, according to the CG representation. It also provides a kit of useful selection macros, coloring methods and backmapping utilities.
Use the command <code class="docutils literal notranslate"><span class="pre">sirah_help</span></code> in the Tcl/Tk console of VMD to access the manual pages. To learn about SIRAH Tools’ capabilities, you can also go to the <a class="reference internal" href="Tutorials%20sirahtools.html#sirah-tools"><span class="std std-ref">SIRAH Tools tutorial</span></a>.</p>
</div>
</section>
<section id="id73">
<h3>8.4. Run the simulation<a class="headerlink" href="#id73" title="Link to this heading"></a></h3>
<p>Make a new folder for the run:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>mkdir<span class="w"> </span>-p<span class="w"> </span>run<span class="p">;</span><span class="w"> </span><span class="nb">cd</span><span class="w"> </span>run
</pre></div>
</div>
<p>The folder <code class="docutils literal notranslate"><span class="pre">sirah.amber/tutorial/8/</span></code> contains typical input files for energy minimization
(<code class="docutils literal notranslate"><span class="pre">em1_WT4.in</span></code> and <code class="docutils literal notranslate"><span class="pre">em2_WT4.in</span></code>), relaxation (or equilibration) (<code class="docutils literal notranslate"><span class="pre">eq1_WT4.in</span></code> and <code class="docutils literal notranslate"><span class="pre">eq2_WT4.in</span></code>) and production (<code class="docutils literal notranslate"><span class="pre">md_WT4.in</span></code>) runs. Please check carefully the
input flags therein, in particular the definition of flag <em>chngmask=0</em> at <em>&amp;ewald</em> section is <strong>mandatory</strong>.</p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<blockquote>
<div><p><strong>Some commonly used flags in Amber</strong></p>
</div></blockquote>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">-i</span></code>: Input file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-o</span></code>: Output file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-p</span></code>: Parameter/topology file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-c</span></code>: Coordinate file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-r</span></code>: Restart file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-x</span></code>: Trajectory file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-ref</span></code>: Reference file</p></li>
</ul>
</div>
<div class="admonition caution">
<p class="admonition-title">Caution</p>
<p>These input files are executed by the <strong>GPU</strong> implementation of <code class="docutils literal notranslate"><span class="pre">pmemd.cuda</span></code>. Other available modules are <code class="docutils literal notranslate"><span class="pre">sander</span></code> or <code class="docutils literal notranslate"><span class="pre">pmemd</span></code>, which are both <strong>CPU</strong> implementations of Amber.</p>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The same input files can be used to run on CPU with the modules <code class="docutils literal notranslate"><span class="pre">pmemd</span></code> or <code class="docutils literal notranslate"><span class="pre">sander</span></code>.</p>
</div>
<p><strong>Energy Minimization of side chains and solvent by restraining the protein backbone and glycan rings:</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/8/em1_WT4.in<span class="w"> </span>-p<span class="w"> </span>../1GYA_cg.prmtop<span class="w"> </span>-c<span class="w"> </span>../1GYA_cg.ncrst<span class="w"> </span>-ref<span class="w"> </span>../1GYA_cg.ncrst<span class="w"> </span>-o<span class="w"> </span>1GYA_cg_em1.out<span class="w"> </span>-r<span class="w"> </span>1GYA_cg_em1.ncrst<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
<p><strong>Energy Minimization of whole system:</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/8/em2_WT4.in<span class="w"> </span>-p<span class="w"> </span>../1GYA_cg.prmtop<span class="w"> </span>-c<span class="w"> </span>../1GYA_cg_em1.ncrst<span class="w"> </span>-o<span class="w"> </span>1GYA_cg_em2.out<span class="w"> </span>-r<span class="w"> </span>1GYA_cg_em2.ncrst<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
<p><strong>Solvent Relaxation (or equlibration) in NPT:</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/8/eq1_WT4.in<span class="w"> </span>-p<span class="w"> </span>../1GYA_cg.prmtop<span class="w"> </span>-c<span class="w"> </span>1GYA_cg_em2.ncrst<span class="w"> </span>-ref<span class="w"> </span>1GYA_cg_em2.ncrst<span class="w"> </span>-o<span class="w"> </span>1GYA_cg_eq1.out<span class="w"> </span>-r<span class="w"> </span>1GYA_cg_eq1.ncrst<span class="w"> </span>-x<span class="w"> </span>1GYA_cg_eq1.nc<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
<div class="admonition caution">
<p class="admonition-title">Caution</p>
<p>Option <strong>restraintmask=:’1-114’</strong> in input file <code class="docutils literal notranslate"><span class="pre">eq1_WT4.in</span></code> must be set specifically for each system to restrain all glycoprotein’s residues.</p>
</div>
<p><strong>Soft Relaxation to improve side chain and glycan solvation (NPT):</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/8/eq2_WT4.in<span class="w"> </span>-p<span class="w"> </span>../1GYA_cg.prmtop<span class="w"> </span>-c<span class="w"> </span>1GYA_cg_eq1.ncrst<span class="w"> </span>-ref<span class="w"> </span>1GYA_cg_eq1.ncrst<span class="w"> </span>-o<span class="w"> </span>1GYA_cg_eq2.out<span class="w"> </span>-r<span class="w"> </span>1GYA_cg_eq2.ncrst<span class="w"> </span>-x<span class="w"> </span>1GYA_cg_eq2.nc<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
<p><strong>Production (1000ns):</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/8/md_WT4.in<span class="w"> </span>-p<span class="w"> </span>../1GYA_cg.prmtop<span class="w"> </span>-c<span class="w"> </span>1GYA_cg_eq2.ncrst<span class="w"> </span>-o<span class="w"> </span>1GYA_cg_md.out<span class="w"> </span>-r<span class="w"> </span>1GYA_cg_md.ncrst<span class="w"> </span>-x<span class="w"> </span>1GYA_cg_md.nc<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
</section>
<section id="id74">
<h3>8.5. Visualizing the simulation<a class="headerlink" href="#id74" title="Link to this heading"></a></h3>
<p>That’s it! Now you can analyze the trajectory.
Process the output trajectory to account for the Periodic Boundary Conditions (PBC):</p>
<blockquote>
<div><div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;autoimage\ngo\nquit\n&quot;</span><span class="w"> </span><span class="p">|</span><span class="w"> </span>cpptraj<span class="w"> </span>-p<span class="w"> </span>../1GYA_cg.prmtop<span class="w"> </span>-y<span class="w"> </span>1GYA_cg_md.nc<span class="w"> </span>-x<span class="w"> </span>1GYA_cg_md_pbc.nc<span class="w"> </span>--interactive
</pre></div>
</div>
</div></blockquote>
<p>Load the processed trajectory in VMD:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">vmd</span> <span class="o">../</span><span class="mi">1</span><span class="n">GYA_cg</span><span class="o">.</span><span class="n">prmtop</span> <span class="o">../</span><span class="mi">1</span><span class="n">GYA_cg</span><span class="o">.</span><span class="n">ncrst</span> <span class="mi">1</span><span class="n">GYA_cg_md</span><span class="o">.</span><span class="n">nc</span> <span class="o">-</span><span class="n">e</span> <span class="o">../</span><span class="n">sirah</span><span class="o">.</span><span class="n">amber</span><span class="o">/</span><span class="n">tools</span><span class="o">/</span><span class="n">sirah_vmdtk</span><span class="o">.</span><span class="n">tcl</span>
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The file <code class="docutils literal notranslate"><span class="pre">sirah_vmdtk.tcl</span></code> is a Tcl script that is part of SIRAH Tools and contains the macros to properly visualize the coarse-grained structures in VMD. Use the command <code class="docutils literal notranslate"><span class="pre">sirah-help</span></code> in the Tcl/Tk console of VMD to access the manual pages. To learn about SIRAH Tools’ capabilities, you can also go to the <a class="reference internal" href="Tutorials%20sirahtools.html#sirah-tools"><span class="std std-ref">SIRAH Tools tutorial</span></a>.</p>
</div>
</section>
</section>
<section id="non-enveloped-viral-like-particle">
<span id="tutorial-9"></span><h2>9. Non-enveloped Viral-Like Particle<a class="headerlink" href="#non-enveloped-viral-like-particle" title="Link to this heading"></a></h2>
<div class="admonition caution">
<p class="admonition-title">Caution</p>
<p>This tutorial was developed for the <a class="reference external" href="https://www.sirahff.com/events.html#Curso_virologia">Workshop and Course on Molecular, Physical, and Computational Virology</a>. Please be aware that the scripts and commands given here might not work right on your system due to software versions diferences.</p>
</div>
<p>This tutorial shows how to use the SIRAH force field to perform a coarse grained (CG) simulation of a
non-enveloped Virus-like Particle (VLP) in explicit solvent (called WatFour, WT4). The main references for
this tutorial are: <a class="reference external" href="https://pubs.acs.org/doi/abs/10.1021/ct100379f">Darré et al.</a>, <a class="reference external" href="https://doi.org/10.1021/acs.jctc.9b00006">Machado et al.</a>, <a class="reference external" href="https://pubs.acs.org/doi/abs/10.1021/acs.jctc.7b00659">Machado et al.</a> and <a class="reference external" href="https://pubs.acs.org/doi/10.1021/acs.jcim.0c01205">Soñora et al.</a>. We strongly advise you to read these articles before starting the tutorial.</p>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>Check the <a class="reference internal" href="#download-amber"><span class="std std-ref">Setting up SIRAH</span></a> section for download and set up details before starting this tutorial.
Since this is <strong>Tutorial 9</strong>, remember to replace <code class="docutils literal notranslate"><span class="pre">X.X</span></code> and the files corresponding to this tutorial can be found in: <code class="docutils literal notranslate"><span class="pre">sirah_[version].amber/tutorial/9/</span></code></p>
</div>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>This tutorial uses tools like Packmol and PDB2PQR that are not part of the AmberTools suite. You can find the latest version of Packmol <a class="reference external" href="https://m3g.github.io/packmol/">here</a>. You can install PDB2PQR locally by:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>conda<span class="w"> </span>install<span class="w"> </span>conda-forge::pdb2pqr
</pre></div>
</div>
</div>
<section id="structure-preprocessing">
<h3>9.1. Structure preprocessing<a class="headerlink" href="#structure-preprocessing" title="Link to this heading"></a></h3>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>This tutorial provides the necessary steps to assemble viral particles for SIRAH CG simulation. However, it is crucial to address any potential errors in the PDB files before proceeding. These preprocessing steps can prevent issues related to incorrect atom labeling, missing residues, clashes, or other common artifacts that are present in raw PDB data. View Prof. Jodi Hadden-Perilla’s fantastic video on <a class="reference external" href="https://www.youtube.com/watch?v=OHf1FqOLxTI">Tips and Tricks for MD Simulation of Viruses</a> for more details.</p>
</div>
<section id="working-with-the-biological-assembly-of-an-entire-vlp">
<h4>9.1.1 Working with the Biological Assembly of an entire VLP<a class="headerlink" href="#working-with-the-biological-assembly-of-an-entire-vlp" title="Link to this heading"></a></h4>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>In this tutorial, we will use a Biological Assembly from the PDB to work with an already-assembled Porcine Circovirus 2 (PCV2) VLP. <strong>In</strong> <a class="reference internal" href="#assalt"><span class="std std-ref">9.1.3</span></a>, <strong>we demonstrate how to generate a VLP of a non-enveloped virus from its asymmetric unit if your system is not already assembled</strong>.</p>
</div>
<p>The PCV2 <a class="reference external" href="https://www.rcsb.org/structure/6OLA">6OLA</a> structure, which includes the VLP and genomic DNA fragments in its inner core, is used in this tutorial. Download the Biological Assembly from the PDB, making sure you are in the <code class="docutils literal notranslate"><span class="pre">sirah.amber/tutorial/9/</span></code> folder:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>wget<span class="w"> </span>https://files.rcsb.org/download/6OLA-assembly1.cif.gz
</pre></div>
</div>
<p>Decompress the Biological Assembly:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>gzip<span class="w"> </span>-d<span class="w"> </span>6OLA-assembly1.cif.gz
</pre></div>
</div>
<p>Open the cif file with VMD:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>vmd<span class="w"> </span>6OLA-assembly1.cif
</pre></div>
</div>
<p>After loading the structure, we must saved it in PDB format by going to <strong>File</strong> &gt; <strong>Save Coordinates</strong> &gt; <strong>Selected Atoms</strong> &gt; <strong>all</strong> option. Enter <strong>File type</strong> as pdb and hit <strong>Save</strong>. Assign filename as <code class="docutils literal notranslate"><span class="pre">6ola_assembly.pdb</span></code> and click <strong>OK</strong>.</p>
<p>You can close VMD now.</p>
<p>Some structures may have structural <strong>issues</strong> or <strong>artifacts</strong> that affect their preparation or simulation. Thus, it is crucial to check the details of the structure and its integrity prior to any step.</p>
<p>Regarding the <strong>6OLA</strong> structure, there are three critical issues that require correction:</p>
<ul class="simple">
<li><p><strong>Issue 1</strong>. The structure contains 60 DNA fragments and each DNA fragment includes a phosphate atom from the subsequent nucleotide. However, this nucleotide was not modeled; only the phosphate was included. Thus, when we attempt to generate the topology using LEaP, the process is stopped since it fails to reconstruct a nucleotide that is absent. This should be corrected by removing the extra phosphate and oxigen atoms. We can correct this using the following command:</p></li>
</ul>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>sed<span class="w"> </span><span class="s1">&#39;/\(P[[:space:]]\+1DC[[:space:]]\+[A-Z][[:space:]]\+1\|OP11DC[[:space:]]\+[A-Z][[:space:]]\+1\|OP21DC[[:space:]]\+[A-Z][[:space:]]\+1\)/d&#39;</span><span class="w"> </span>6ola_assembly.pdb<span class="w"> </span>&gt;<span class="w"> </span>6ola_modificated_1.pdb
</pre></div>
</div>
<p>This line of code creates a new PDB file by removing the lines containing the <strong>P</strong>, <strong>OP1</strong>, and <strong>OP2</strong> atoms from the original structure.</p>
<ul class="simple">
<li><p><strong>Issue 2</strong>. Another issue is the incorrect assignment of residue names according to their protonation state. Following AMBER force field rules, histidine residue name can vary depending on whether the protonation occurs at the delta nitrogen (HID), epsilon nitrogen (HIE), or if the residue is doubly protonated (HIP). Different residue names can also be given to ASP (ASH), GLU (GLH), and CYS (CYM or CYX if they are part of a disulfide bond).</p></li>
</ul>
<p>In order to comply with AMBER force field rules, the PDB file must be edited to update residues protonations. To achieve this, we will first inspect the PDB structure in VMD to identify the relevant protonations.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>In this part of the tutorial, we established the protonation states using the proton locations as provided in the original Biological Assembly of <a class="reference external" href="https://www.rcsb.org/structure/6OLA">6OLA</a> structure.</p>
</div>
<p>Open VMD:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>vmd<span class="w"> </span>6ola_modificated_1.pdb
</pre></div>
</div>
<p>Once in VMD, let’s examine one of the fragments (or protomers) of the PCV2 capsid. To do this, go to the <strong>Graphics</strong> tab and select <strong>Representations</strong>. In the <strong>Graphics Representations</strong> menu, go to <strong>Selected Atoms</strong> and make the selection of histidines from fragment 0 using the following selection command:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>resname<span class="w"> </span>HIS<span class="w"> </span>and<span class="w"> </span>fragment<span class="w"> </span><span class="m">0</span>
</pre></div>
</div>
<p>You should run this command again and check the protons position for all the other protonable residues, such as ASP, GLU, and CYS.</p>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>The only residues in this structure that appear to have distinct protonation states are the three histidines. Use your mouse to navigate and identify which nitrogens are protonated to correctly adjust their residue’s names. Following proton location, HIS 113 is protonated at the epsilon nitrogen (<span class="math notranslate nohighlight">\(N_{\epsilon}\)</span>), HIS 122 is protonated at the delta nitrogen (<span class="math notranslate nohighlight">\(N_{\delta}\)</span>), and HIS 160 is protonated at the epsilon nitrogen (<span class="math notranslate nohighlight">\(N_{\epsilon}\)</span>).</p>
</div>
<p>Close VMD. Use the command below to modify the names of the histidine’s residues:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>sed<span class="w"> </span>-e<span class="w"> </span><span class="s1">&#39;/HIS[[:space:]]\+[A-Z][[:space:]]113/s/HIS/HIE/&#39;</span><span class="w"> </span>-e<span class="w"> </span><span class="s1">&#39;/HIS[[:space:]]\+[A-Z][[:space:]]122/s/HIS/HID/&#39;</span><span class="w"> </span>-e<span class="w"> </span><span class="s1">&#39;/HIS[[:space:]]\+[A-Z][[:space:]]160/s/HIS/HIE/&#39;</span><span class="w"> </span>6ola_modificated_1.pdb<span class="w"> </span>&gt;<span class="w"> </span>6ola_modificated_2.pdb
</pre></div>
</div>
<ul class="simple">
<li><p><strong>Issue 3</strong>. Another important step for this structure is to correct the atom index numbers, ensuring they range from 0 to 99999. Additionally, a <cite>TER</cite> record should be added at the end of each chain, and an <cite>END</cite> record should be included at the end of the structure.</p></li>
</ul>
<p>To do this, we can use the <code class="docutils literal notranslate"><span class="pre">fixpdb.tcl</span></code> VMD script found in the <code class="docutils literal notranslate"><span class="pre">sirah.amber/tutorial/9/</span></code> folder:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>vmd<span class="w"> </span>-dispdev<span class="w"> </span>text<span class="w"> </span>6ola_modificated_2.pdb<span class="w"> </span>-e<span class="w"> </span>./sirah.amber/tutorial/9/fixpdb.tcl
</pre></div>
</div>
<p>Then</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>mv<span class="w"> </span>6ola_modificated_2_final.pdb<span class="w"> </span>6ola_modificated_final.pdb
</pre></div>
</div>
<p>If all stages were completed successfully, the final PDB file <code class="docutils literal notranslate"><span class="pre">6ola_modificated_final.pdb</span></code> is generated.</p>
<ul class="simple">
<li><p><strong>Issue 4 (Issue 2 - Alternative)</strong>. Instead of assigning protonation states by hand, you can use tools that take into account the surrounding residues and the pH conditions to do it automatically. You can use servers, like the <a class="reference external" href="https://www.charmm-gui.org/">CHARMM-GUI server</a> and <a class="reference external" href="https://server.poissonboltzmann.org/pdb2pqr">PDB2PQR server</a>. The <code class="docutils literal notranslate"><span class="pre">6ola_modificated_final.pdb</span></code> should work without any issues, though it may take a while to execute.</p></li>
</ul>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>An additional approach to circumvent potential server execution issues is to submit a minimal fragment of your symmetrized VLP that includes all of the interfaces between subunits as input.</p>
</div>
<div class="admonition caution">
<p class="admonition-title">Caution</p>
<p>The mapping to CG requires the correct protonation state of each residue at a given pH. We recommend using the <a class="reference external" href="https://www.charmm-gui.org/">CHARMM-GUI server</a> and use the <strong>PDB Reader &amp; Manipulator</strong> to prepare your system. An account is required to access any of the CHARMM-GUI Input Generator modules, and it can take up to 24 hours to obtain one.</p>
<p>Other option is the <a class="reference external" href="https://server.poissonboltzmann.org/pdb2pqr">PDB2PQR server</a> and choosing the output naming scheme of AMBER for best compatibility. This server was utilized to generate the <em>PQR</em> file featured in this tutorial. Be aware that modified residues lacking parameters such as: MSE (seleno MET), TPO (phosphorylated TYR), SEP (phosphorylated SER) or others are deleted from the PQR file by the server. In that case, mutate the residues to their unmodified form before submitting the structure to the server.</p>
</div>
<p>Given the scale and complexity of the system, using the <code class="docutils literal notranslate"><span class="pre">pdb2pqr</span></code> script to perform the calculations locally is one method to get around potential server problems. Therefore, we can assign protonation states using pH ~ 7.0 with the following command:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pdb2pqr<span class="w"> </span>--ff<span class="o">=</span>AMBER<span class="w"> </span>--keep-chain<span class="w"> </span>--ffout<span class="o">=</span>AMBER<span class="w"> </span>--titration-state-method<span class="o">=</span>propka<span class="w"> </span>--with-ph<span class="o">=</span><span class="m">7</span>.0<span class="w"> </span>--include-header<span class="w"> </span>./6ola_modificated_final.pdb<span class="w"> </span>./6ola_modificated_final.pqr
</pre></div>
</div>
<p>A PQR file <code class="docutils literal notranslate"><span class="pre">6ola_modificated_final.pqr</span></code> is generated containing protonation state of each residue at the given pH.</p>
</section>
<section id="calculate-system-charge">
<h4>9.1.2 Calculate system charge<a class="headerlink" href="#calculate-system-charge" title="Link to this heading"></a></h4>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>As a result of the system’s size and the various methods available for assigning hydrogen atoms, the most straightforward method of verifying the current charge of the VLP is to calculate the number of charged amino acids and phosphate atoms within the system.</p>
</div>
<p>To do this, we can use the <code class="docutils literal notranslate"><span class="pre">CountCharge.sh</span></code> script found in the <code class="docutils literal notranslate"><span class="pre">sirah.amber/tutorial/9/</span></code> folder by:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>./sirah.amber/tutorial/9/CountCharge.sh<span class="w"> </span>6ola_modificated_final.pdb
</pre></div>
</div>
<p>This script evaluates the number of positively and negatively charged residues in the system and provides the net charge that can be used in later steps of preparing the VLP for simulations.</p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>The file <code class="docutils literal notranslate"><span class="pre">CountCharge.sh</span></code> was made for this system, we suggest that you inspect the  file to gain a more comprehensive understanding of the script and then change it to work with your own system.</p>
</div>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="go">+--------------------+----------+</span>
<span class="go">|      Molecule      | Count    |</span>
<span class="go">+====================+==========+</span>
<span class="go">| LYS (CA)           |  540     |</span>
<span class="go">| ARG (CA)           | 1020     |</span>
<span class="go">| GLU (CA)           |  240     |</span>
<span class="go">| ASP (CA)           |  720     |</span>
<span class="go">| Phosphate (P)      |  180     |</span>
<span class="go">|--------------------|----------|</span>
<span class="go">|Total Positv Charge | 1560     |</span>
<span class="go">|Total Negatv Charge | 1140     |</span>
<span class="go">|Net Charge          |  420     |</span>
<span class="go">+--------------------|----------+</span>
</pre></div>
</div>
</section>
<section id="packing-the-assimetric-unit-to-build-an-entire-vlp-alternative">
<span id="assalt"></span><h4>9.1.3 Packing the assimetric unit to build an entire VLP (Alternative)<a class="headerlink" href="#packing-the-assimetric-unit-to-build-an-entire-vlp-alternative" title="Link to this heading"></a></h4>
<div class="admonition caution">
<p class="admonition-title">Caution</p>
<p>This step should be performed only when the input file (.pdb or .cif) is the asymmetric unit. For more details about asymetric unit and biological assemblies, check the <a class="reference external" href="https://pdb101.rcsb.org/learn/guide-to-understanding-pdb-data/biological-assemblies">RCSB PDB website</a>.</p>
</div>
<p>In complex systems like viruses, some biological assemblies are constructed from the asymmetric unit, which in virology may be represented by a single protein or a small number of them, referred to as a capsomer or protomer. In these cases, it is essential to assemble the whole VLP by employing symmetry operations (rotations and translations) with the symmetry matrices included in the original PDB file. Thus, based on the available structure, it may be necessary to initially assemble the entire VLP.</p>
<p>In <code class="docutils literal notranslate"><span class="pre">sirah.amber/tutorial/9/example_3JCI</span></code> you will find the asymmetric unit of the capsid protein of the PCV2 (PDB code: <a class="reference external" href="https://www.rcsb.org/structure/3JCI">3JCI</a>) along with the necessary files to assemble its VLP.</p>
<p>The first step to assemble a VLP is to minimize the protomer. To do this, we need to create a <code class="docutils literal notranslate"><span class="pre">3JCI.leap</span></code> file to generate the molecular topology and initial coordinate files:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="gp"># </span>Load<span class="w"> </span>AMBER<span class="w"> </span>force<span class="w"> </span>field
<span class="go">source leaprc.protein.ff14SB</span>

<span class="gp"># </span>Load<span class="w"> </span>model
<span class="go">pcv2 = loadPdb 3jci.pdb</span>

<span class="gp"># </span>Save<span class="w"> </span>Parms
<span class="go">saveamberparm pcv2 3JCI_aa.prmtop 3JCI_aa.inpcrd</span>

<span class="gp"># </span>EXIT
<span class="go">quit</span>
</pre></div>
</div>
<p>Run the LEaP application with:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>tleap<span class="w"> </span>-f<span class="w"> </span>3JCI.leap
</pre></div>
</div>
<p>Make a new folder to run the minimization:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>mkdir<span class="w"> </span>-p<span class="w"> </span>run<span class="p">;</span><span class="w"> </span><span class="nb">cd</span><span class="w"> </span>run
</pre></div>
</div>
<p>Run the minimization with sander:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>mpirun<span class="w"> </span>-np<span class="w"> </span><span class="m">4</span><span class="w"> </span>sander.MPI<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../em_cpu.in<span class="w"> </span>-p<span class="w"> </span>../3JCI_aa.prmtop<span class="w"> </span>-c<span class="w"> </span>../3JCI_aa.inpcrd<span class="w"> </span>-ref<span class="w"> </span>../3JCI_aa.inpcrd<span class="w"> </span>-inf<span class="w"> </span>mdinfo<span class="w"> </span>-o<span class="w"> </span>3JCI_aa_em.out<span class="w"> </span>-r<span class="w"> </span>3JCI_aa_em.rst7<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The same input file can be used to run on CPU with the modules <code class="docutils literal notranslate"><span class="pre">pmemd</span></code>, <code class="docutils literal notranslate"><span class="pre">sander</span></code> or <code class="docutils literal notranslate"><span class="pre">sander.MPI</span></code>. Or the GPU implementation, <code class="docutils literal notranslate"><span class="pre">pmemd.cuda</span></code>.</p>
</div>
<p>Once the protomer has been minimized, return to the original folder:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="nb">cd</span><span class="w"> </span>..
</pre></div>
</div>
<p>And build the VLP using the <code class="docutils literal notranslate"><span class="pre">capsid.tcl</span></code> VMD script:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>vmd<span class="w"> </span>-dispdev<span class="w"> </span>text<span class="w"> </span>-e<span class="w"> </span>./sirah.amber/tutorial/9/capsid.tcl
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Due to its complexity, we suggest that you inspect the <code class="docutils literal notranslate"><span class="pre">capsid.tcl</span></code> file to gain a more comprehensive understanding of the procedure.</p>
</div>
<p>That’s it! You can now use VMD to see what the VLP looks like:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>vmd<span class="w"> </span>PCV2_capsid_OK.pdb
</pre></div>
</div>
</section>
</section>
<section id="id82">
<h3>9.2. Build CG representations<a class="headerlink" href="#id82" title="Link to this heading"></a></h3>
<div class="admonition caution">
<p class="admonition-title">Caution</p>
<p>We are using the <code class="docutils literal notranslate"><span class="pre">6ola_modificated_final.pdb</span></code> file for this section of the tutorial, which makes use of the assigned protonation states and hydrogen atoms as given in the original Biological Assembly VLP. But you may also use the automatically calculated protonation states file <code class="docutils literal notranslate"><span class="pre">6ola_modificated_final.pqr</span></code>.</p>
</div>
<p>Map the atomistic structure of VLP that was prepared in the previous step to its CG representation:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>./sirah.amber/tools/CGCONV/cgconv.pl<span class="w"> </span>-i<span class="w"> </span>6ola_modificated_final.pdb<span class="w"> </span>-o<span class="w"> </span>6ola_modificated_final_cg.pdb
</pre></div>
</div>
<p>The input file <code class="docutils literal notranslate"><span class="pre">-i</span></code> 6ola_modificated_final.pdb has the atomistic model of the VLP that has had its structure issues fixed,, while the output <code class="docutils literal notranslate"><span class="pre">-o</span></code> 6ola_modificated_final_cg.pdb is its SIRAH CG representation.</p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>This is the basic usage of the script <strong>cgconv.pl</strong>, you can learn other capabilities from its help by typing:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>./sirah.amber/tools/CGCONV/cgconv.pl<span class="w"> </span>-h
</pre></div>
</div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p><strong>Pay attention to residue names when mapping structures from other atomistic force fields or experimental structures.</strong> Although we provide compatibility for naming schemes in PDB, GMX, GROMOS, CHARMM and OPLS, there might always be some ambiguity in the residue naming, specially regarding protonation states, that may lead to a wrong mapping. For example, SIRAH Tools always maps the residue name “HIS” to a Histidine protonated at the epsilon nitrogen (<span class="math notranslate nohighlight">\(N_{\epsilon}\)</span>) regardless the actual proton placement. Similarly, protonated Glutamic and Aspartic acid residues must be named “GLH” and “ASH”, otherwise they will be treated as negative charged residues. In addition, protonated and disulfide bonded Cysteines must be named “CYS” and “CYX” respectively. These kind of situations need to be carefully checked by the users. In all cases the residues preserve their identity when mapping and back-mapping the structures. Hence, the total charge of the protein should be the same at atomistic and SIRAH levels. You can check the following mapping file to be sure of the compatibility: <code class="docutils literal notranslate"><span class="pre">sirah.amber/tools/CGCONV/maps/sirah_prot.map</span></code>.</p>
</div>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>By default, charged termini are used, but it is possible to set them neutral by renaming the residues from <strong>s</strong>[code] to <strong>a</strong>[code] (Nt-acetylated) or <strong>m</strong>[code] (Ct-amidated) after mapping to CG, where [code] is the root residue name in SIRAH. For example, to set a neutral N-terminal Histidine protonated at epsilon nitrogen (<span class="math notranslate nohighlight">\(N_{\epsilon}\)</span>) rename it from “sHe” to “aHe”.</p>
</div>
<p>Please check both structures using VMD:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>vmd<span class="w"> </span>-m<span class="w"> </span>6ola_modificated_final.pdb<span class="w"> </span>6ola_modificated_final_cg.pdb
</pre></div>
</div>
</section>
<section id="wrapping-up-vlp-system-with-packmol">
<h3>9.3. Wrapping up VLP system with Packmol<a class="headerlink" href="#wrapping-up-vlp-system-with-packmol" title="Link to this heading"></a></h3>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>Before packing the VLP system, it is necessary to estimate the number of coarse-grained water molecules (WT4) per layer (see <strong>Figure 1</strong>), i.e., the inner-virus and outer-virus regions. In this tutorial, we will use the following radius to pack the system with Packmol (left) and LEaP (right).</p>
<figure class="align-center">
<a class="reference internal image-reference" href="../images/water_box_virus.png"><img alt="../images/water_box_virus.png" src="../images/water_box_virus.png" style="width: 100%;" />
</a>
</figure>
<p><strong>Figure 1.</strong> Dimensions of the solvation box used in this tutorial.</p>
</div>
<p>To calculate the approximate number of solvent molecules in each layer, you can use a function in the <code class="docutils literal notranslate"><span class="pre">calc_n.awk</span></code> file that estimates the number of molecules in a given volume.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>awk<span class="w"> </span>-f<span class="w"> </span>./sirah.amber/tutorial/9/calc_n.awk<span class="w"> </span>./sirah.amber/tutorial/9/layers_radius.dat
</pre></div>
</div>
<p>The two lines in the <code class="docutils literal notranslate"><span class="pre">layers_radius.dat</span></code> are read by this code. One line calculates the number of WT4 outside the VLP, and the other line calculates the number of WT4 inside the VLP (see <strong>Figure 1</strong>).</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Each line of the <code class="docutils literal notranslate"><span class="pre">layers_radius.dat</span></code> has five parameters. The first and second ones are WT4’s molecular weight and density, respectively. The third and fourth parameters are the starting and end point of the compartment and defines its radius in Angstroms (Å). The fourth value needs to be greater than the third. The fifth and final parameter gives a label to each compartment. Depending on your system, you may need to adjust the radius values in the <code class="docutils literal notranslate"><span class="pre">layers_radius.dat</span></code> file.</p>
</div>
<p>The output of this command is:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="nv">WT4_out</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">18235</span>
<span class="nv">WT4_in</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">6393</span>
</pre></div>
</div>
<p>The next step is to use the program Packmol to put together the system’s parts: VLP, solvent (both inside and outside), and ions (inside) (see <strong>Figure 1</strong>).</p>
<p>To accurately add the right number of solvent and ion molecules using Packmol, it is important to consider the charge of the VLP. As previously determined, the net charge of the VLP is <strong>+420</strong>.</p>
<p>To balance this, we need to introduce at least <strong>420</strong> negative ions (Cl⁻) into the inner core of the VLP. However, previous simulations conducted by us and other researchers demonstrated through trial and error that the inner core of the virion must have a minimum of <strong>620 Cl⁻</strong> ions to not only neutralize the VLP’s charge but also compensate for the lack of a complete genome within the particle. As a result, the inner solvent layer contains:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>620 Cl⁻ beads (ClW in its CG naming).

6393 - 620 = 5773 WT4 molecules at the inner layer.
</pre></div>
</div>
<div class="admonition caution">
<p class="admonition-title">Caution</p>
<p>It is important to be aware that the charges may vary when using the <code class="docutils literal notranslate"><span class="pre">6ola_modificated_final.pqr</span></code>, and as a result, the number of ions necessary to neutralize the charge imbalance may differ.</p>
</div>
<p>This information is provided to the <code class="docutils literal notranslate"><span class="pre">PCV2.pkm</span></code> file, which is then utilized by Packmol to pack the system. Use the following command to execute Packmol:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>packmol<span class="w"> </span>&lt;<span class="w"> </span>./sirah.amber/tutorial/9/PCV2.pkm<span class="w"> </span>&gt;&gt;<span class="w"> </span>PCV2_packmol.log<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>Edit the <code class="docutils literal notranslate"><span class="pre">PCV2.pkm</span></code> file to fix the number of solvent and ions according to your system.</p>
</div>
<p>In order for LEaP to recognize proteins and DNA as distinct molecules, we also need to add <cite>TER</cite> records at the end of each chain and a <cite>END</cite> record at the end of the PDB file.</p>
<p>Thus, we can use the <code class="docutils literal notranslate"><span class="pre">fixpdb.tcl</span></code> VMD script found in the <code class="docutils literal notranslate"><span class="pre">sirah.amber/tutorial/9/</span></code> folder again:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>vmd<span class="w"> </span>-dispdev<span class="w"> </span>text<span class="w"> </span>PCV2_6OLA_CG_solv.pdb<span class="w"> </span>-e<span class="w"> </span>./sirah.amber/tutorial/9/fixpdb.tcl
</pre></div>
</div>
<p>A PDB file <code class="docutils literal notranslate"><span class="pre">PCV2_6OLA_CG_solv_final.pdb</span></code> is generated and ready for LEaP.</p>
<p>From now on it is just normal Amber stuff!</p>
</section>
<section id="id83">
<h3>9.4 Prepare LEaP input<a class="headerlink" href="#id83" title="Link to this heading"></a></h3>
<p>In this step, we will set the truncated octahedral box and the salt concentration to 150 mM with LEaP. There are 35,720 WT4 molecules in the system after LEaP (an increase of 11,712 WT4). To reach 150 mM NaCl concentration on the outer layer of WT4, we need to add:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="mi">35720</span><span class="o">/</span><span class="mi">34</span> <span class="o">=</span> <span class="mi">1050</span> <span class="n">ionic</span> <span class="n">pairs</span>
</pre></div>
</div>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<p>The available electrolyte species in SIRAH force field are: <code class="docutils literal notranslate"><span class="pre">Na⁺</span></code> (NaW), <code class="docutils literal notranslate"><span class="pre">K⁺</span></code> (KW) and <code class="docutils literal notranslate"><span class="pre">Cl⁻</span></code> (ClW) which represent solvated ions in solution. One ion pair (e.g., NaW-ClW) each 34 WT4 molecules results in a salt concentration of ~0.15M (see <a class="reference internal" href="#appendix"><span class="std std-ref">Appendix</span></a> for details). Counterions were added according to <a class="reference external" href="https://pubs.acs.org/doi/10.1021/acs.jctc.9b00953">Machado et al.</a>.</p>
</div>
<p>However, keep in mind that the VLP currently has a 200 negative charge:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="mi">420</span> <span class="o">-</span> <span class="mi">620</span> <span class="o">=</span> <span class="o">-</span><span class="mi">200</span>
</pre></div>
</div>
<p>So, in order to achieve a more balanced system, we can split this charge. In this case, the number of counterions to add is:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="mi">1</span><span class="p">,</span><span class="mi">050</span> <span class="o">+</span> <span class="mi">100</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span><span class="mi">150</span> <span class="n">NaW</span>

<span class="mi">1</span><span class="p">,</span><span class="mi">050</span> <span class="o">-</span> <span class="mi">100</span> <span class="o">=</span> <span class="mi">950</span> <span class="n">ClW</span>
</pre></div>
</div>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<p>Counterions were added according to <a class="reference external" href="https://pubs.acs.org/doi/10.1021/acs.jctc.9b00953">Machado et al.</a>.</p>
</div>
<p>Use a text editor to create the file <code class="docutils literal notranslate"><span class="pre">6OLA_gensystem.leap</span></code> including the following lines:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="gp"># </span>Load<span class="w"> </span>SIRAH<span class="w"> </span>force<span class="w"> </span>field
<span class="go">addPath ./sirah.amber</span>
<span class="go">source leaprc.sirah</span>

<span class="gp"># </span>Load<span class="w"> </span>model
<span class="go">pcv2 = loadpdb PCV2_6OLA_CG_solv_final.pdb</span>

<span class="gp"># </span>Info<span class="w"> </span>on<span class="w"> </span>system<span class="w"> </span>charge
<span class="go">charge pcv2</span>

<span class="gp"># </span>Add<span class="w"> </span>solvent<span class="w"> </span>and<span class="w"> </span>build<span class="w"> </span>a<span class="w"> </span>truncated<span class="w"> </span>octahedral<span class="w"> </span>box
<span class="gp"># </span>Tuned<span class="w"> </span>solute-solvent<span class="w"> </span>closeness<span class="w"> </span><span class="k">for</span><span class="w"> </span>best<span class="w"> </span>hydration
<span class="go">solvateOct pcv2 WT4BOX 17 0.9</span>

<span class="gp"># </span>Add<span class="w"> </span>counterions<span class="w"> </span>and<span class="w"> </span><span class="m">0</span>.15M<span class="w"> </span>NaCl
<span class="go">addIonsRand pcv2 NaW 1150 ClW 950</span>

<span class="gp"># </span>Save<span class="w"> </span>Parms
<span class="go">saveAmberParmNetcdf pcv2 PCV2_6OLA_CG.prmtop PCV2_6OLA_CG.ncrst</span>

<span class="gp"># </span>Save<span class="w"> </span>pdb<span class="w"> </span>file
<span class="go">savepdb pcv2 PCV2_6OLA_CG.pdb</span>

<span class="gp"># </span>EXIT
<span class="go">quit</span>
</pre></div>
</div>
<div class="admonition caution">
<p class="admonition-title">Caution</p>
<p>There is only one CYS in this VLP, and there are no disulfide bridges in it. However, if you are working with a system that includes these bonds, please take note of the following:</p>
<p>Each disulfide bond must be defined explicitly in LEaP using the command bond, e.g.: “<em>bond unit.ri.BSG unit.rj.BSG</em>”. Where <em>ri</em> and <em>rj</em> correspond to the residue index in the topology file starting from 1, which may differ from the biological sequence in the PDB file. You can try the command <em>pdb4amber</em> to get those indexes from the atomistic structure, but be aware that it may not work if the Cysteine residues are too far away:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pdb4amber<span class="w"> </span>-i<span class="w"> </span>YOUR_VLP.pqr<span class="w"> </span>-o<span class="w"> </span>YOUR_VLP_aa.pdb<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span>cat<span class="w"> </span>YOUR_VLP_aa_sslink
</pre></div>
</div>
</div>
</section>
<section id="id84">
<h3>9.5 Run LEaP<a class="headerlink" href="#id84" title="Link to this heading"></a></h3>
<p>Run LEaP to generate the molecular topology and initial coordinate files:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>tleap<span class="w"> </span>-f<span class="w"> </span>6OLA_gensystem.leap
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Warning messages about long, triangular or square bonds in <code class="docutils literal notranslate"><span class="pre">leap.log</span></code> file are fine and expected due to the CG topology of some residues.</p>
</div>
<p>This should create a topology file <code class="docutils literal notranslate"><span class="pre">PCV2_6OLA_CG.prmtop</span></code>, a coordinate file <code class="docutils literal notranslate"><span class="pre">PCV2_6OLA_CG.ncrst</span></code> and a pdb file <code class="docutils literal notranslate"><span class="pre">PCV2_6OLA_CG.pdb</span></code>.</p>
<p>Use VMD to check how the CG model looks like and particularly the presence of disulfide bonds whether your system has them:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>vmd<span class="w"> </span>PCV2_6OLA_CG.prmtop<span class="w"> </span>PCV2_6OLA_CG.ncrst<span class="w"> </span>-e<span class="w"> </span>./sirah.amber/tools/sirah_vmdtk.tcl
</pre></div>
</div>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>VMD assigns default radius to unknown atom types, the script <code class="docutils literal notranslate"><span class="pre">sirah_vmdtk.tcl</span></code> sets the right
ones, according to the CG representation. It also provides a kit of useful selection macros, coloring methods and backmapping utilities.
Use the command <code class="docutils literal notranslate"><span class="pre">sirah_help</span></code> in the Tcl/Tk console of VMD to access the manual pages. To learn about SIRAH Tools’ capabilities, you can also go to the <a class="reference internal" href="Tutorials%20sirahtools.html#sirah-tools"><span class="std std-ref">SIRAH Tools tutorial</span></a>.</p>
</div>
</section>
<section id="id85">
<h3>9.6. Run the simulation<a class="headerlink" href="#id85" title="Link to this heading"></a></h3>
<p>Make a new folder for the run:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>mkdir<span class="w"> </span>-p<span class="w"> </span>run<span class="p">;</span><span class="w"> </span><span class="nb">cd</span><span class="w"> </span>run
</pre></div>
</div>
<p>The folder <code class="docutils literal notranslate"><span class="pre">sirah.amber/tutorial/9/</span></code> contains typical input files for energy minimization
(<code class="docutils literal notranslate"><span class="pre">VLP_cg_em1.in</span></code> to <code class="docutils literal notranslate"><span class="pre">VLP_cg_em5.in</span></code>), equilibration (<code class="docutils literal notranslate"><span class="pre">VLP_cg_eq1.in</span></code> to <code class="docutils literal notranslate"><span class="pre">VLP_cg_eq4.in</span></code>) and production (<code class="docutils literal notranslate"><span class="pre">VLP_cg_md1.in</span></code>) runs. Please check carefully the
input flags therein, in particular the definition of flag <em>chngmask=0</em> at <em>&amp;ewald</em> section is <strong>mandatory</strong>.</p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<blockquote>
<div><p><strong>Some commonly used flags in Amber</strong></p>
</div></blockquote>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">-i</span></code>: Input file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-o</span></code>: Output file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-p</span></code>: Parameter/topology file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-c</span></code>: Coordinate file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-r</span></code>: Restart file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-x</span></code>: Trajectory file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-ref</span></code>: Reference file</p></li>
</ul>
</div>
<div class="admonition caution">
<p class="admonition-title">Caution</p>
<p>These input files are executed by the <strong>GPU</strong> implementation of <code class="docutils literal notranslate"><span class="pre">pmemd.cuda</span></code>. Other available modules are <code class="docutils literal notranslate"><span class="pre">sander</span></code> or <code class="docutils literal notranslate"><span class="pre">pmemd</span></code>, which are both <strong>CPU</strong> implementations of Amber.</p>
</div>
<p><strong>Energy Minimization of solvent and ions by restraining the whole protein and DNA fragments:</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/9/VLP_cg_em1.in<span class="w"> </span>-p<span class="w"> </span>PCV2_6OLA_CG.prmtop<span class="w"> </span>-c<span class="w"> </span>PCV2_6OLA_CG.ncrst<span class="w"> </span>-ref<span class="w"> </span>PCV2_6OLA_CG.ncrst<span class="w"> </span>-o<span class="w"> </span>PCV2_6OLA_CG_em1.out<span class="w"> </span>-r<span class="w"> </span>PCV2_6OLA_CG_em1.ncrst<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
<p><strong>Energy Minimization of side chains and solvent by restraining the backbone of proteins and DNA:</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/9/VLP_cg_em2.in<span class="w"> </span>-p<span class="w"> </span>PCV2_6OLA_CG.prmtop<span class="w"> </span>-c<span class="w"> </span>PCV2_6OLA_CG_em1.ncrst<span class="w"> </span>-ref<span class="w"> </span>PCV2_6OLA_CG_em1.ncrst<span class="w"> </span>-o<span class="w"> </span>PCV2_6OLA_CG_em2.out<span class="w"> </span>-r<span class="w"> </span>PCV2_6OLA_CG_em2.ncrst<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
<p><strong>Energy Minimization of whole system:</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/9/VLP_cg_em3.in<span class="w"> </span>-p<span class="w"> </span>PCV2_6OLA_CG.prmtop<span class="w"> </span>-c<span class="w"> </span>PCV2_6OLA_CG_em2.ncrst<span class="w"> </span>-ref<span class="w"> </span>PCV2_6OLA_CG_em2.ncrst<span class="w"> </span>-o<span class="w"> </span>PCV2_6OLA_CG_em3.out<span class="w"> </span>-r<span class="w"> </span>PCV2_6OLA_CG_em3.ncrst<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
<p><strong>Energy Minimization of side chains and solvent by restraining the backbone:</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/9/VLP_cg_em4.in<span class="w"> </span>-p<span class="w"> </span>PCV2_6OLA_CG.prmtop<span class="w"> </span>-c<span class="w"> </span>PCV2_6OLA_CG_em3.ncrst<span class="w"> </span>-ref<span class="w"> </span>PCV2_6OLA_CG_em3.ncrst<span class="w"> </span>-o<span class="w"> </span>PCV2_6OLA_CG_em4.out<span class="w"> </span>-r<span class="w"> </span>PCV2_6OLA_CG_em4.ncrst<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
<p><strong>Energy Minimization of whole system:</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/9/VLP_cg_em5.in<span class="w"> </span>-p<span class="w"> </span>PCV2_6OLA_CG.prmtop<span class="w"> </span>-c<span class="w"> </span>PCV2_6OLA_CG_em4.ncrst<span class="w"> </span>-ref<span class="w"> </span>PCV2_6OLA_CG_em4.ncrst<span class="w"> </span>-o<span class="w"> </span>PCV2_6OLA_CG_em5.out<span class="w"> </span>-r<span class="w"> </span>PCV2_6OLA_CG_em5.ncrst<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
<p><strong>Solvent Equilibration (NPT) by restraining the whole protein and DNA fragments:</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/9/VLP_cg_eq1.in<span class="w"> </span>-p<span class="w"> </span>PCV2_6OLA_CG.prmtop<span class="w"> </span>-c<span class="w"> </span>PCV2_6OLA_CG_em5.ncrst<span class="w"> </span>-ref<span class="w"> </span>PCV2_6OLA_CG_em5.ncrst<span class="w"> </span>-o<span class="w"> </span>PCV2_6OLA_CG_eq1.out<span class="w"> </span>-r<span class="w"> </span>PCV2_6OLA_CG_eq1.ncrst<span class="w"> </span>-x<span class="w"> </span>PCV2_6OLA_CG_eq1.nc<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
<p><strong>Soft equilibration to improve side chain solvation (NPT) by restraining the backbone of proteins and DNA:</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/9/VLP_cg_eq2.in<span class="w"> </span>-p<span class="w"> </span>PCV2_6OLA_CG.prmtop<span class="w"> </span>-c<span class="w"> </span>PCV2_6OLA_CG_eq1.ncrst<span class="w"> </span>-ref<span class="w"> </span>PCV2_6OLA_CG_eq1.ncrst<span class="w"> </span>-o<span class="w"> </span>PCV2_6OLA_CG_eq2.out<span class="w"> </span>-r<span class="w"> </span>PCV2_6OLA_CG_eq2.ncrst<span class="w"> </span>-x<span class="w"> </span>PCV2_6OLA_CG_eq2.nc<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
<p><strong>Soft solvent Equilibration (NPT) by restraining the proteins’ backbone:</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/9/VLP_cg_eq3.in<span class="w"> </span>-p<span class="w"> </span>PPCV2_6OLA_CG.prmtop<span class="w"> </span>-c<span class="w"> </span>PCV2_6OLA_CG_eq2.ncrst<span class="w"> </span>-ref<span class="w"> </span>PCV2_6OLA_CG_eq2.ncrst<span class="w"> </span>-o<span class="w"> </span>PCV2_6OLA_CG_eq3.out<span class="w"> </span>-r<span class="w"> </span>PCV2_6OLA_CG_eq3.ncrst<span class="w"> </span>-x<span class="w"> </span>PCV2_6OLA_CG_eq3.nc<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
<p><strong>Whole System Equilibration (NPT) without restraints:</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/9/VLP_cg_eq4.in<span class="w"> </span>-p<span class="w"> </span>PCV2_6OLA_CG.prmtop<span class="w"> </span>-c<span class="w"> </span>PCV2_6OLA_CG_eq3.ncrst<span class="w"> </span>-ref<span class="w"> </span>PCV2_6OLA_CG_eq3.ncrst<span class="w"> </span>-o<span class="w"> </span>PCV2_6OLA_CG_eq4.out<span class="w"> </span>-r<span class="w"> </span>PCV2_6OLA_CG_eq4.ncrst<span class="w"> </span>-x<span class="w"> </span>PCV2_6OLA_CG_eq4.nc<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
<p><strong>Production (1000ns):</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/9/VLP_cg_md1.in<span class="w"> </span>-p<span class="w"> </span>PCV2_6OLA_CG.prmtop<span class="w"> </span>-c<span class="w"> </span>PCV2_6OLA_CG_eq4.ncrst<span class="w"> </span>-ref<span class="w"> </span>PCV2_6OLA_CG_eq4.ncrst<span class="w"> </span>-o<span class="w"> </span>PCV2_6OLA_CG_md1.out<span class="w"> </span>-r<span class="w"> </span>PCV2_6OLA_CG_md1.ncrst<span class="w"> </span>-x<span class="w"> </span>PCV2_6OLA_CG_md1.nc<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
</section>
<section id="id86">
<h3>9.7. Visualizing the simulation<a class="headerlink" href="#id86" title="Link to this heading"></a></h3>
<p>That’s it! Now you can analyze the trajectory.
Process the output trajectory to account for the Periodic Boundary Conditions (PBC):</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;center (@GN,GC,GO)\n image familiar\n go\n quit\n&quot;</span><span class="w"> </span><span class="p">|</span><span class="w"> </span>cpptraj<span class="w"> </span>-p<span class="w"> </span>PCV2_6OLA_CG.prmtop<span class="w"> </span>-y<span class="w"> </span>PCV2_6OLA_CG_md1.nc<span class="w"> </span>-x<span class="w"> </span>PCV2_6OLA_CG_md1_pbc.nc<span class="w"> </span>--interactive
</pre></div>
</div>
<p>Load the processed trajectory in VMD:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">vmd</span> <span class="n">PCV2_6OLA_CG</span><span class="o">.</span><span class="n">prmtop</span> <span class="n">PCV2_6OLA_CG</span><span class="o">.</span><span class="n">ncrst</span> <span class="n">PCV2_6OLA_CG_md1_pbc</span><span class="o">.</span><span class="n">nc</span> <span class="o">-</span><span class="n">e</span> <span class="o">../</span><span class="n">sirah</span><span class="o">.</span><span class="n">amber</span><span class="o">/</span><span class="n">tools</span><span class="o">/</span><span class="n">sirah_vmdtk</span><span class="o">.</span><span class="n">tcl</span>
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The file <code class="docutils literal notranslate"><span class="pre">sirah_vmdtk.tcl</span></code> is a Tcl script that is part of SIRAH Tools and contains the macros to properly visualize the coarse-grained structures in VMD. Use the command <code class="docutils literal notranslate"><span class="pre">sirah-help</span></code> in the Tcl/Tk console of VMD to access the manual pages. To learn about SIRAH Tools’ capabilities, you can also go to the <a class="reference internal" href="Tutorials%20sirahtools.html#sirah-tools"><span class="std std-ref">SIRAH Tools tutorial</span></a>.</p>
</div>
</section>
</section>
<section id="appendix">
<span id="id87"></span><h2>Appendix<a class="headerlink" href="#appendix" title="Link to this heading"></a></h2>
<section id="calculating-ionic-concentrations">
<h3>Calculating ionic concentrations<a class="headerlink" href="#calculating-ionic-concentrations" title="Link to this heading"></a></h3>
<div class="math notranslate nohighlight">
\[\]</div>
<script type="text/javascript" src="http://localhost/mathjax/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>\[\rho_{\text{WT4}} = \rho_{\text{H}_2\text{O}} = 1000 \, \text{g/L} \]

 \[MW_{\text{H}_2\text{O}} = 18 \, \text{g/mol} \]

 \[1 \text{WT4} \sim 11 \text{H}_2\text{O} \]

 \[\text{M}=\frac{\text{mol}}{V} \quad ; \quad \text{n}={\text{N}_A} \quad ; \quad \rho = \frac{\text{m}}{V} \quad ; \quad \text{m} = \text{mol} \cdot \text{M} \cdot \text{W}\]

 \[\text{V} = \frac{\text{m}}{\rho} = \frac{\text{mol} \cdot \text{MW}_{\text{H}_2\text{O}}}{\rho} = \frac{\text{n}_{\text{H}_2\text{O}} \cdot \text{MW}_{\text{H}_2\text{O}}}{{\text{N}_A} \cdot \rho}\]


 \[\text{M}=\frac{\text{mol}}{V} = \frac{\text{n}_{\text{ion}}}{\text{N}_{\text{A}} \cdot \text{V}} = \frac{\text{n}_{\text{ion}}}{\text{N}_{\text{A}}} \frac{\text{N}_{\text{A}} \cdot \rho}{\text{n}_{\text{H}_2\text{O}} \cdot \text{MW}_{\text{H}_2\text{O}}} = \frac{\text{n}_{\text{ion}} \cdot 1000}{\text{n}_{\text{WT}_4} \cdot \text{(11)} \cdot \text{(18)}} \sim 5 \frac{\text{n}_{\text{ion}}}{\text{n}_{\text{WT}_4}}  \]<p>Number of <em>WT4</em> molecules per ion at 0.15M: <span class="raw-latex">\(\text{n}_{WT4} = 5 {{\text{n}_{\text{ion}}} \over \text{M}} = {5(1) \over 0.15} \sim 34 \)</span></p>
</section>
<section id="available-mapping-files">
<span id="table"></span><h3>Available mapping files<a class="headerlink" href="#available-mapping-files" title="Link to this heading"></a></h3>
<p>Table 1. Available mapping files (MAPs) at folder <code class="docutils literal notranslate"><span class="pre">sirah.amber/tools/CGCONV/maps/</span></code> for converting atomistic lipid structures to SIRAH models. <strong>Important!</strong> MAPs can not inter-convert different name conventions, e.g. amber_lipid.map won’t generate fragment-based residues from residue-based force fields. Due to possible nomenclature conflicts, users are advised to check and modify the MAPs as required.</p>
<table class="docutils align-default">
<tbody>
<tr class="row-odd"><td><p><strong>Map</strong></p></td>
<td><p><strong>Type</strong>*</p></td>
<td><p><strong>Compatibility</strong></p></td>
<td><p><strong>Source</strong></p></td>
</tr>
<tr class="row-even"><td><p>amber_lipid.map</p></td>
<td><p>F</p></td>
<td><p>AMBER Lipid11-17 force fields</p></td>
<td><div class="line-block">
<div class="line"><a class="reference external" href="http://ambermd.org/">AMBER</a></div>
<div class="line"><a class="reference external" href="https://software.acellera.com/htmd/index.html">HTMD</a></div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p>GAFF_lipid.map</p></td>
<td><p>R</p></td>
<td><p>AMBER GAFF force field</p></td>
<td><p><a class="reference external" href="https://lipidbook.org/">LipidBook</a></p></td>
</tr>
<tr class="row-even"><td><p>charmm_lipid.map</p></td>
<td><p>R</p></td>
<td><div class="line-block">
<div class="line">CHARMM 27/36 force field, and “CHARMM</div>
<div class="line">compatible” GAFF nomenclature</div>
</div>
</td>
<td><div class="line-block">
<div class="line"><a class="reference external" href="https://charmm-gui.org/">CHARMM-GUI</a></div>
<div class="line"><a class="reference external" href="https://www.gromacs.org/">GROMACS</a></div>
<div class="line"><a class="reference external" href="https://lipidbook.org/">LipidBook</a></div>
<div class="line"><a class="reference external" href="http://bioinf.modares.ac.ir/software/mb/">MemBuilder</a></div>
<div class="line"><a class="reference external" href="https://software.acellera.com/htmd/index.html">HTMD</a></div>
<div class="line"><a class="reference external" href="https://www.ks.uiuc.edu/Research/vmd/">VMD</a></div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p>slipids.map</p></td>
<td><p>R</p></td>
<td><p>Stockholm lipids force field</p></td>
<td><div class="line-block">
<div class="line"><a class="reference external" href="http://www.fos.su.se/~sasha/SLipids/About.html">SLIPIDS</a></div>
<div class="line"><a class="reference external" href="http://bioinf.modares.ac.ir/software/mb/">MemBuilder</a></div>
</div>
</td>
</tr>
<tr class="row-even"><td><p>OPLSA-AA_2014_lipid.map</p></td>
<td><p>R</p></td>
<td><p>All-atoms lipids for OPLS force field</p></td>
<td><div class="line-block">
<div class="line"><a class="reference external" href="https://doi.org/10.1021/jp5016627">Maciejewski</a></div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p>OPLSA-UA_lipid.map</p></td>
<td><p>R</p></td>
<td><p>United-atom lipids for OPLS force field</p></td>
<td><p><a class="reference external" href="https://lipidbook.org/">LipidBook</a></p></td>
</tr>
<tr class="row-even"><td><p>GROMOS43a1_lipid.map</p></td>
<td><p>R</p></td>
<td><div class="line-block">
<div class="line">United-atom lipids for GROMOS 43a1 and</div>
<div class="line">CKP force fields</div>
</div>
</td>
<td><div class="line-block">
<div class="line"><a class="reference external" href="https://lipidbook.org/">LipidBook</a></div>
<div class="line"><a class="reference external" href="http://bioinf.modares.ac.ir/software/mb/">MemBuilder</a></div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p>GROMOS43a1-s3_lipid.map</p></td>
<td><p>R</p></td>
<td><div class="line-block">
<div class="line">United-atom lipids for GROMOS 43a1-s3</div>
<div class="line">force field</div>
</div>
</td>
<td><div class="line-block">
<div class="line"><a class="reference external" href="https://www.gromacs.org/">GROMACS</a></div>
<div class="line"><a class="reference external" href="https://lipidbook.org/">LipidBook</a></div>
<div class="line"><a class="reference external" href="http://bioinf.modares.ac.ir/software/mb/">MemBuilder</a></div>
</div>
</td>
</tr>
<tr class="row-even"><td><p>GROMOS53a6_lipid.map</p></td>
<td><p>R</p></td>
<td><div class="line-block">
<div class="line">United-atom lipids for GROMOS 53a6</div>
<div class="line">force field</div>
</div>
</td>
<td><div class="line-block">
<div class="line"><a class="reference external" href="https://www.gromacs.org/">GROMACS</a></div>
<div class="line"><a class="reference external" href="https://lipidbook.org/">LipidBook</a></div>
<div class="line"><a class="reference external" href="http://bioinf.modares.ac.ir/software/mb/">MemBuilder</a></div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p>tieleman_lipid.map</p></td>
<td><p>R</p></td>
<td><div class="line-block">
<div class="line">Berger lipids as implemented by</div>
<div class="line">Tieleman et al. for GROMOS</div>
<div class="line">force fields.</div>
</div>
</td>
<td><div class="line-block">
<div class="line"><a class="reference external" href="https://doi.org/10.1021/ja0624321">Tieleman</a></div>
<div class="line"><a class="reference external" href="https://lipidbook.org/">LipidBook</a></div>
</div>
</td>
</tr>
</tbody>
</table>
<p>* Fragment-based (F) or Residue-based (R) topology.</p>
</section>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="Developers.html" class="btn btn-neutral float-left" title="Developers" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="Tutorials%20gromacs.html" class="btn btn-neutral float-right" title="GROMACS" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2023, BA, SL, SM.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>