<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>SIRAH tools &mdash; SIRAH Force Field v_1.0 documentation</title>
      <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
      <link rel="stylesheet" href="_static/copybutton.css" type="text/css" />
      <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script src="_static/jquery.js?v=5d32c60e"></script>
        <script src="_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
        <script src="_static/documentation_options.js?v=3060a628"></script>
        <script src="_static/doctools.js?v=888ff710"></script>
        <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
        <script src="_static/clipboard.min.js?v=a7894cd8"></script>
        <script src="_static/copybutton.js?v=f281be69"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="prev" title="NAMD" href="Tutorials%20namd.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="index.html" class="icon icon-home">
            SIRAH Force Field
          </a>
              <div class="version">
                v_1.0
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Manual</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="About%20SIRAH.html">About SIRAH</a></li>
<li class="toctree-l1"><a class="reference internal" href="Citation.html">Citations</a></li>
<li class="toctree-l1"><a class="reference internal" href="Further%20reading.html">Further Reading</a></li>
<li class="toctree-l1"><a class="reference internal" href="FAQ.html">FAQs</a></li>
<li class="toctree-l1"><a class="reference internal" href="Developers.html">Developers</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Tutorials</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="Tutorials%20amber.html">AMBER</a></li>
<li class="toctree-l1"><a class="reference internal" href="Tutorials%20gromacs.html">GROMACS</a></li>
<li class="toctree-l1"><a class="reference internal" href="Tutorials%20namd.html">NAMD</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">SIRAH tools</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#vmd-plugin">VMD Plugin</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#visualization-of-cg-systems">Visualization of CG systems</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#sirah-macros">SIRAH macros</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#structural-analysis-of-cg-systems">Structural analysis of CG systems</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#secondary-structure-analysis">Secondary structure analysis</a></li>
<li class="toctree-l4"><a class="reference internal" href="#backmapping-analysis">Backmapping analysis</a></li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">SIRAH Force Field</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">SIRAH tools</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/Tutorials sirahtools.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="sirah-tools">
<span id="id1"></span><h1>SIRAH tools<a class="headerlink" href="#sirah-tools" title="Link to this heading"></a></h1>
<section id="vmd-plugin">
<span id="id2"></span><h2>VMD Plugin<a class="headerlink" href="#vmd-plugin" title="Link to this heading"></a></h2>
<p>This tutorial shows how to visualize and analize trajectory files using the <strong>SIRAH Tools Plugin</strong> for VMD. The main reference
for this tutorial is <a class="reference external" href="https://academic.oup.com/bioinformatics/article/32/10/1568/1743152">Machado &amp; Pantano</a>. The protein−DNA complex used here is the 3′ Repair Exonuclease 1 (TREX 1, PDB id: <a class="reference external" href="https://www.rcsb.org/structure/5YWS">5YWS</a>) and its simulation was previously dicussed by <a class="reference external" href="https://doi.org/10.1021/acs.jcim.0c00160">Klein et al.</a>. We strongly advise you to read these articles before starting the tutorial.</p>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>This tutorial requires the use of <strong>VMD</strong> and <strong>AmberTools</strong>. Please install these applications if you do not already have them. If you need to install any of these applications, please visit the following websites for installation instructions and additional information: <a class="reference external" href="https://ambermd.org/GetAmber.php#ambertools">Install AmberTools</a> and <a class="reference external" href="https://www.ks.uiuc.edu/Research/vmd/current/ig/node6.html">Installation VMD guide</a>.</p>
</div>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>In this tutorial, we will focus on the <strong>sirah_vmdtk.tcl</strong> plugin. The utilization of the SIRAH Tools scripts <code class="docutils literal notranslate"><span class="pre">cgconv.pl</span></code> to map AA to CG, and <code class="docutils literal notranslate"><span class="pre">g_top2psf.pl</span></code> to convert GROMACS’ topologies to PSF files, is covered elsewhere in the documentation. We recommend that you review the other tutorial sections.</p>
</div>
<section id="visualization-of-cg-systems">
<h3>Visualization of CG systems<a class="headerlink" href="#visualization-of-cg-systems" title="Link to this heading"></a></h3>
<p>One of the features of the SIRAH Tools plugin for VMD, <code class="docutils literal notranslate"><span class="pre">sirah_vmdtk.tcl</span></code>, is that it improves the visualization and analysis of SIRAH’s CG trajectories. This plugin sets the correct van der Waals radii, sets the correct coloring code by atom and residue types, and has macros for selecting molecular components on SIRAH trajectories.</p>
<p>After processing the output trajectory to account for the Periodic Boundary Conditions (PBC) (see <a class="reference internal" href="Tutorials%20amber.html#id1"><span class="std std-ref">Amber</span></a>, <a class="reference internal" href="Tutorials%20gromacs.html#id1"><span class="std std-ref">GROMACS</span></a> or <a class="reference internal" href="Tutorials%20namd.html#id1"><span class="std std-ref">NAMD</span></a> tutorials for examples of how you can do this), load the processed trajectory and the <code class="docutils literal notranslate"><span class="pre">sirah_vmdtk.tcl</span></code> file in VMD:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>vmd<span class="w"> </span>../5YWS_cg.psf<span class="w"> </span>../5YWS_cg.gro<span class="w"> </span>5YWS_cg_md_pbc.nc<span class="w"> </span>-e<span class="w"> </span>../sirah.amber/tools/sirah_vmdtk.tcl
</pre></div>
</div>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>You can also load <code class="docutils literal notranslate"><span class="pre">sirah_vmdtk.tcl</span></code> inside VMD. Go to <em>Extensions</em> &gt; <em>Tk Console</em> and enter:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="nb">source</span><span class="w"> </span>../sirah.amber/tools/sirah_vmdtk.tcl
</pre></div>
</div>
<p>or</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="nb">source</span><span class="w"> </span>../sirah.ff/tools/sirah_vmdtk.tcl
</pre></div>
</div>
</div>
<p>If <code class="docutils literal notranslate"><span class="pre">sirah_vmdtk.tcl</span></code> is not loaded in VMD, the trajectory will still be loaded, but the correct molecular connectivity, bead size, charges, etc. will not be generated (see <strong>Figure 1</strong>).</p>
<figure class="align-center" id="id3">
<a class="reference internal image-reference" href="_images/sirah_tools_1.png"><img alt="_images/sirah_tools_1.png" src="_images/sirah_tools_1.png" style="width: 100%;" /></a>
<figcaption>
<p><span class="caption-text"><strong>Figure 1.</strong> SIRAH CG simulation loaded in VMD. A) The trajectory imported without the <code class="docutils literal notranslate"><span class="pre">sirah_vmdtk.tcl</span></code> plugin. B) Trajectory imported using the <code class="docutils literal notranslate"><span class="pre">sirah_vmdtk.tcl</span></code> plugin.</span><a class="headerlink" href="#id3" title="Link to this image"></a></p>
</figcaption>
</figure>
<section id="sirah-macros">
<h4>SIRAH macros<a class="headerlink" href="#sirah-macros" title="Link to this heading"></a></h4>
<p>In VMD, a macro is a text that represents a selection. Macros are a useful feature of VMD when you use certain selections often. In the SIRAH Tools plugin for VMD, <code class="docutils literal notranslate"><span class="pre">sirah_vmdtk.tcl</span></code>, 10 macros are available:</p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">sirah_membrane</span></code> - select only lipid residues;</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">sirah_nucleic</span></code> - select only acid nucleic residues;</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">sirah_protein</span></code> - select only protein residues;</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">sirah_basic</span></code> - select only basic amino acid residues;</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">sirah_acidic</span></code> - select only acidic amino acid residues;</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">sirah_polar</span></code> - select only polar amino acid residues;</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">sirah_neutral</span></code> - select only neutral amino acid residues;</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">sirah_backbone</span></code> - select the backbone beads of protein (GN, GC, and GO) or acid nucleic (PX, C5X, and O3’);</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">sirah_water</span></code> - select all solvent available in the force field (WT4 and WLS);</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">sirah_ions</span></code> - select all ions available in the force field (KW, NaW, ClW, MgX, CaX, and Znx).</p></li>
</ul>
<p>To use any SIRAH macro, go to <em>Graphics</em> &gt; <em>Representations</em> (See <strong>Figure 2</strong>) and click on the <em>Create Rep</em> button. In the <em>Selected Atoms</em> box, erase the text that appears there and type any of the available <em>Singlewords</em> above.</p>
<figure class="align-center" id="id4">
<a class="reference internal image-reference" href="_images/sirah_tools_2.png"><img alt="_images/sirah_tools_2.png" src="_images/sirah_tools_2.png" style="width: 100%;" /></a>
<figcaption>
<p><span class="caption-text"><strong>Figure 2.</strong> Our trajectory loaded in VMD with the default representation and without any selection.</span><a class="headerlink" href="#id4" title="Link to this image"></a></p>
</figcaption>
</figure>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>To see all available VMD’s <em>Singlewords</em>, click on the <em>Selections</em> tab in the <em>Graphics</em> &gt; <em>Representations</em> window and look through the <em>Singlewords</em> panel. You can also double-click on a singleword, and then, click on the <em>Apply</em> button to use a VMD Macro.</p>
</div>
<p>Since our system is a protein−DNA complex with coordinating magnesium ions, we can use four macros: <code class="docutils literal notranslate"><span class="pre">sirah_protein</span></code> (see <strong>Figure 3A</strong>), <code class="docutils literal notranslate"><span class="pre">sirah_nucleic</span></code> (see <strong>Figure 3B</strong>), <code class="docutils literal notranslate"><span class="pre">sirah_water</span></code> (see <strong>Figure 3C</strong>), and <code class="docutils literal notranslate"><span class="pre">sirah_ions</span></code> (see <strong>Figure 3D</strong>).</p>
<figure class="align-center" id="id5">
<a class="reference internal image-reference" href="_images/sirah_tools_3.png"><img alt="_images/sirah_tools_3.png" src="_images/sirah_tools_3.png" style="width: 100%;" /></a>
<figcaption>
<p><span class="caption-text"><strong>Figure 3.</strong> Each of the macros of our systems shown as a single image. A) The <code class="docutils literal notranslate"><span class="pre">sirah_protein</span></code> macro using the <strong>Licorice</strong> as <em>Drawing Method</em>. The backbone beads of the protein are pink, and the sidechains ones are blue when using the <em>Coloring Method</em> Name. B) The <code class="docutils literal notranslate"><span class="pre">sirah_nucleic</span></code> macro using the <em>Licorice</em> as <em>Drawing Method</em>. For the DNA, the backbone beads PX, C5X, and O3’ are colored dark yellow, cyan, and red, respectively, and the sidechain ones are colored red and blue using the <strong>Coloring Method</strong> Name. C) The <code class="docutils literal notranslate"><span class="pre">sirah_water</span></code> macro using the <em>CPK</em> as <em>Drawing Method</em>. In this system, only WT4 was used and is colored green by the <em>Coloring Method</em> Name. D) The <code class="docutils literal notranslate"><span class="pre">sirah_ions</span></code> macro using the <em>VDW</em> as <em>Drawing Method</em>. The ClW, NaW, and MgW ions are colored cyan, blue, and purple, respectively, when using the <em>Coloring Method</em> Name.</span><a class="headerlink" href="#id5" title="Link to this image"></a></p>
</figcaption>
</figure>
<p>To use all representations together, just create different representations for each one (clicking on the <em>Create Rep</em> button in the <em>Graphics</em> &gt; <em>Representations</em> window). Your protein-DNA complex should now look similar to the one in <strong>Figure 4</strong>.</p>
<figure class="align-center" id="id6">
<a class="reference internal image-reference" href="_images/sirah_tools_4.png"><img alt="_images/sirah_tools_4.png" src="_images/sirah_tools_4.png" style="width: 100%;" /></a>
<figcaption>
<p><span class="caption-text"><strong>Figure 4.</strong> Our protein-DNA complex after using the SIRAH macros <code class="docutils literal notranslate"><span class="pre">sirah_backbone</span></code>, <code class="docutils literal notranslate"><span class="pre">sirah_water</span></code>, and <code class="docutils literal notranslate"><span class="pre">sirah_ions</span></code>. To improve visualization, we used Licorice as <em>Drawing Method</em> for <code class="docutils literal notranslate"><span class="pre">sirah_backbone</span></code> and CPK for <code class="docutils literal notranslate"><span class="pre">sirah_water</span></code> and <code class="docutils literal notranslate"><span class="pre">sirah_ions</span></code>. In addition, the <strong>Material</strong> of the water beads was changed to Ghost and we hid some elements of the system.</span><a class="headerlink" href="#id6" title="Link to this image"></a></p>
</figcaption>
</figure>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>As you can see, macros can be very useful and when saving your work in a saved state (<em>File</em> &gt; <em>Save Visualization State</em>), macros are included in the saved state file to be used later (<em>File</em> &gt; <em>Load Visualization State</em>).</p>
</div>
</section>
</section>
<section id="structural-analysis-of-cg-systems">
<h3>Structural analysis of CG systems<a class="headerlink" href="#structural-analysis-of-cg-systems" title="Link to this heading"></a></h3>
<p>Besides the features that enhance the visualization of SIRAH CG simulations, two additional SIRAH Tools features can be used to analyze the trajectory: <code class="docutils literal notranslate"><span class="pre">sirah_ss</span></code> and <code class="docutils literal notranslate"><span class="pre">sirah_backmap</span></code>. They will be discussed below.</p>
<section id="secondary-structure-analysis">
<h4>Secondary structure analysis<a class="headerlink" href="#secondary-structure-analysis" title="Link to this heading"></a></h4>
<p>The utility <code class="docutils literal notranslate"><span class="pre">sirah_ss</span></code> assigns secondary structures to CG proteins in SIRAH, classifying residues into <em>α-helix (H)</em>, <em>extended β-sheet (E)</em> or, otherwise, <em>coil (C)</em> conformations, based on the instantaneous values of the backbone’s torsional angles and Hydrogen bond-like (HB) interactions (see <a class="reference external" href="https://pubs.acs.org/doi/10.1021/ct5007746">Darre et al.</a> for more information). The <code class="docutils literal notranslate"><span class="pre">sirah_ss</span></code> feature produces ASCII files of average and by-frame results, which can be visualized as color plots using the python script <code class="docutils literal notranslate"><span class="pre">plot_ss.py</span></code>.</p>
<p>With the <code class="docutils literal notranslate"><span class="pre">sirah_vmdtk.tcl</span></code> file loaded, we can access the <code class="docutils literal notranslate"><span class="pre">sirah_help</span></code> feature by going to <em>Extensions</em> &gt; <em>Tk Console</em> and entering</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="go">sirah_help</span>
</pre></div>
</div>
<p>to see all the SIRAH tools available options. Or, we can go straight to the help for the <code class="docutils literal notranslate"><span class="pre">sirah_ss</span></code> feature by typing</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="go">sirah_ss help</span>
</pre></div>
</div>
<p>The output will be the following:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="go">&gt;&gt;&gt;&gt; sirah_ss &lt;&lt;&lt;&lt;</span>

<span class="go">version: 1.1 [Dec 2015]</span>

<span class="go">Description:</span>

<span class="go">  Command to calculate the secondary structure of SIRAH proteins.</span>

<span class="go">  If mol is omitted then all options are applied to top molecule.</span>
<span class="go">  By default the secondary structure data is stored in memory</span>
<span class="go">  as the array sscache_data(mol,frame) and the coloring method</span>
<span class="go">  &#39;Secondary Structure&#39; will display correctly. By default the</span>
<span class="go">  script reads the sscache_data if available in memory and does</span>
<span class="go">  not recalculate the conformation, unless flag ramach is set or</span>
<span class="go">  the sscache_data is cleaned. The outname and noprint keywords</span>
<span class="go">  control the generated output files.</span>

<span class="go">  The current command version does not support PBC options.</span>

<span class="go">Usage: sirah_ss [options]</span>

<span class="go">Options: These are the optional arguments</span>

<span class="go">  mol       Set the molecule ID, default top</span>

<span class="go">  first     First frame to analyze</span>

<span class="go">  last      Last frame to analyze</span>

<span class="go">  sel       &quot;VMD selection&quot;, default &quot;all&quot;</span>

<span class="go">  outname   List {} of keywords and names for output files.</span>
<span class="go">            Keywords and default names:</span>

<span class="go">              byframe   ss_by_frame.xvg</span>
<span class="go">              byres     ss_by_res.xvg</span>
<span class="go">              global    ss_global.xvg</span>
<span class="go">              mtx       ss.mtx</span>
<span class="go">              phi       phi.mtx</span>
<span class="go">              psi       psi.mtx</span>

<span class="go">  load      Load secondary structure data from file (e.g ss.mtx) into molecule (mol)</span>

<span class="go">  noprint   Flag to avoid printing standard results to out files</span>

<span class="go">  ramach    Flag to write phi and psi angles of residue selection (sel) to out files</span>

<span class="go">  nocahe    Flag to avoid saving data to sscache_data</span>

<span class="go">  now       Flag to calculate the secondary structure at current frame.</span>
<span class="go">            No sscache_data or output file is saved.</span>

<span class="go">  clean     Flag to clean the sscache_data of the selected molecule (mol) and exit</span>

<span class="go">  help      Display help and exit</span>


<span class="go">Examples:</span>

<span class="go">  1.        Load secondary structure data into molecules 2 and extend the calculation</span>
<span class="go">            sirah_ss mol 2 load ss.mtx</span>
<span class="go">            sirah_ss mol 2</span>

<span class="go">  2.        Set custom output file names</span>
<span class="go">            sirah_ss outname {mtx myss.mtx byframe myss_by_frame.xvg}</span>
</pre></div>
</div>
<p>All the available options for <code class="docutils literal notranslate"><span class="pre">sirah_ss</span></code> are documented in this help. By default, four outputs are created by <code class="docutils literal notranslate"><span class="pre">sirah_ss</span></code>: <code class="docutils literal notranslate"><span class="pre">ss_by_frame.xvg</span></code>, <code class="docutils literal notranslate"><span class="pre">ss_by_res.xvg</span></code>, <code class="docutils literal notranslate"><span class="pre">ss_global.xvg</span></code> and <code class="docutils literal notranslate"><span class="pre">ss.mtx</span></code>. The <code class="docutils literal notranslate"><span class="pre">ss_by_frame.xvg</span></code> file gives the secondary structure percentages of the three classifications (H, E, or C) by each frame of the simulations. The <code class="docutils literal notranslate"><span class="pre">ss_by_res.xvg</span></code> file gives the secondary structure percentages of H, E, and C by each residue. The <code class="docutils literal notranslate"><span class="pre">ss_global.xvg</span></code> file shows the overall percentages and standard deviation of H, E, and C of the entire simulation. And the <code class="docutils literal notranslate"><span class="pre">ss.mtx</span></code> file gives a matrix of the secondary structure transitions, between H, E, or C, of the residues versus the simulation time.</p>
<p>We can also choose on which <em>mol</em> and on which frames to do the analysis within VMD. For example, if we want to calculate the secondary structure for the protein (loaded on <em>top</em>) on all frames, we can:</p>
<ol class="arabic">
<li><p>Make a VMD selection. By default <code class="docutils literal notranslate"><span class="pre">sirah_ss</span></code> selection is <em>all</em>, but if we need to change the selection to choose only the protein, we can type in the <em>Tk Console</em>:</p>
<blockquote>
<div><div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="go">set protein [atomselect top sirah_protein]</span>
</pre></div>
</div>
<p>Here we selected only the protein beads.</p>
</div></blockquote>
</li>
<li><p>Calculate the secondary structure information of the selection by typing:</p>
<blockquote>
<div><div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="go">sirah_ss</span>
</pre></div>
</div>
</div></blockquote>
</li>
</ol>
<p>This will, by default, calculate the secondary structure for all frames of the trajectory of our selection and generate the four files.</p>
<p>Furthermore, we have prepared a python script <code class="docutils literal notranslate"><span class="pre">plot_ss.py</span></code> that can be used to plot three of the four files (<code class="docutils literal notranslate"><span class="pre">ss_by_frame.xvg</span></code>, <code class="docutils literal notranslate"><span class="pre">ss_by_res.xvg</span></code> and <code class="docutils literal notranslate"><span class="pre">ss.mtx</span></code>).</p>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>The <code class="docutils literal notranslate"><span class="pre">plot_ss.py</span></code> script works properly with <strong>Python 3.9</strong>. You can for example create a conda environment using:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="go">conda create --name plot_ss python=3.9</span>
</pre></div>
</div>
<p>and activate it using:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="go">conda activate plot_ss</span>
</pre></div>
</div>
<p>Finally check de python version:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="go">python --version</span>
</pre></div>
</div>
</div>
<p>Once we know we are using the right version of Python, we can use the <code class="docutils literal notranslate"><span class="pre">plot_ss.py</span></code> script by typing:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">python</span> <span class="n">plot_ss</span><span class="o">.</span><span class="n">py</span> <span class="o">-</span><span class="n">h</span>
</pre></div>
</div>
<p>This will show us the options we have within the script, most of the flags are for changing the appearance of the graph (colors, font size, label size, etc).</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="go">Create a PNG image from an input file (ss.mtx, ss_by_frame.xvg, or ss_by_res.xvg) generated by SIRAH.</span>

<span class="go">optional arguments:</span>
<span class="go">  -h, --help            Show this help message and exit</span>
<span class="go">  -i [input]            Input file name (ss.mtx, ss_by_frame.xvg, ss_by_res.xvg)</span>
<span class="go">  -d [dpi]              DPI (dots per inch) for saving the figure (default: 300)</span>
<span class="go">  -tu [tu]              Time unit (us or ns) (default: us)</span>
<span class="go">  -dt [dt]              Time between consecutive frames (default: 1e-04)</span>
<span class="go">  -H [helix-color]      Alpha-helix color (default: darkviolet)</span>
<span class="go">  -E [beta-sheet color] Beta-sheet color (default: yellow)</span>
<span class="go">  -C [coil color]       Coil color (default: aqua)</span>
<span class="go">  -o [out name]         Image output name (default: input name)</span>
<span class="go">  -wt [width]           Image width (inches) (default: 10)</span>
<span class="go">  -ht [height]          Image height (inches) (default: 8)</span>
<span class="go">  -xfs [xlab fontsize]  Fontsize for x-axis labels (default: 14)</span>
<span class="go">  -yfs [ylab fontsize]  Fontsize for y-axis labels (default: 14)</span>
<span class="go">  -yticks [y # ticks]   Number of major tick locators on the y-axis (default: 10)</span>
<span class="go">  -xticks [x # ticks]   Number of major tick locators on the x-axis (default: 10)</span>
<span class="go">  -xtsize [xtsize]      Size of ticks on the x-axis (default: 12)</span>
<span class="go">  -ytsize [ytsize]      Size of ticks on the y-axis (default: 12)</span>
<span class="go">  -title [title]        Title of the plot (default: None)</span>
<span class="go">  -ttsize [title_size]  Size of the plot title (default: 16)</span>
<span class="go">  --version             Print version and exit</span>
</pre></div>
</div>
<p>The basic use of the script is:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="go">python plot_ss.py -i filename</span>
</pre></div>
</div>
<p>where <code class="docutils literal notranslate"><span class="pre">filename</span></code> is one of the files <code class="docutils literal notranslate"><span class="pre">ss_by_res.xvg</span></code>, <code class="docutils literal notranslate"><span class="pre">ss_by_frame.xvg</span></code> or <code class="docutils literal notranslate"><span class="pre">ss.mtx</span></code>.</p>
<p>In the case of the <code class="docutils literal notranslate"><span class="pre">ss.mtx</span></code> matrix it may take some time (no more than a couple of minutes), so be patient.</p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>To use the flags that modify the colors, any of the following entries is valid. For example, if we would like to use red as the color for the α-helix, we can type:</p>
<p><code class="docutils literal notranslate"><span class="pre">-H</span> <span class="pre">red</span></code>
<code class="docutils literal notranslate"><span class="pre">-H</span> <span class="pre">r``</span></code>
<code class="docutils literal notranslate"><span class="pre">-H</span> <span class="pre">&quot;red</span></code>
<code class="docutils literal notranslate"><span class="pre">-H</span> <span class="pre">&quot;r&quot;</span></code>
<code class="docutils literal notranslate"><span class="pre">-H</span> <span class="pre">&quot;#FF0000&quot;</span></code></p>
<p>In the case of HEX code usage, quotation marks are required.</p>
</div>
<p>The script generates the plots shown in <strong>Figure 5</strong>, remember that you must plot one by one, selecting the filename according to your interest.</p>
<figure class="align-center" id="id7">
<a class="reference internal image-reference" href="_images/sirah_tools_5.png"><img alt="_images/sirah_tools_5.png" src="_images/sirah_tools_5.png" style="width: 100%;" /></a>
<figcaption>
<p><span class="caption-text"><strong>Figure 5.</strong> Outcomes of <code class="docutils literal notranslate"><span class="pre">plot_ss.py</span></code> where the secondary structural elements α-helix (H), extended β-sheet (E), and coil (C) are colored purple, yellow, and blue, respectively. A) The <code class="docutils literal notranslate"><span class="pre">ss.mtx</span></code> matrix plot shows the protein’s secondary structure transitions during the 3.0 μs MD simulation. B)  The <code class="docutils literal notranslate"><span class="pre">ss_by_frame.xvg</span></code> plot shows the three secondary structure elements (H, E, and C) percentages by each frame of the 3.0 μs MD simulation. C) The <code class="docutils literal notranslate"><span class="pre">ss_by_res.xvg</span></code> plot shows the three secondary structure elements (H, E, and C) percentages by each residue during the 3.0 μs MD simulation. Here, we put the three plots together and edited the figure to use only one legend. However, the script makes three separate plots, each with its own legend.</span><a class="headerlink" href="#id7" title="Link to this image"></a></p>
</figcaption>
</figure>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The files <code class="docutils literal notranslate"><span class="pre">ss_by_res.xvg</span></code>, <code class="docutils literal notranslate"><span class="pre">ss_by_frame.xvg</span></code>, and <code class="docutils literal notranslate"><span class="pre">ss.mtx</span></code> can also be plotted using other plotting programs such as <a class="reference external" href="https://plasma-gate.weizmann.ac.il/Grace/">Grace</a> or <a class="reference external" href="https://www.r-project.org/">R</a>. In addition, the <code class="docutils literal notranslate"><span class="pre">plot_ss.py</span></code> script can be modified or improved according to the necessity of the user.</p>
</div>
</section>
<section id="backmapping-analysis">
<h4>Backmapping analysis<a class="headerlink" href="#backmapping-analysis" title="Link to this heading"></a></h4>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>By default, <code class="docutils literal notranslate"><span class="pre">sirah_backmap</span></code> minimizes the structures after the backmapping procedure. <a class="reference external" href="http://ambermd.org/AmberTools.php">AmberTools</a> is used to accomplish the minimization task. Please install this application if you do not already have it. If you are using AmberTools via conda, AmberTools environment should be activated before opening VMD.</p>
<p>However, the <code class="docutils literal notranslate"><span class="pre">nomin</span></code> option can be used to disable the minimization step. Consequently, you can minimize backmapped outputs by utilizing other software/force fields outside of VMD.</p>
</div>
<p>The utility <code class="docutils literal notranslate"><span class="pre">sirah_backmap</span></code> retrieves pseudo-atomistic information from the CG model. The atomistic positions are built on a by-residue basis following the geometrical reconstruction proposed by <a class="reference external" href="https://onlinelibrary.wiley.com/doi/10.1002/jcc.20237">Parsons et al.</a>. Bond distances and angles are derived from rough organic chemistry considerations stored in backmapping libraries.</p>
<p>With the <code class="docutils literal notranslate"><span class="pre">sirah_vmdtk.tcl</span></code> file loaded, we can access the <code class="docutils literal notranslate"><span class="pre">sirah_help</span></code> feature by going to <em>Extensions</em> &gt; <em>Tk Console</em> and entering</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="go">sirah_help</span>
</pre></div>
</div>
<p>to see all the SIRAH tools available options. Or, we can go straight to the help for the <code class="docutils literal notranslate"><span class="pre">sirah_backmap</span></code> feature by typing</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="go">sirah_backmap help</span>
</pre></div>
</div>
<p>The output will be the following:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="go">&gt;&gt;&gt;&gt; sirah_backmap &lt;&lt;&lt;&lt;</span>

<span class="go">version: 1.0 [Nov 2015]</span>

<span class="go">Description:</span>

<span class="go">  Command to recover atomistic information from coarse-grained or multiscale</span>
<span class="go">  systems.</span>

<span class="go">  Geometric operations are applied to reconstruct the atomic coordinates then</span>
<span class="go">  a minimization is performed to refine the structure of the system.</span>
<span class="go">  The minimization requires AMBERTOOLS 14 (free at http://ambermd.org/)</span>
<span class="go">  or later properly installed. MPI option requires mpirun and parallel</span>
<span class="go">  compilation of AMBER code. Be aware that small systems may fail to run or</span>
<span class="go">  converge in parallel execution due to decomposition problems. Notice, the</span>
<span class="go">  cuda version requires installing the AMBER licensed suite.</span>

<span class="go">  By default the force field ff14SB is used for the atomistic refinement,</span>
<span class="go">  any residue or unit not defined within it will generate an execution error.</span>
<span class="go">  The minimization protocol consists on 100+50 steps of steepest descent and</span>
<span class="go">  conjugate gradient in vacuum conditions with a 0.12 nm cut-off for</span>
<span class="go">  electrostatic.</span>

<span class="go">  The current command version does not support PBC options, so make sure the</span>
<span class="go">  molecules are whole before running the backmap.</span>

<span class="go">Usage: sirah_backmap [options]</span>

<span class="go">Options: These are the optional arguments</span>

<span class="go">  mol       Set the molecule ID, default top.</span>

<span class="go">  now       Backmap the current frame.</span>

<span class="go">  first     First frame to process.</span>

<span class="go">  last      Last frame to process.</span>

<span class="go">  each      Process frames each number of steps, default 1.</span>

<span class="go">  frames    List of frames to process, e.g. {1 2 10 21 22 23 30}.</span>

<span class="go">  outname   Root names for output PDB file.</span>

<span class="go">  noload    Flag to avoid loading the atomistic trajectory to the VMD session.</span>

<span class="go">  nomin     Flag to avoid minimizing the system.</span>

<span class="go">  mpi       MPI processes to use during minimization, default 1.</span>

<span class="go">  cuda      Flag to use pmemd.cuda, sets gbsa on, cutoff to 999 and no MPI.</span>

<span class="go">  gbsa      Flag to use implicit solvation GBSA (igb=1), default off (igb=0).</span>

<span class="go">  cutoff    Set cut-off value (in angstroms) for non-bonded interactions, default 12.</span>

<span class="go">  maxcyc    Set total number of minimization steps, default 150.</span>

<span class="go">  ncyc      Set the initial number of steepest descent steps, default 100.</span>

<span class="go">  help      Display help and exit.</span>
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Currently, backmapping libraries contain instructions for proteins, DNA, and ions; however, user-defined procedures for other models can be included.</p>
</div>
<p>All the available options for <code class="docutils literal notranslate"><span class="pre">sirah_backmap</span></code> are documented in this help. By default, all trajectory frames are used. Due to the fact that our protein-DNA complex is a 3.0 μs MD simulation with 30,000 frames, processing the entire trajectory could take some time. Thus, we chose to analyze the trajectory by extracting one frame for every 1,000 frames with the <code class="docutils literal notranslate"><span class="pre">each</span></code> option. To do that we type:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="go">sirah_backmap each 1000</span>
</pre></div>
</div>
<p>The output is a 300-frame file named <code class="docutils literal notranslate"><span class="pre">backmap.pdb</span></code>. This file is displayed as an animated GIF in <strong>Figure 6</strong>.</p>
<figure class="align-center" id="id8">
<a class="reference internal image-reference" href="_images/backmapping_bad.gif"><img alt="_images/backmapping_bad.gif" src="_images/backmapping_bad.gif" style="width: 100%;" /></a>
<figcaption>
<p><span class="caption-text"><strong>Figure 6.</strong> The 300-frame backmapped output from the 3.0 μs MD simulation using the default minimization arguments of <code class="docutils literal notranslate"><span class="pre">sirah_backmap</span></code>.</span><a class="headerlink" href="#id8" title="Link to this image"></a></p>
</figcaption>
</figure>
<p>The original atomistic PDB file reveals that a few nucleotides at the DNA molecule’s extremities are unpaired, allowing them to interact with neighboring molecules. Since we used this original PDB file to generate our CG representation, we anticipated that the unpaired nucleotides would exhibit a flexible behavior. However, in the backmapped PDB file, we can observe that the DNA extremities undergo unusual deformations and movements. These events are not observed in the CG simulation, as shown in <strong>Figure 7</strong>.</p>
<figure class="align-center" id="id9">
<a class="reference internal image-reference" href="_images/CG.gif"><img alt="_images/CG.gif" src="_images/CG.gif" style="width: 100%;" /></a>
<figcaption>
<p><span class="caption-text"><strong>Figure 7.</strong> The same 300-frame trajectory from the 3.0 μs MD simulation in CG representation does not show the unusual behavior at the DNA extremities.</span><a class="headerlink" href="#id9" title="Link to this image"></a></p>
</figcaption>
</figure>
<p>We believe that this unusual backmapping outcome could be the product of excessive minimization in this particular system. In most cases, the default parameters of 100 steps of steepest descent (<code class="docutils literal notranslate"><span class="pre">ncyc</span></code>) and 50 steps of conjugate gradient (total of 150 <code class="docutils literal notranslate"><span class="pre">maxcyc</span></code> steps) in vacuum conditions are sufficient to produce a suitable result, but in this instance, they produced artifacts that were absent from the CG simulation. To overcome this, we modified the total minimization steps to 50 for <code class="docutils literal notranslate"><span class="pre">maxcyc</span></code> and to 25 for <code class="docutils literal notranslate"><span class="pre">ncyc</span></code> as follows:</p>
<div class="admonition caution">
<p class="admonition-title">Caution</p>
<p>Return the original CG trajectory’s <strong>top</strong> status in VMD prior to typing the command.</p>
</div>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="go">sirah_backmap each 1000 maxcyc 50 ncyc 25</span>
</pre></div>
</div>
<p>The output is a new 300-frame file named <code class="docutils literal notranslate"><span class="pre">backmap.pdb</span></code>, displayed as an animated GIF in <strong>Figure 8</strong>, exhibiting a similar behavior to the CG representation simulation.</p>
<figure class="align-center" id="id10">
<a class="reference internal image-reference" href="_images/backmapping_ok.gif"><img alt="_images/backmapping_ok.gif" src="_images/backmapping_ok.gif" style="width: 100%;" /></a>
<figcaption>
<p><span class="caption-text"><strong>Figure 8.</strong> The final 300-frame backmapped output from the 3.0 μs MD simulation using less minimization steps by modifying the <code class="docutils literal notranslate"><span class="pre">maxcyc</span></code> and <code class="docutils literal notranslate"><span class="pre">ncyc</span></code> arguments of <code class="docutils literal notranslate"><span class="pre">sirah_backmap</span></code>.</span><a class="headerlink" href="#id10" title="Link to this image"></a></p>
</figcaption>
</figure>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>Always review both the original CG trajectory and the backmapping output to identify out-of-the-ordinary behavior and adjust arguments accordingly.</p>
</div>
<p>It is important to try different combinations of settings to find the one that works best for your system. To keep the backmapped files you already have, you can always change the name of the result with the <code class="docutils literal notranslate"><span class="pre">outname</span></code> option:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="go">sirah_backmap each 1000 maxcyc 50 ncyc 25 outname backmap_less_min.pdb</span>
</pre></div>
</div>
</section>
</section>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="Tutorials%20namd.html" class="btn btn-neutral float-left" title="NAMD" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2023, BA, SL, SM.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>