<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>AMBER &mdash; SIRAH Force Field v_1.0 documentation</title>
      <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
      <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script src="_static/jquery.js?v=5d32c60e"></script>
        <script src="_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
        <script src="_static/documentation_options.js?v=3060a628"></script>
        <script src="_static/doctools.js?v=888ff710"></script>
        <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
        <script async="async" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="prev" title="Developers" href="Developers.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="index.html" class="icon icon-home">
            SIRAH Force Field
          </a>
              <div class="version">
                v_1.0
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Manual</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="About%20SIRAH.html">About SIRAH</a></li>
<li class="toctree-l1"><a class="reference internal" href="Citation.html">Citations</a></li>
<li class="toctree-l1"><a class="reference internal" href="Further%20reading.html">Further Reading</a></li>
<li class="toctree-l1"><a class="reference internal" href="FAQ.html">FAQs</a></li>
<li class="toctree-l1"><a class="reference internal" href="Developers.html">Developers</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Tutorials</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="current reference internal" href="#">AMBER</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#download">Download</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#required-software">Required Software</a></li>
<li class="toctree-l3"><a class="reference internal" href="#prior-knowledge">Prior knowledge</a></li>
<li class="toctree-l3"><a class="reference internal" href="#set-up">Set up</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#dna-molecule-in-implicit-solvent">1. DNA molecule in implicit solvent</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#build-cg-representations">1.1. Build CG representations</a></li>
<li class="toctree-l3"><a class="reference internal" href="#prepare-leap-input">1.2. Prepare LEaP input</a></li>
<li class="toctree-l3"><a class="reference internal" href="#run-leap">1.3. Run LEaP</a></li>
<li class="toctree-l3"><a class="reference internal" href="#run-the-simulation">1.4. Run the simulation</a></li>
<li class="toctree-l3"><a class="reference internal" href="#visualizing-the-simulation">1.5. Visualizing the simulation</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#dna-molecule-in-explicit-solvent">2. DNA molecule in explicit solvent</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#id6">2.1. Build CG representations</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id7">2.2. Prepare LEaP input</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id8">2.3. Run LEaP</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id9">2.4. Run the simulation</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id10">2.5. Visualizing the simulation</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#multiscale-simulation-in-implicit-solvent">3. Multiscale simulation in implicit solvent</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#id12">3.1. Build CG representations</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id13">3.2. Prepare LEaP input</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id14">3.3. Run LEaP</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id15">3.4. Run the simulation</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id16">3.5. Visualizing the simulation</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#closed-circular-dna-in-implicit-solvent">4. Closed circular DNA in implicit solvent</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#id20">4.1. Build CG representations</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id21">4.2. Prepare LEaP input</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id22">4.3. Run LEaP</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id23">4.4. Run the simulation</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id24">4.5. Visualizing the simulation</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#proteins-in-explicit-solvent">5. Proteins in explicit solvent</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#id27">5.1. Build CG representations</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id29">5.2. Prepare LEaP input</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id31">5.3. Run LEaP</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id32">5.4. Run the simulation</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id33">5.5. Visualizing the simulation</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#lipid-bilayers-in-explicit-solvent">6. Lipid bilayers in explicit solvent</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#id35">6.1. Build CG representations</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id36">6.2. Prepare LEaP input</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id38">6.3. Run LEaP</a></li>
<li class="toctree-l3"><a class="reference internal" href="#resize-the-box-with-cpptraj">6.4. Resize the box with CPPTRAJ</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id39">6.5. Run the simulation</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id40">6.6. Visualizing the simulation</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#membrane-proteins-in-explicit-solvent">7. Membrane proteins in explicit solvent</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#id41">7.1. Build CG representations</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#embed-the-protein-in-a-lipid-bilayer">7.1.1. Embed the protein in a lipid bilayer</a></li>
<li class="toctree-l4"><a class="reference internal" href="#delete-close-contact-lipid-molecules">7.1.2. Delete close contact lipid molecules</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#id42">7.2. Prepare LEaP input</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id44">7.3. Run LEaP</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id45">7.4. Resize the box with CPPTRAJ</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id46">7.5. Run the simulation</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id47">7.6. Visualizing the simulation</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#appendix">Appendix</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#calculating-ionic-concentrations">Calculating ionic concentrations</a></li>
<li class="toctree-l3"><a class="reference internal" href="#available-mapping-files">Available mapping files</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="#gromacs">GROMACS</a></li>
<li class="toctree-l1"><a class="reference internal" href="#namd">NAMD</a></li>
<li class="toctree-l1"><a class="reference internal" href="#id51">SIRAH tools</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">SIRAH Force Field</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">AMBER</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/Tutorials.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="amber">
<h1>AMBER<a class="headerlink" href="#amber" title="Link to this heading"></a></h1>
<section id="download">
<span id="download-amber"></span><span id="id1"></span><h2>Download<a class="headerlink" href="#download" title="Link to this heading"></a></h2>
<section id="required-software">
<h3>Required Software<a class="headerlink" href="#required-software" title="Link to this heading"></a></h3>
<p><strong>AMBER</strong></p>
<p>AMBER (version 16) and AMBER Tools (version 16) or later versions properly installed and running in your computer.</p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p><a class="reference external" href="_ambertools">Download</a> lastest version of amber. See <em>How to obtain Amber23</em> section.
<a class="reference external" href="_amber_install">Install amber</a>. Select according to the operating system you are using
<a class="reference external" href="_amber">Install amberTools</a>.  If you have amber installed, it is not necessary to install ambertools separately.</p>
</div>
<p><strong>VMD</strong></p>
<p>The molecular visualization program <a class="reference external" href="https://www.ks.uiuc.edu/Research/vmd/">VMD</a>, version 1.9.3 or later is (<a class="reference external" href="https://www.ks.uiuc.edu/Development/Download/download.cgi?PackageName=VMD">freely available download</a>). Check the <a class="reference external" href="https://www.ks.uiuc.edu/Research/vmd/current/ig/node6.html">installation guide</a> for more instructions.</p>
</section>
<section id="prior-knowledge">
<h3>Prior knowledge<a class="headerlink" href="#prior-knowledge" title="Link to this heading"></a></h3>
<p>How to perform a standard atomistic molecular dynamics simulations with AMBER and basic usage of
VMD.</p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p><strong>AMBER</strong></p>
<p>If you are not familiar with atomistic molecular dynamics simulations we strongly recommend you to perform the basic usage tutorials of Amber and AmberTools: <a class="reference external" href="https://ambermd.org/tutorials/BuildingSystems.php">1.Building Systems</a>, <a class="reference external" href="https://ambermd.org/tutorials/Relaxation.php">3.Creating Stable Systems and Running MD</a> and <a class="reference external" href="https://ambermd.org/tutorials/Introductory.php">5.Case Studies</a>. For more details and documentation, check <a class="reference external" href="https://ambermd.org/doc12/Amber23.pdf">Amber23 Manual</a>.</p>
<p><strong>VMD</strong></p>
<p>If you are not familiar with VMD we strongly recommend you to perform the basic usage tutorial of VMD (<a class="reference external" href="https://www.ks.uiuc.edu/Training/Tutorials/vmd/tutorial-html/index.html">using VMD</a>). For more details and documentation, check <a class="reference external" href="https://www.ks.uiuc.edu/Research/vmd/current/ug/ug.html">VMD User Guide</a> and the <a class="reference external" href="https://www.ks.uiuc.edu/Research/vmd/current/docs.html#tutorials">official VMD documentation</a>.</p>
</div>
</section>
<section id="set-up">
<h3>Set up<a class="headerlink" href="#set-up" title="Link to this heading"></a></h3>
<p>Download <a class="reference external" href="sirah_amber">SIRAH for amber</a> and uncompress it into your working directory.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>tar<span class="w"> </span>-xzvf<span class="w"> </span>sirah_x2.2_20-08.amber.tgz
</pre></div>
</div>
<p>In case of using another version of sirah:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>tar<span class="w"> </span>-xzvf<span class="w"> </span>sirah_<span class="o">[</span>version<span class="o">]</span>.amber.tgz
</pre></div>
</div>
<p>You will get a folder <code class="docutils literal notranslate"><span class="pre">sirah_[version].amber/</span></code> containing the force field definition, the SIRAH Tools in
<code class="docutils literal notranslate"><span class="pre">sirah_[version].amber/tools/</span></code>, molecular structures to build up systems in <code class="docutils literal notranslate"><span class="pre">sirah_[version].amber/PDB/</span></code> and the required material to perform tutorials in <code class="docutils literal notranslate"><span class="pre">sirah_[version].amber/tutorial/X.X/</span></code></p>
<div class="admonition caution">
<p class="admonition-title">Caution</p>
<p>Remember to change <strong>X.X</strong> to the number that corresponds to the tutorial you are going to do.</p>
</div>
<p>Make a new folder for this tutorial in your working directory:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>mkdir<span class="w"> </span>tutorialX.X<span class="p">;</span><span class="w"> </span><span class="nb">cd</span><span class="w"> </span>tutorialX.X
</pre></div>
</div>
<p>Create the following symbolic link in the folder tutorialX.X:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>ln<span class="w"> </span>-s<span class="w"> </span>../sirah_x2.2_20-08.amber<span class="w"> </span>sirah.amber
</pre></div>
</div>
</section>
</section>
<section id="dna-molecule-in-implicit-solvent">
<span id="tutorial-1"></span><h2>1. DNA molecule in implicit solvent<a class="headerlink" href="#dna-molecule-in-implicit-solvent" title="Link to this heading"></a></h2>
<p>This tutorial shows how to perform a coarse grained (CG) simulation of a double stranded DNA using the Generalized Born model for implicit solvent (GB) and the SIRAH force field. The main references
for this tutorial are: <a class="reference external" href="https://pubs.acs.org/doi/abs/10.1021/ct900653p">SIRAH DNA</a> (latest parameters are those reported in: <a class="reference external" href="https://pubs.acs.org/doi/abs/10.1021/ct100379f">WAT4</a>, <a class="reference external" href="https://academic.oup.com/bioinformatics/article/32/10/1568/1743152">SIRAH Tools</a>. We strongly advise you to read these articles before starting the tutorial.</p>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>Check <a class="reference internal" href="#download-amber"><span class="std std-ref">download</span></a> section for download and set up details before to start this tutorial.
Since this is <strong>tutorial 1</strong>, remember to replace <code class="docutils literal notranslate"><span class="pre">X.X</span></code>, the files corresponding to this tutorial can be found in: <code class="docutils literal notranslate"><span class="pre">sirah_[version].amber/tutorial/1/</span></code></p>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>If you are not familiar with DNA stuff we strongly recommend you to first perform the <a class="reference external" href="http://ambermd.org/tutorials/basic/tutorial1">AMBER
tutorial on DNA</a>.</p>
</div>
<section id="build-cg-representations">
<h3>1.1. Build CG representations<a class="headerlink" href="#build-cg-representations" title="Link to this heading"></a></h3>
<p>Map the atomistic structure of a 20-mer DNA to its CG representation:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>./sirah.amber/tools/CGCONV/cgconv.pl<span class="w"> </span>-i<span class="w"> </span>./sirah.amber/tutorial/1/dna.pdb<span class="w">  </span>-o<span class="w"> </span>dna_cg.pdb
</pre></div>
</div>
<p>The input file <code class="docutils literal notranslate"><span class="pre">-i</span></code> dna.pdb contains all the heavy atoms composing the DNA molecule, while the  output <code class="docutils literal notranslate"><span class="pre">-o</span></code> dna_cg.pdb preserves a few of them.
Please check both PDB structures using VMD:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>vmd<span class="w"> </span>-m<span class="w"> </span>./sirah.amber/tutorial/1/dna.pdb<span class="w"> </span>dna_cg.pdb
</pre></div>
</div>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>This is the basic usage of the script <strong>cgconv.pl</strong>, you can learn other capabilities from its help:
<code class="docutils literal notranslate"><span class="pre">./sirah.amber/tools/CGCONV/cgconv.pl</span> <span class="pre">-h</span></code></p>
</div>
<p>From now on it is just normal AMBER stuff!</p>
</section>
<section id="prepare-leap-input">
<h3>1.2. Prepare LEaP input<a class="headerlink" href="#prepare-leap-input" title="Link to this heading"></a></h3>
<p>Use a text editor to create the file <code class="docutils literal notranslate"><span class="pre">gensystem.leap</span></code> including the following lines:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="gp"># </span>Load<span class="w"> </span>SIRAH<span class="w"> </span>force<span class="w"> </span>field
<span class="go">addPath ./sirah.amber</span>
<span class="go">source leaprc.sirah</span>

<span class="gp"># </span>Load<span class="w"> </span>model
<span class="go">dna = loadpdb dna_cg.pdb</span>

<span class="gp"># </span>Save<span class="w"> </span>Parms
<span class="go">saveAmberParmNetcdf dna dna_cg.prmtop dna_cg.ncrst</span>

<span class="gp"># </span>EXIT
<span class="go">quit</span>
</pre></div>
</div>
</section>
<section id="run-leap">
<h3>1.3. Run LEaP<a class="headerlink" href="#run-leap" title="Link to this heading"></a></h3>
<p>Run the LEAP application to generate the molecular topology and initial coordinate files:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>tleap<span class="w"> </span>-f<span class="w"> </span>gensystem.leap
</pre></div>
</div>
<div class="admonition caution">
<p class="admonition-title">Caution</p>
<p>Warning messages about long, triangular or square bonds in <code class="docutils literal notranslate"><span class="pre">leap.log</span></code> file are fine and
expected due to the CG topology.</p>
</div>
<p>This should create a topology file dna_cg.prmtop and a coordinate file dna_cg.ncrst.</p>
<p>Use VMD to check how the CG model looks like:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>vmd<span class="w"> </span>dna_cg.prmtop<span class="w"> </span>dna_cg.ncrst<span class="w"> </span>-e<span class="w"> </span>./sirah.amber/tools/sirah_vmdtk.tcl
</pre></div>
</div>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>VMD assigns default radius to unknown atom types, the script <code class="docutils literal notranslate"><span class="pre">sirah_vmdtk.tcl</span></code> sets the right
ones. It also provides a kit of useful selection macros, coloring methods and backmapping utilities.
Use the command <code class="docutils literal notranslate"><span class="pre">sirah_help</span></code> in the Tcl/Tk console of VMD to access the manual pages.</p>
</div>
</section>
<section id="run-the-simulation">
<h3>1.4. Run the simulation<a class="headerlink" href="#run-the-simulation" title="Link to this heading"></a></h3>
<p>Make a new folder for the run:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>mkdir<span class="w"> </span>-p<span class="w"> </span>run<span class="p">;</span><span class="w"> </span><span class="nb">cd</span><span class="w"> </span>run
</pre></div>
</div>
<p>The folder <code class="docutils literal notranslate"><span class="pre">sirah.amber/tutorial/1/PMEMD.GPU/</span></code> contains typical input files for energy minimization
(<code class="docutils literal notranslate"><span class="pre">em_GB.in</span></code>), equilibration (<code class="docutils literal notranslate"><span class="pre">eq_GB.in</span></code>) and production (<code class="docutils literal notranslate"><span class="pre">md_GB.in</span></code>) runs. Please check carefully the input
flags therein.</p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<blockquote>
<div><p><strong>Some flags used in AMBER</strong></p>
</div></blockquote>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">-i</span></code>: Input file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-o</span></code>: Output file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-p</span></code>: Parameter/topology file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-c</span></code>: Coordinate file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-r</span></code>: Restart file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-x</span></code>: Trajectory file.</p></li>
</ul>
</div>
<div class="admonition caution">
<p class="admonition-title">Caution</p>
<p>These input files are executed by the <strong>GPU</strong> implementation of <code class="docutils literal notranslate"><span class="pre">pmemd.cuda</span></code>. Other available implementations that could be used: <code class="docutils literal notranslate"><span class="pre">sander</span></code>  or <code class="docutils literal notranslate"><span class="pre">pmemd</span></code>, both <strong>CPU</strong> implementations of AMBER.</p>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>You can find example input files for CPU versions of sander and pmemd at folders <code class="docutils literal notranslate"><span class="pre">SANDER/</span></code> and  <code class="docutils literal notranslate"><span class="pre">PMEMD.CPU/</span></code>, within <code class="docutils literal notranslate"><span class="pre">sirah.amber/tutorial/1/</span></code></p>
</div>
<p><strong>Energy Minimization:</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/1/PMEMD.GPU/em_GB.in<span class="w"> </span>-p<span class="w"> </span>../dna_cg.prmtop<span class="w"> </span>-c<span class="w"> </span>../dna_cg.ncrst<span class="w"> </span>-o<span class="w"> </span>dna_cg_em.out<span class="w"> </span>-r<span class="w"> </span>dna_cg_em.ncrst<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>In the course of long MD simulations the capping residues may eventually separate, this effect is
called helix fraying. To avoid such behavior is necessary to set Watson-Crick restraints for the capping base pairs of this CG DNA at the end of <code class="docutils literal notranslate"><span class="pre">eq_GB.in</span></code> and <code class="docutils literal notranslate"><span class="pre">md_GB.in</span></code> files. Check the files lines that start with <em>&amp;rst</em>.</p>
</div>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>If you are using SANDER to avoid such behavior create a symbolic link to the file <code class="docutils literal notranslate"><span class="pre">dna_cg.RST</span></code>, which
contains the definition of Watson-Crick restraints for the capping base pairs of this CG DNA:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>ln<span class="w"> </span>-s<span class="w"> </span>../sirah.amber/tutorial/1/SANDER/dna_cg.RST
</pre></div>
</div>
<p>The file dna_cg.RST can only be read by SANDER, PMEMD reads a different restrain format.</p>
</div>
<p><strong>Equilibration:</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/1/PMEMD.GPU/eq_GB.in<span class="w"> </span>-p<span class="w"> </span>../dna_cg.prmtop<span class="w"> </span>-c<span class="w"> </span>dna_cg_em.ncrst<span class="w"> </span>-o<span class="w"> </span>dna_cg_eq.out<span class="w"> </span>-r<span class="w"> </span>dna_cg_eq.ncrst<span class="w"> </span>-x<span class="w"> </span>dna_cg_eq.nc<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
<p><strong>Production (100ns):</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/1/PMEMD.GPU/md_GB.in<span class="w"> </span>-p<span class="w"> </span>../dna_cg.prmtop<span class="w"> </span>-c<span class="w"> </span>dna_cg_eq.ncrst<span class="w"> </span>-o<span class="w"> </span>dna_cg_md.out<span class="w"> </span>-r<span class="w"> </span>dna_cg_md.ncrst<span class="w">  </span>-x<span class="w"> </span>dna_cg_md.nc<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
</section>
<section id="visualizing-the-simulation">
<h3>1.5. Visualizing the simulation<a class="headerlink" href="#visualizing-the-simulation" title="Link to this heading"></a></h3>
<p>Now you can load, visualize and analize the trajectory file in VMD:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">vmd</span> <span class="o">../</span><span class="n">dna_cg</span><span class="o">.</span><span class="n">prmtop</span> <span class="o">../</span><span class="n">dna_cg</span><span class="o">.</span><span class="n">ncrst</span> <span class="n">dna_cg_md</span><span class="o">.</span><span class="n">nc</span> <span class="o">-</span><span class="n">e</span> <span class="o">../</span><span class="n">sirah</span><span class="o">.</span><span class="n">amber</span><span class="o">/</span><span class="n">tools</span><span class="o">/</span><span class="n">sirah_vmdtk</span><span class="o">.</span><span class="n">tcl</span>
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The file <code class="docutils literal notranslate"><span class="pre">sirah_vmdtk.tcl</span></code> is a Tcl script that is part of SIRAH Tools and contains the macros to properly visualize the coarse-grained structures in VMD. Use the command <code class="docutils literal notranslate"><span class="pre">sirah-help</span></code> in the Tcl/Tk console of VMD to access the manual pages.</p>
</div>
</section>
</section>
<section id="dna-molecule-in-explicit-solvent">
<span id="tutorial-2"></span><h2>2. DNA molecule in explicit solvent<a class="headerlink" href="#dna-molecule-in-explicit-solvent" title="Link to this heading"></a></h2>
<p>This tutorial shows how to use the SIRAH force field to perform a coarse grained (CG) simulation of a
double stranded DNA embedded in explicit solvent (called WatFour, WT4). The main references for
this tutorial are: <a class="reference external" href="https://pubs.acs.org/doi/abs/10.1021/ct900653p">SIRAH DNA</a> (latest parameters are those reported in: <a class="reference external" href="https://pubs.acs.org/doi/abs/10.1021/ct100379f">WAT4</a>, <a class="reference external" href="https://academic.oup.com/bioinformatics/article/32/10/1568/1743152">SIRAH Tools</a>.
We strongly advise you to read these articles before starting the tutorial.</p>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>Check <a class="reference internal" href="#download-amber"><span class="std std-ref">download</span></a> section for download and set up details before to start this tutorial.
Since this is <strong>tutorial 2</strong>, remember to replace <code class="docutils literal notranslate"><span class="pre">X.X</span></code>, the files corresponding to this tutorial can be found in: <code class="docutils literal notranslate"><span class="pre">sirah_[version].amber/tutorial/2/</span></code></p>
</div>
<section id="id6">
<h3>2.1. Build CG representations<a class="headerlink" href="#id6" title="Link to this heading"></a></h3>
<p>Map the atomistic structure of a 20-mer DNA to its CG representation:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>./sirah.amber/tools/CGCONV/cgconv.pl<span class="w"> </span>-i<span class="w"> </span>./sirah.amber/tutorial/2/dna.pdb<span class="w"> </span>-o<span class="w"> </span>dna_cg.pdb
</pre></div>
</div>
<p>The input file <code class="docutils literal notranslate"><span class="pre">-i</span></code> dna.pdb contains all the heavy atoms composing the DNA molecule, while the  output <code class="docutils literal notranslate"><span class="pre">-o</span></code> dna_cg.pdb preserves a few of them.
Please check both PDB structures using VMD:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>vmd<span class="w"> </span>-m<span class="w"> </span>./sirah.amber/tutorial/2/dna.pdb<span class="w"> </span>dna_cg.pdb
</pre></div>
</div>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>This is the basic usage of the script <strong>cgconv.pl</strong>, you can learn other capabilities from its help:
<code class="docutils literal notranslate"><span class="pre">./sirah.amber/tools/CGCONV/cgconv.pl</span> <span class="pre">-h</span></code></p>
</div>
<p>From now on it is just normal AMBER stuff!</p>
</section>
<section id="id7">
<h3>2.2. Prepare LEaP input<a class="headerlink" href="#id7" title="Link to this heading"></a></h3>
<p>Use a text editor to create the file <code class="docutils literal notranslate"><span class="pre">gensystem.leap</span></code> including the following lines:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="gp"># </span>Load<span class="w"> </span>SIRAH<span class="w"> </span>force<span class="w"> </span>field
<span class="go">addPath ./sirah.amber</span>
<span class="go">source leaprc.sirah</span>

<span class="gp"># </span>Load<span class="w"> </span>model
<span class="go">dna = loadpdb dna_cg.pdb</span>

<span class="gp"># </span>Info<span class="w"> </span>on<span class="w"> </span>system<span class="w"> </span>charge
<span class="go">charge dna</span>

<span class="gp"># </span>Add<span class="w"> </span>solvent,<span class="w"> </span>counterions<span class="w"> </span>and<span class="w"> </span><span class="m">0</span>.15M<span class="w"> </span>NaCl
<span class="gp"># </span>Tuned<span class="w"> </span>solute-solvent<span class="w"> </span>closeness<span class="w"> </span><span class="k">for</span><span class="w"> </span>best<span class="w"> </span>hydration
<span class="go">solvateoct dna WT4BOX 20 0.7</span>
<span class="go">addionsrand dna NaW 88 ClW 50</span>

<span class="gp"># </span>Save<span class="w"> </span>Parms
<span class="go">saveAmberParmNetcdf dna dna_cg.prmtop dna_cg.ncrst</span>

<span class="gp"># </span>EXIT
<span class="go">quit</span>
</pre></div>
</div>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<p>The available ionic species in SIRAH force field are: <code class="docutils literal notranslate"><span class="pre">Na⁺</span></code> (NaW), <code class="docutils literal notranslate"><span class="pre">K⁺</span></code> (KW) and <code class="docutils literal notranslate"><span class="pre">Cl⁻</span></code> (ClW). One
ion pair (e.g. NaW-ClW) each 34 WT4 molecules renders a salt concentration of ~0.15M (see <a class="reference internal" href="#appendix"><span class="std std-ref">Appendix</span></a> for details).
Counterions were added according to <a class="reference external" href="https://pubs.acs.org/doi/10.1021/acs.jctc.9b00953">Machado et al.</a>.</p>
</div>
</section>
<section id="id8">
<h3>2.3. Run LEaP<a class="headerlink" href="#id8" title="Link to this heading"></a></h3>
<p>Run the LEaP application to generate the molecular topology and initial coordinate files:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>tleap<span class="w"> </span>-f<span class="w"> </span>gensystem.leap
</pre></div>
</div>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>Warning messages about long, triangular or square bonds in <code class="docutils literal notranslate"><span class="pre">leap.log</span></code> file are fine and
expected due to the CG topology of some residues.</p>
</div>
<p>This should create a topology file <code class="docutils literal notranslate"><span class="pre">dna_cg.prmtop</span></code> and a coordinate file <code class="docutils literal notranslate"><span class="pre">dna_cg.ncrst</span></code>.</p>
<p>Use VMD to check how the CG model looks like:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>vmd<span class="w"> </span>dna_cg.prmtop<span class="w"> </span>dna_cg.ncrst<span class="w"> </span>-e<span class="w"> </span>./sirah.amber/tools/sirah_vmdtk.tcl
</pre></div>
</div>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>VMD assigns default radius to unknown atom types, the script <code class="docutils literal notranslate"><span class="pre">sirah_vmdtk.tcl</span></code> sets the right
ones. It also provides a kit of useful selection macros, coloring methods and backmapping utilities.
Use the command <code class="docutils literal notranslate"><span class="pre">sirah_help</span></code> in the Tcl/Tk console of VMD to access the manual pages.</p>
</div>
</section>
<section id="id9">
<h3>2.4. Run the simulation<a class="headerlink" href="#id9" title="Link to this heading"></a></h3>
<p>Make a new folder for the run:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>mkdir<span class="w"> </span>-p<span class="w"> </span>run<span class="p">;</span><span class="w"> </span><span class="nb">cd</span><span class="w"> </span>run
</pre></div>
</div>
<p>The folder <code class="docutils literal notranslate"><span class="pre">sirah.amber/tutorial/2/PMEMD/</span></code> contains typical input files for energy minimization
(<code class="docutils literal notranslate"><span class="pre">em_WT4.in</span></code>), equilibration (<code class="docutils literal notranslate"><span class="pre">eq_WT4.in</span></code>) and production (<code class="docutils literal notranslate"><span class="pre">md_WT4.in</span></code>) runs. Please check carefully the input flags therein, in particular the definition of flag <em>chngmask=0</em> at <em>&amp;ewald</em> section is <strong>mandatory</strong>.</p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<blockquote>
<div><p><strong>Some flags used in AMBER</strong></p>
</div></blockquote>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">-i</span></code>: Input file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-o</span></code>: Output file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-p</span></code>: Parameter/topology file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-c</span></code>: Coordinate file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-r</span></code>: Restart file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-x</span></code>: Trajectory file.</p></li>
</ul>
</div>
<div class="admonition caution">
<p class="admonition-title">Caution</p>
<p>These input files are executed by the <strong>GPU</strong> implementation of <code class="docutils literal notranslate"><span class="pre">pmemd.cuda</span></code>. Other available implementations that could be used: <code class="docutils literal notranslate"><span class="pre">sander</span></code>  or <code class="docutils literal notranslate"><span class="pre">pmemd</span></code>, both <strong>CPU</strong> implementations of AMBER.</p>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>You can find example input files for sander at <code class="docutils literal notranslate"><span class="pre">SANDER/</span></code> and pmemd at <code class="docutils literal notranslate"><span class="pre">PMEMD/</span></code>, within <code class="docutils literal notranslate"><span class="pre">sirah.amber/tutorial/2/</span></code></p>
</div>
<p><strong>Energy Minimization:</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/2/PMEMD/em_WT4.in<span class="w"> </span>-p<span class="w"> </span>../dna_cg.prmtop<span class="w"> </span>-c<span class="w"> </span>../dna_cg.ncrst<span class="w"> </span>-o<span class="w"> </span>dna_cg_em.out<span class="w"> </span>-r<span class="w"> </span>dna_cg_em.ncrst<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>In the course of long MD simulations the capping residues may eventually separate, this effect is
called helix fraying. To avoid such behavior is necessary to set Watson-Crick restraints for the capping base pairs of this CG DNA at the end of <code class="docutils literal notranslate"><span class="pre">eq_GB.in</span></code> and <code class="docutils literal notranslate"><span class="pre">md_GB.in</span></code> files. Check the files lines that start with <em>&amp;rst</em>.</p>
</div>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>If you are using SANDER to avoid such behavior create a symbolic link to the file <code class="docutils literal notranslate"><span class="pre">dna_cg.RST</span></code>, which
contains the definition of Watson-Crick restraints for the capping base pairs of this CG DNA:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>ln<span class="w"> </span>-s<span class="w"> </span>../sirah.amber/tutorial/1/SANDER/dna_cg.RST
</pre></div>
</div>
<p>The file dna_cg.RST can only be read by SANDER, PMEMD reads a different restrain format.</p>
</div>
<p><strong>Equilibration (NPT):</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/2/PMEMD/eq_WT4.in<span class="w"> </span>-p<span class="w"> </span>../dna_cg.prmtop<span class="w"> </span>-c<span class="w"> </span>dna_cg_em.ncrst<span class="w"> </span>-ref<span class="w"> </span>dna_cg_em.ncrst<span class="w"> </span>-o<span class="w"> </span>dna_cg_eq.out<span class="w"> </span>-r<span class="w"> </span>dna_cg_eq.ncrst<span class="w"> </span>-x<span class="w"> </span>dna_cg_eq.nc<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
<p><strong>Production (100ns):</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/2/PMEMD/md_WT4.in<span class="w"> </span>-p<span class="w"> </span>../dna_cg.prmtop<span class="w"> </span>-c<span class="w"> </span>dna_cg_eq.ncrst<span class="w"> </span>-o<span class="w"> </span>dna_cg_md.out<span class="w"> </span>-r<span class="w"> </span>dna_cg_md.ncrst<span class="w"> </span>-x<span class="w"> </span>dna_cg_md.nc<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
</section>
<section id="id10">
<h3>2.5. Visualizing the simulation<a class="headerlink" href="#id10" title="Link to this heading"></a></h3>
<p>That’s it! Now you can analyze the trajectory.
Process the output trajectory to account for the Periodic Boundary Conditions (PBC):</p>
<blockquote>
<div><div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;autoimage\ngo\nquit\n&quot;</span><span class="w"> </span><span class="p">|</span><span class="w"> </span>cpptraj<span class="w"> </span>-p<span class="w"> </span>../dna_cg.prmtop<span class="w"> </span>-y<span class="w"> </span>dna_cg_md.nc<span class="w"> </span>-x<span class="w"> </span>dna_cg_md_pbc.nc<span class="w"> </span>--interactive
</pre></div>
</div>
</div></blockquote>
<p><strong>Load the processed trajectory in VMD:</strong></p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">vmd</span> <span class="o">../</span><span class="n">dna_cg</span><span class="o">.</span><span class="n">prmtop</span> <span class="o">../</span><span class="n">dna_cg</span><span class="o">.</span><span class="n">ncrst</span> <span class="n">dna_cg_md</span><span class="o">.</span><span class="n">nc</span> <span class="o">-</span><span class="n">e</span> <span class="o">../</span><span class="n">sirah</span><span class="o">.</span><span class="n">amber</span><span class="o">/</span><span class="n">tools</span><span class="o">/</span><span class="n">sirah_vmdtk</span><span class="o">.</span><span class="n">tcl</span>
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The file <code class="docutils literal notranslate"><span class="pre">sirah_vmdtk.tcl</span></code> is a Tcl script that is part of SIRAH Tools and contains the macros to properly visualize the coarse-grained structures in VMD. Use the command <code class="docutils literal notranslate"><span class="pre">sirah-help</span></code> in the Tcl/Tk console of VMD to access the manual pages.</p>
</div>
</section>
</section>
<section id="multiscale-simulation-in-implicit-solvent">
<span id="tutorial-3"></span><h2>3. Multiscale simulation in implicit solvent<a class="headerlink" href="#multiscale-simulation-in-implicit-solvent" title="Link to this heading"></a></h2>
<p>This tutorial shows how to apply a multiscale representation of DNA, which is available in the SIRAH
force field, to study protein-DNA interactions. In this approach, the protein and its binding region are
treated with an atomistic force field, while the contextual DNA is represented at coarse-grained (CG)
level. The simulated system is composed of the human TATA binding protein (TBP PDB: <a class="reference external" href="https://www.rcsb.org/structure/1c9b">1C9B</a>) bounded to the TATA box at the promoter region (-64 to +13) of the Human Immunodeficiency Virus type 1 (HIV-1, GenBank: <a class="reference external" href="https://www.ncbi.nlm.nih.gov/nuccore/K03455.1">K03455</a>, <strong>Figure 1</strong>). Solvent effects are considered implicitly using the Generalized Born model (GB).</p>
<p>The main references for this tutorial are: <a class="reference external" href="https://pubs.rsc.org/en/Content/ArticleLanding/2011/CP/c1cp21248f">All-atoms/CG DNA</a> and <a class="reference external" href="https://academic.oup.com/bioinformatics/article/32/10/1568/1743152">SIRAH Tools</a>. We strongly advise you to read these articles before starting the tutorial.</p>
<figure class="align-center" id="id54">
<a class="reference internal image-reference" href="_images/Tuto3.png"><img alt="map to buried treasure" src="_images/Tuto3.png" style="width: 100%;" /></a>
<figcaption>
<p><span class="caption-text"><strong>Figure 1.</strong> Promoter region of HIV-1. The TATA box is highlighted in orange, while base pairs at 0.7nm of TBP are colored in yellow.</span><a class="headerlink" href="#id54" title="Link to this image"></a></p>
</figcaption>
</figure>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>Check <a class="reference internal" href="#download-amber"><span class="std std-ref">download</span></a> section for download and set up details before to start this tutorial.
Since this is <strong>tutorial 3</strong>, remember to replace <code class="docutils literal notranslate"><span class="pre">X.X</span></code>, the files corresponding to this tutorial can be found in: <code class="docutils literal notranslate"><span class="pre">sirah_[version].amber/tutorial/3/</span></code></p>
</div>
<section id="id12">
<h3>3.1. Build CG representations<a class="headerlink" href="#id12" title="Link to this heading"></a></h3>
<p>Map only the atomistic structure of nucleotides outside the TATA box of the HIV-1 promoter to CG:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>./sirah.amber/tools/CGCONV/cgconv.pl<span class="w"> </span>-i<span class="w"> </span>./sirah.amber/tutorial/3/tbphiv.pdb<span class="w"> </span>-R<span class="w"> </span><span class="m">1</span>-32,45-110,123-154<span class="w"> </span>-o<span class="w"> </span>dna_hyb.pdb
</pre></div>
</div>
<p>The input file <code class="docutils literal notranslate"><span class="pre">-i</span></code> tbphiv.pdb contains all the heavy atoms composing the protein-DNA system. Mapped
residues are selected through option <code class="docutils literal notranslate"><span class="pre">-R</span></code>. The selection considers a buffer of two base pairs at each
side of the TATA box (<strong>Figure 1</strong>). The atomistic/CG interface must always be a step of B form DNA. The
resulting coordinates are saved in the output <code class="docutils literal notranslate"><span class="pre">-o</span></code> dna_hyb.pdb.</p>
<p>Please check both PDB structures in VMD:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>vmd<span class="w"> </span>-m<span class="w"> </span>./sirah.amber/tutorial/3/tbphiv.pdb<span class="w"> </span>dna_hyb.pdb
</pre></div>
</div>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>This is the basic usage of the script <strong>cgconv.pl</strong>, you can learn other capabilities from its help:
<code class="docutils literal notranslate"><span class="pre">./sirah.amber/tools/CGCONV/cgconv.pl</span> <span class="pre">-h</span></code></p>
</div>
<p>From now on it is just normal AMBER stuff!</p>
</section>
<section id="id13">
<h3>3.2. Prepare LEaP input<a class="headerlink" href="#id13" title="Link to this heading"></a></h3>
<p>Use a text editor to create the file <code class="docutils literal notranslate"><span class="pre">gensystem.leap</span></code> including the following lines:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="gp"># </span>Load<span class="w"> </span>AMBER<span class="w"> </span>force<span class="w"> </span>field<span class="w"> </span><span class="o">(</span>parm14SB/parmbsc0<span class="o">)</span>
<span class="go">source oldff/leaprc.ff14SB</span>

<span class="gp"># </span>Load<span class="w"> </span>SIRAH<span class="w"> </span>force<span class="w"> </span>field
<span class="go">addPath ./sirah.amber</span>
<span class="go">source leaprc.sirah</span>

<span class="gp"># </span>Load<span class="w"> </span>model
<span class="go">dna = loadpdb dna_hyb.pdb</span>

<span class="gp"># </span>Save<span class="w"> </span>Parms
<span class="go">saveAmberParmNetcdf dna dna_hyb.prmtop dna_hyb.ncrst</span>

<span class="gp"># </span>EXIT
<span class="go">quit</span>
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Notice: According to AMBER version 10, 11 or 12 (14) the source file for parm99SB/bsc0 is leaprc.ff99bsc0, leaprc.ff10 or leaprc.ff12SB respectively.</p>
</div>
</section>
<section id="id14">
<h3>3.3. Run LEaP<a class="headerlink" href="#id14" title="Link to this heading"></a></h3>
<p>Run the LEaP application to generate the molecular topology and initial coordinate files:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>tleap<span class="w"> </span>-f<span class="w"> </span>gensystem.leap
</pre></div>
</div>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>Warning messages about long, triangular or square bonds in <code class="docutils literal notranslate"><span class="pre">leap.log</span></code> file are fine and
expected due to the CG topology of some residues.</p>
</div>
<p>This should create a topology file <code class="docutils literal notranslate"><span class="pre">dna_hyb.prmtop</span></code> and a coordinate file <code class="docutils literal notranslate"><span class="pre">dna_hyb.ncrst</span></code>.</p>
<p>Use VMD to check how the multiscale model looks like:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>vmd<span class="w"> </span>dna_hyb.prmtop<span class="w"> </span>dna_hyb.ncrst<span class="w"> </span>-e<span class="w"> </span>./sirah.amber/tools/sirah_vmdtk.tcl
</pre></div>
</div>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>VMD assigns default radius to unknown atom types, the script <code class="docutils literal notranslate"><span class="pre">sirah_vmdtk.tcl</span></code> sets the right
ones. It also provides a kit of useful selection macros, coloring methods and backmapping utilities.
Use the command <code class="docutils literal notranslate"><span class="pre">sirah_help</span></code> in the Tcl/Tk console of VMD to access the manual pages.</p>
</div>
</section>
<section id="id15">
<h3>3.4. Run the simulation<a class="headerlink" href="#id15" title="Link to this heading"></a></h3>
<p>Make a new folder for the run:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>mkdir<span class="w"> </span>-p<span class="w"> </span>run<span class="p">;</span><span class="w"> </span><span class="nb">cd</span><span class="w"> </span>run
</pre></div>
</div>
<p>The folder <code class="docutils literal notranslate"><span class="pre">sirah.amber/tutorial/3/PMEMD.GPU/</span></code> contains typical input files for energy minimization
(<code class="docutils literal notranslate"><span class="pre">em_HYB.in</span></code>), equilibration (<code class="docutils literal notranslate"><span class="pre">eq_HYB.in</span></code>) and production (<code class="docutils literal notranslate"><span class="pre">md_HYB.in</span></code>) runs. Please check carefully the input flags therein.</p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<blockquote>
<div><p><strong>Some flags used in AMBER</strong></p>
</div></blockquote>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">-i</span></code>: Input file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-o</span></code>: Output file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-p</span></code>: Parameter/topology file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-c</span></code>: Coordinate file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-r</span></code>: Restart file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-x</span></code>: Trajectory file.</p></li>
</ul>
</div>
<div class="admonition caution">
<p class="admonition-title">Caution</p>
<p>These input files are executed by the <strong>GPU</strong> implementation of <code class="docutils literal notranslate"><span class="pre">pmemd.cuda</span></code>. Other available implementations that could be used: <code class="docutils literal notranslate"><span class="pre">sander</span></code>  or <code class="docutils literal notranslate"><span class="pre">pmemd</span></code>, both <strong>CPU</strong> implementations of AMBER.</p>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>You can find example input files for CPU versions of sander and pmemd at folders <code class="docutils literal notranslate"><span class="pre">SANDER/</span></code> and  <code class="docutils literal notranslate"><span class="pre">PMEMD.CPU/</span></code>, within <code class="docutils literal notranslate"><span class="pre">sirah.amber/tutorial/3/</span></code></p>
<p>This simulation is very time consuming owing to the system’s size, so try a parallel or CUDA implementation of AMBER.</p>
</div>
<p><strong>Energy Minimization:</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>sander<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/3/SANDER/em_HYB.in<span class="w"> </span>-p<span class="w"> </span>../dna_hyb.prmtop<span class="w"> </span>-c<span class="w"> </span>../dna_hyb.ncrst<span class="w"> </span>-o<span class="w"> </span>dna_hyb_em.out<span class="w"> </span>-r<span class="w"> </span>dna_hyb_em.ncrst<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>In the course of long MD simulations the capping residues may eventually separate, this effect is
called helix fraying. To avoid such behavior is necessary to set Watson-Crick restraints for the capping base pairs of this CG DNA at the end of <code class="docutils literal notranslate"><span class="pre">eq_GB.in</span></code> and <code class="docutils literal notranslate"><span class="pre">md_GB.in</span></code> files. Check the files lines that start with <em>&amp;rst</em>.</p>
</div>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>If you are using SANDER to avoid such behavior create a symbolic link to the file <code class="docutils literal notranslate"><span class="pre">dna_cg.RST</span></code>, which
contains the definition of Watson-Crick restraints for the capping base pairs of this CG DNA:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>ln<span class="w"> </span>-s<span class="w"> </span>../sirah.amber/tutorial/3/SANDER/dna_cg.RST
</pre></div>
</div>
<p>The file dna_cg.RST can only be read by SANDER, PMEMD reads a different restrain format.</p>
</div>
<p><strong>Equilibration:</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>sander<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/3/SANDER/eq_HYB.in<span class="w"> </span>-p<span class="w"> </span>../dna_hyb.prmtop<span class="w"> </span>-c<span class="w"> </span>dna_hyb_em.ncrst<span class="w"> </span>-o<span class="w"> </span>dna_hyb_eq.out<span class="w"> </span>-r<span class="w"> </span>dna_hyb_eq.ncrst<span class="w"> </span>-x<span class="w"> </span>dna_hyb_eq.nc<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
<p><strong>Production (10ns):</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>sander<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/3/SANDER/md_HYB.in<span class="w"> </span>-p<span class="w"> </span>../dna_hyb.prmtop<span class="w"> </span>-c<span class="w"> </span>dna_hyb_eq.ncrst<span class="w"> </span>-o<span class="w"> </span>dna_hyb_md.out<span class="w"> </span>-r<span class="w"> </span>dna_hyb_md.ncrst<span class="w"> </span>-x<span class="w"> </span>dna_hyb_md.nc<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>You can find example input files for CPU and GPU versions of pmemd at folders PMEMD.CPU/ and PMEMD.GPU/ within sirah.amber/tutorial/3/</p>
</div>
</section>
<section id="id16">
<h3>3.5. Visualizing the simulation<a class="headerlink" href="#id16" title="Link to this heading"></a></h3>
<p>That’s it! Now you can load, visualize and analize the trajectory file in VMD:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">vmd</span> <span class="o">../</span><span class="n">dna_hyb</span><span class="o">.</span><span class="n">prmtop</span> <span class="o">../</span><span class="n">dna_hyb</span><span class="o">.</span><span class="n">ncrst</span> <span class="n">dna_hyb_md</span><span class="o">.</span><span class="n">nc</span> <span class="o">-</span><span class="n">e</span> <span class="o">../</span><span class="n">sirah</span><span class="o">.</span><span class="n">amber</span><span class="o">/</span><span class="n">tools</span><span class="o">/</span><span class="n">sirah_vmdtk</span><span class="o">.</span><span class="n">tcl</span>
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The file <code class="docutils literal notranslate"><span class="pre">sirah_vmdtk.tcl</span></code> is a Tcl script that is part of SIRAH Tools and contains the macros to properly visualize the coarse-grained structures in VMD. Use the command <code class="docutils literal notranslate"><span class="pre">sirah-help</span></code> in the Tcl/Tk console of VMD to access the manual pages.</p>
</div>
</section>
</section>
<section id="closed-circular-dna-in-implicit-solvent">
<span id="tutorial-4"></span><h2>4. Closed circular DNA in implicit solvent<a class="headerlink" href="#closed-circular-dna-in-implicit-solvent" title="Link to this heading"></a></h2>
<p>This tutorial shows how to use the SIRAH force field to perform a coarse grained (CG) simulation of a
closed circular DNA using the Generalized Born model (GB) for implicit solvent.</p>
<p>The main references for this tutorial are: <a class="reference external" href="https://pubs.acs.org/doi/abs/10.1021/ct900653p">SIRAH DNA</a> (latest parameters are those reported in: <a class="reference external" href="https://pubs.acs.org/doi/abs/10.1021/ct100379f">WAT4</a>) and <a class="reference external" href="https://academic.oup.com/bioinformatics/article/32/10/1568/1743152">SIRAH Tools</a>. We strongly advise you to read these articles before starting the tutorial.</p>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>Check <a class="reference internal" href="#download-amber"><span class="std std-ref">download</span></a> section for download and set up details before to start this tutorial.
Since this is <strong>tutorial 4</strong>, remember to replace <code class="docutils literal notranslate"><span class="pre">X.X</span></code>, the files corresponding to this tutorial can be found in: <code class="docutils literal notranslate"><span class="pre">sirah_[version].amber/tutorial/4/</span></code></p>
</div>
<section id="id20">
<h3>4.1. Build CG representations<a class="headerlink" href="#id20" title="Link to this heading"></a></h3>
<p>Map the atomistic structure of the closed circular DNA to its CG representation:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>./sirah.amber/tools/CGCONV/cgconv.pl<span class="w"> </span>-i<span class="w"> </span>./sirah.amber/tutorial/4/ccdna.pdb<span class="w"> </span><span class="p">|</span><span class="w"> </span>sed<span class="w"> </span>-e<span class="w"> </span><span class="s1">&#39;s/DCX A 1/CW5 A 1/&#39;</span><span class="w">  </span>-e<span class="w"> </span><span class="s1">&#39;s/DCX B 1/CW5 B 1/&#39;</span><span class="w"> </span>-e<span class="w"> </span><span class="s1">&#39;s/DGX A 100/GW3 A 100/&#39;</span><span class="w"> </span>-e<span class="w"> </span><span class="s1">&#39;s/DGX B 100/GW3 B 100/&#39;</span><span class="w"> </span>&gt;<span class="w"> </span>ccdna_cg.pdb
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>5’ and 3’ end residues mast be renamed to AW5, TW5, GW5 or CW5 and AW3, TW3, GW3 or CW3 to represent the corresponding Adenine, Thymine, Guanine or Cytosine extremes in a closed circular DNA.</p>
</div>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>This is the basic usage of the script <strong>cgconv.pl</strong>, you can learn other capabilities from its help:
<code class="docutils literal notranslate"><span class="pre">./sirah.amber/tools/CGCONV/cgconv.pl</span> <span class="pre">-h</span></code></p>
</div>
<p>The input file <code class="docutils literal notranslate"><span class="pre">-i</span></code> ccdna.pdb contains all the heavy atoms composing the DNA molecule, while the output <code class="docutils literal notranslate"><span class="pre">-o</span></code> ccdna_cg.pdb preserves a few of them.</p>
<p>Please check both PDB structures using VMD:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>vmd<span class="w"> </span>-m<span class="w"> </span>./sirah.amber/tutorial/4/ccdna.pdb<span class="w"> </span>ccdna_cg.pdb
</pre></div>
</div>
<p>From now on it is just normal AMBER stuff!</p>
</section>
<section id="id21">
<h3>4.2. Prepare LEaP input<a class="headerlink" href="#id21" title="Link to this heading"></a></h3>
<p>Use a text editor to create the file <code class="docutils literal notranslate"><span class="pre">gensystem.leap</span></code> including the following lines:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="gp"># </span>Load<span class="w"> </span>SIRAH<span class="w"> </span>force<span class="w"> </span>field
<span class="go">addPath ./sirah.amber</span>
<span class="go">source leaprc.sirah</span>

<span class="gp"># </span>Load<span class="w"> </span>model
<span class="go">dna = loadpdb ccdna_cg.pdb</span>

<span class="gp"># </span>Make<span class="w"> </span>a<span class="w"> </span>covalent<span class="w"> </span>closed<span class="w"> </span>circular<span class="w"> </span>DNA
<span class="go">bond dna.1.PX dna.100.C1X</span>
<span class="go">bond dna.101.PX dna.200.C1X</span>

<span class="gp"># </span>Save<span class="w"> </span>Parms
<span class="go">saveAmberParmNetcdf dna ccdna_cg.prmtop ccdna_cg.ncrst</span>

<span class="gp"># </span>EXIT
<span class="go">quit</span>
</pre></div>
</div>
</section>
<section id="id22">
<h3>4.3. Run LEaP<a class="headerlink" href="#id22" title="Link to this heading"></a></h3>
<p>Run the LEAP application to generate the molecular topology and initial coordinate files:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>tleap<span class="w"> </span>-f<span class="w"> </span>gensystem.leap
</pre></div>
</div>
<div class="admonition caution">
<p class="admonition-title">Caution</p>
<p>Warning messages about long, triangular or square bonds in <code class="docutils literal notranslate"><span class="pre">leap.log</span></code> file are fine and
expected due to the CG topology.</p>
</div>
<p>This should create a topology file <code class="docutils literal notranslate"><span class="pre">ccdna_cg.prmtop</span></code> and a coordinate file <code class="docutils literal notranslate"><span class="pre">ccdna_cg.ncrst</span></code>.</p>
<p>Use VMD to check how the CG model looks like:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">vmd</span> <span class="n">ccdna_cg</span><span class="o">.</span><span class="n">prmtop</span> <span class="n">ccdna_cg</span><span class="o">.</span><span class="n">ncrst</span> <span class="o">-</span><span class="n">e</span> <span class="o">./</span><span class="n">sirah</span><span class="o">.</span><span class="n">amber</span><span class="o">/</span><span class="n">tools</span><span class="o">/</span><span class="n">sirah_vmdtk</span><span class="o">.</span><span class="n">tcl</span>
</pre></div>
</div>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>VMD assigns default radius to unknown atom types, the script <code class="docutils literal notranslate"><span class="pre">sirah_vmdtk.tcl</span></code> sets the right
ones. It also provides a kit of useful selection macros, coloring methods and backmapping utilities.
Use the command <code class="docutils literal notranslate"><span class="pre">sirah_help</span></code> in the Tcl/Tk console of VMD to access the manual pages.</p>
</div>
</section>
<section id="id23">
<h3>4.4. Run the simulation<a class="headerlink" href="#id23" title="Link to this heading"></a></h3>
<p>Make a new folder for the run:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>mkdir<span class="w"> </span>-p<span class="w"> </span>run<span class="p">;</span><span class="w"> </span><span class="nb">cd</span><span class="w"> </span>run
</pre></div>
</div>
<p>The folder <code class="docutils literal notranslate"><span class="pre">sirah.amber/tutorial/4/GPU</span></code> contains typical input files for energy minimization (<code class="docutils literal notranslate"><span class="pre">em_GBSA.in</span></code>), equilibration (<code class="docutils literal notranslate"><span class="pre">eq_GBSA.in</span></code>) and production (<code class="docutils literal notranslate"><span class="pre">md_GBSA.in</span></code>) runs. Please check carefully the input flags therein.</p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<blockquote>
<div><p><strong>Some flags used in AMBER</strong></p>
</div></blockquote>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">-i</span></code>: Input file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-o</span></code>: Output file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-p</span></code>: Parameter/topology file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-c</span></code>: Coordinate file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-r</span></code>: Restart file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-x</span></code>: Trajectory file.</p></li>
</ul>
</div>
<div class="admonition caution">
<p class="admonition-title">Caution</p>
<p>These input files are executed by the <strong>GPU</strong> implementation of <code class="docutils literal notranslate"><span class="pre">pmemd.cuda</span></code>. Other available implementations that could be used: <code class="docutils literal notranslate"><span class="pre">sander</span></code>  or <code class="docutils literal notranslate"><span class="pre">pmemd</span></code>, both <strong>CPU</strong> implementations of AMBER.</p>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>You can find example input files for CPU and GPU, within <code class="docutils literal notranslate"><span class="pre">sirah.amber/tutorial/4/</span></code></p>
</div>
<p><strong>Energy Minimization:</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/4/GPU/em_GBSA.in<span class="w"> </span>-p<span class="w"> </span>../ccdna_cg.prmtop<span class="w"> </span>-c<span class="w"> </span>../ccdna_cg.ncrst<span class="w"> </span>-o<span class="w"> </span>ccdna_cg_em.out<span class="w"> </span>-r<span class="w"> </span>ccdna_cg_em.ncrst<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
<p><strong>Equilibration:</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/4/GPU/eq_GBSA.in<span class="w"> </span>-p<span class="w"> </span>../ccdna_cg.prmtop<span class="w"> </span>-c<span class="w"> </span>ccdna_cg_em.ncrst<span class="w"> </span>-o<span class="w"> </span>ccdna_cg_eq.out<span class="w"> </span>-r<span class="w"> </span>ccdna_cg_eq.ncrst<span class="w"> </span>-x<span class="w"> </span>ccdna_cg_eq.nc<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
<p><strong>Production (100ns):</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/4/GPU/md_GBSA.in<span class="w"> </span>-p<span class="w"> </span>../ccdna_cg.prmtop<span class="w"> </span>-c<span class="w"> </span>ccdna_cg_eq.ncrst<span class="w"> </span>-o<span class="w"> </span>ccdna_cg_md.out<span class="w"> </span>-r<span class="w"> </span>ccdna_cg_md.ncrst<span class="w"> </span>-x<span class="w"> </span>ccdna_cg_md.nc<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
</section>
<section id="id24">
<h3>4.5. Visualizing the simulation<a class="headerlink" href="#id24" title="Link to this heading"></a></h3>
<p>That’s it! Now you can load, visualize and analize the trajectory file in VMD:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">vmd</span> <span class="o">../</span><span class="n">ccdna_cg</span><span class="o">.</span><span class="n">prmtop</span> <span class="o">../</span><span class="n">ccdna_cg</span><span class="o">.</span><span class="n">ncrst</span> <span class="n">ccdna_cg_md</span><span class="o">.</span><span class="n">nc</span> <span class="o">-</span><span class="n">e</span> <span class="o">../</span><span class="n">sirah</span><span class="o">.</span><span class="n">amber</span><span class="o">/</span><span class="n">tools</span><span class="o">/</span><span class="n">sirah_vmdtk</span><span class="o">.</span><span class="n">tcl</span>
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The file <code class="docutils literal notranslate"><span class="pre">sirah_vmdtk.tcl</span></code> is a Tcl script that is part of SIRAH Tools and contains the macros to properly visualize the coarse-grained structures in VMD. Use the command <code class="docutils literal notranslate"><span class="pre">sirah-help</span></code> in the Tcl/Tk console of VMD to access the manual pages.</p>
</div>
</section>
</section>
<section id="proteins-in-explicit-solvent">
<span id="tutorial-5"></span><h2>5. Proteins in explicit solvent<a class="headerlink" href="#proteins-in-explicit-solvent" title="Link to this heading"></a></h2>
<p>This tutorial shows how to use the SIRAH force field to perform a coarse grained (CG) simulation of a
protein in explicit solvent (called WatFour, WT4). The main references for
this tutorial are: <a class="reference external" href="https://pubs.acs.org/doi/abs/10.1021/ct100379f">WAT4</a>, <a class="reference external" href="https://doi.org/10.1021/acs.jctc.9b00006">SIRAH 2.0</a> and <a class="reference external" href="https://academic.oup.com/bioinformatics/article/32/10/1568/1743152">SIRAH Tools</a>. We strongly advise you to read these articles before starting the tutorial.</p>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>Check <a class="reference internal" href="#download-amber"><span class="std std-ref">download</span></a> section for download and set up details before to start this tutorial.
Since this is <strong>tutorial 5</strong>, remember to replace <code class="docutils literal notranslate"><span class="pre">X.X</span></code> in your folder directory. The files corresponding to this tutorial can be found in: <code class="docutils literal notranslate"><span class="pre">sirah_[version].amber/tutorial/5/</span></code></p>
</div>
<section id="id27">
<h3>5.1. Build CG representations<a class="headerlink" href="#id27" title="Link to this heading"></a></h3>
<div class="admonition caution">
<p class="admonition-title">Caution</p>
<p>The mapping to CG requires the correct protonation state of each residue at a given pH. We recommend using the <a class="reference external" href="https://server.poissonboltzmann.org/pdb2pqr">PDB2PQR server</a> and choosing the output naming scheme of AMBER for best compatibility. Be aware that modified residues lacking parameters such as: MSE (seleno MET), TPO (phosphorylated THY), SEP (phosphorylated SER) or others are deleted from the PQR file by the server. In that case, mutate the residues to their unmodified form before submitting the structure to the server.</p>
</div>
<p>Map the protonated atomistic structure of protein <a class="reference external" href="https://www.rcsb.org/structure/1CRN">1CRN</a> to its CG representation:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>./sirah.amber/tools/CGCONV/cgconv.pl<span class="w"> </span>-i<span class="w"> </span>./sirah.amber/tutorial/5/1CRN.pqr<span class="w"> </span>-o<span class="w"> </span>1CRN_cg.pdb
</pre></div>
</div>
<p>The input file <code class="docutils literal notranslate"><span class="pre">-i</span></code> 1CRN.pqr contains the atomistic representation of <a class="reference external" href="https://www.rcsb.org/structure/1CRN">1CRN</a> structure at pH <strong>7.0</strong>, while the output <code class="docutils literal notranslate"><span class="pre">-o</span></code> 1CRN_cg.pdb is its SIRAH CG representation.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p><strong>Pay attention to residue names when mapping structures from other atomistic force fields or experimental structures.</strong> Although we provide compatibility for naming schemes in PDB, GMX, GROMOS, CHARMM and OPLS, there always may be some ambiguity in the residue naming, specially regarding protonation states, that may lead to a wrong mapping. For example, SIRAH Tools always maps the residue name “HIS” to a Histidine protonated at epsilon nitrogen (<span class="math notranslate nohighlight">\(N_{\epsilon}\)</span>) regardless the actual proton placement. Similarly, protonated Glutamic and Aspartic acid residues must be named “GLH” and “ASH”, otherwise they will be treated as negative charged residues. In addition, protonated and disulfide bonded Cysteines must be named “CYS” and “CYX” respectively. These kind of situations need to be carefully checked by the users. In all cases the residues preserve their identity when mapping and back-mapping the structures. Hence, the total charge of the protein should be the same at atomistic and SIRAH level. You can check the following mapping file to be sure of the compatibility: <code class="docutils literal notranslate"><span class="pre">sirah.amber/tools/CGCONV/maps/sirah_prot.map</span></code>.</p>
</div>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>By default charged termini are used, but it is possible to set then neutral by renaming the residues from <strong>s</strong>[code] to <strong>a</strong>[code] (Nt-acetylated) or <strong>m</strong>[code] (Ct-amidated) after mapping to CG, where [code] is the root residue name in SIRAH. For example, to set a neutral N-terminal Histidine protonated at epsilon nitrogen (<span class="math notranslate nohighlight">\(N_{\epsilon}\)</span>) rename it from “sHe” to “aHe”.</p>
</div>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>This is the basic usage of the script <strong>cgconv.pl</strong>, you can learn other capabilities from its help:
<code class="docutils literal notranslate"><span class="pre">./sirah.amber/tools/CGCONV/cgconv.pl</span> <span class="pre">-h</span></code></p>
</div>
<p>Please check both PDB and PQR structures using VMD:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>vmd<span class="w"> </span>-m<span class="w"> </span>sirah.amber/tutorial/5/1CRN.pqr<span class="w"> </span>1CRN_cg.pdb
</pre></div>
</div>
<p>From now on it is just normal AMBER stuff!</p>
</section>
<section id="id29">
<h3>5.2. Prepare LEaP input<a class="headerlink" href="#id29" title="Link to this heading"></a></h3>
<p>Use a text editor to create the file <code class="docutils literal notranslate"><span class="pre">gensystem.leap</span></code> including the following lines:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="gp"># </span>Load<span class="w"> </span>SIRAH<span class="w"> </span>force<span class="w"> </span>field
<span class="go">addPath ./sirah.amber</span>
<span class="go">source leaprc.sirah</span>

<span class="gp"># </span>Load<span class="w"> </span>model
<span class="go">protein = loadpdb 1CRN_cg.pdb</span>

<span class="gp"># </span>Info<span class="w"> </span>on<span class="w"> </span>system<span class="w"> </span>charge
<span class="go">charge protein</span>

<span class="gp"># </span>Set<span class="w"> </span>S-S<span class="w"> </span>bridges
<span class="go">bond protein.3.BSG protein.40.BSG</span>
<span class="go">bond protein.4.BSG protein.32.BSG</span>
<span class="go">bond protein.16.BSG protein.26.BSG</span>

<span class="gp"># </span>Add<span class="w"> </span>solvent,<span class="w"> </span>counterions<span class="w"> </span>and<span class="w"> </span><span class="m">0</span>.15M<span class="w"> </span>NaCl
<span class="gp"># </span>Tuned<span class="w"> </span>solute-solvent<span class="w"> </span>closeness<span class="w"> </span><span class="k">for</span><span class="w"> </span>best<span class="w"> </span>hydration
<span class="go">solvateOct protein WT4BOX 20 0.7</span>
<span class="go">addIonsRand protein NaW 22 ClW 22</span>

<span class="gp"># </span>Save<span class="w"> </span>Parms
<span class="go">saveAmberParmNetcdf protein 1CRN_cg.prmtop 1CRN_cg.ncrst</span>

<span class="gp"># </span>EXIT
<span class="go">quit</span>
</pre></div>
</div>
<div class="admonition caution">
<p class="admonition-title">Caution</p>
<p>Each disulfide bond must be defined explicitly in LEaP using the command bond, e.g.: “<em>bond unit.ri.BSG unit.rj.BSG</em>”. Where <em>ri</em> and <em>rj</em> correspond to the residue index in the topology file starting from 1, which may differ from the biological sequence in the PDB file. You can try the command <em>pdb4amber</em> to get those indexes from the atomistic structure, but be aware that it may not work if the Cysteine residues are too far away:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pdb4amber<span class="w"> </span>-i<span class="w"> </span>sirah.amber/tutorial/5/1CRN.pqr<span class="w"> </span>-o<span class="w"> </span>1CRN_aa.pdb<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span>cat<span class="w"> </span>1CRN_aa_sslink
</pre></div>
</div>
</div>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<p>The available ionic species in SIRAH force field are: <code class="docutils literal notranslate"><span class="pre">Na⁺</span></code> (NaW), <code class="docutils literal notranslate"><span class="pre">K⁺</span></code> (KW) and <code class="docutils literal notranslate"><span class="pre">Cl⁻</span></code> (ClW). One ion pair (e.g. NaW-ClW) each 34 WT4 molecules renders a salt concentration of ~0.15M (see <a class="reference internal" href="#appendix"><span class="std std-ref">Appendix</span></a> for details).
Counterions were added according to <a class="reference external" href="https://pubs.acs.org/doi/10.1021/acs.jctc.9b00953">Machado et al.</a>.</p>
</div>
</section>
<section id="id31">
<h3>5.3. Run LEaP<a class="headerlink" href="#id31" title="Link to this heading"></a></h3>
<p>Run the LEaP application to generate the molecular topology and initial coordinate files:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>tleap<span class="w"> </span>-f<span class="w"> </span>gensystem.leap
</pre></div>
</div>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>Warning messages about long, triangular or square bonds in <code class="docutils literal notranslate"><span class="pre">leap.log</span></code> file are fine and expected due to the CG topology.</p>
</div>
<p>This should create a topology file <code class="docutils literal notranslate"><span class="pre">1CRN_cg.prmtop</span></code> and a coordinate file <code class="docutils literal notranslate"><span class="pre">1CRN_cg.ncrst</span></code>.</p>
<p>Use VMD to check how the CG model looks like and particularly the presence of disulfide bonds:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>vmd<span class="w"> </span>1CRN_cg.prmtop<span class="w"> </span>1CRN_cg.ncrst<span class="w"> </span>-e<span class="w"> </span>./sirah.amber/tools/sirah_vmdtk.tcl
</pre></div>
</div>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>VMD assigns default radius to unknown atom types, the script <code class="docutils literal notranslate"><span class="pre">sirah_vmdtk.tcl</span></code> sets the right
ones. It also provides a kit of useful selection macros, coloring methods and backmapping utilities.
Use the command <code class="docutils literal notranslate"><span class="pre">sirah_help</span></code> in the Tcl/Tk console of VMD to access the manual pages.</p>
</div>
</section>
<section id="id32">
<h3>5.4. Run the simulation<a class="headerlink" href="#id32" title="Link to this heading"></a></h3>
<p>Make a new folder for the run:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>mkdir<span class="w"> </span>-p<span class="w"> </span>run<span class="p">;</span><span class="w"> </span><span class="nb">cd</span><span class="w"> </span>run
</pre></div>
</div>
<p>The folder <code class="docutils literal notranslate"><span class="pre">sirah.amber/tutorial/5/</span></code> contains typical input files for energy minimization
(<code class="docutils literal notranslate"><span class="pre">em1_WT4.in</span></code> and <code class="docutils literal notranslate"><span class="pre">em2_WT4.in</span></code>), equilibration (<code class="docutils literal notranslate"><span class="pre">eq1_WT4.in</span></code> and <code class="docutils literal notranslate"><span class="pre">eq2_WT4.in</span></code>) and production (<code class="docutils literal notranslate"><span class="pre">md_WT4.in</span></code>) runs. Please check carefully the
input flags therein, in particular the definition of flag <em>chngmask=0</em> at <em>&amp;ewald</em> section is <strong>mandatory</strong>.</p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<blockquote>
<div><p><strong>Some flags used in AMBER</strong></p>
</div></blockquote>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">-i</span></code>: Input file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-o</span></code>: Output file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-p</span></code>: Parameter/topology file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-c</span></code>: Coordinate file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-r</span></code>: Restart file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-x</span></code>: Trajectory file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-ref</span></code>: Reference file</p></li>
</ul>
</div>
<div class="admonition caution">
<p class="admonition-title">Caution</p>
<p>These input files are executed by the <strong>GPU</strong> implementation of <code class="docutils literal notranslate"><span class="pre">pmemd.cuda</span></code>. Other available implementations that could be used: <code class="docutils literal notranslate"><span class="pre">sander</span></code>  or <code class="docutils literal notranslate"><span class="pre">pmemd</span></code>, both <strong>CPU</strong> implementations of AMBER.</p>
</div>
<p><strong>Energy Minimization of side chains and solvent by restraining the backbone:</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/5/em1_WT4.in<span class="w"> </span>-p<span class="w"> </span>../1CRN_cg.prmtop<span class="w"> </span>-c<span class="w"> </span>../1CRN_cg.ncrst<span class="w"> </span>-ref<span class="w"> </span>../1CRN_cg.ncrst<span class="w"> </span>-o<span class="w"> </span>1CRN_cg_em1.out<span class="w"> </span>-r<span class="w"> </span>1CRN_cg_em1.ncrst<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
<p><strong>Energy Minimization of whole system:</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/5/em2_WT4.in<span class="w"> </span>-p<span class="w"> </span>../1CRN_cg.prmtop<span class="w"> </span>-c<span class="w"> </span>../1CRN_cg_em1.ncrst<span class="w"> </span>-o<span class="w"> </span>1CRN_cg_em2.out<span class="w"> </span>-r<span class="w"> </span>1CRN_cg_em2.ncrst<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
<p><strong>Solvent Equilibration (NPT):</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/5/eq1_WT4.in<span class="w"> </span>-p<span class="w"> </span>../1CRN_cg.prmtop<span class="w"> </span>-c<span class="w"> </span>1CRN_cg_em2.ncrst<span class="w"> </span>-ref<span class="w"> </span>1CRN_cg_em2.ncrst<span class="w"> </span>-o<span class="w"> </span>1CRN_cg_eq1.out<span class="w"> </span>-r<span class="w"> </span>1CRN_cg_eq1.ncrst<span class="w"> </span>-x<span class="w"> </span>1CRN_cg_eq1.nc<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
<div class="admonition caution">
<p class="admonition-title">Caution</p>
<p>Option <strong>restraintmask=:’1-46’</strong> in input file <code class="docutils literal notranslate"><span class="pre">eq1_WT4.in</span></code> must be set specifically for each system to embrace all protein’s residues.</p>
</div>
<p><strong>Soft equilibration to improve side chain solvation (NPT):</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/5/eq2_WT4.in<span class="w"> </span>-p<span class="w"> </span>../1CRN_cg.prmtop<span class="w"> </span>-c<span class="w"> </span>1CRN_cg_eq1.ncrst<span class="w"> </span>-ref<span class="w"> </span>1CRN_cg_eq1.ncrst<span class="w"> </span>-o<span class="w"> </span>1CRN_cg_eq2.out<span class="w"> </span>-r<span class="w"> </span>1CRN_cg_eq2.ncrst<span class="w"> </span>-x<span class="w"> </span>1CRN_cg_eq2.nc<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
<p><strong>Production (1000ns):</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/5/md_WT4.in<span class="w"> </span>-p<span class="w"> </span>../1CRN_cg.prmtop<span class="w"> </span>-c<span class="w"> </span>1CRN_cg_eq2.ncrst<span class="w"> </span>-o<span class="w"> </span>1CRN_cg_md.out<span class="w"> </span>-r<span class="w"> </span>1CRN_cg_md.ncrst<span class="w"> </span>-x<span class="w"> </span>1CRN_cg_md.nc<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>The same input files can be used to run on CPU <em>pmemd</em> or <em>sander</em>.</p>
</div>
</section>
<section id="id33">
<h3>5.5. Visualizing the simulation<a class="headerlink" href="#id33" title="Link to this heading"></a></h3>
<p>That’s it! Now you can analyze the trajectory.
Process the output trajectory to account for the Periodic Boundary Conditions (PBC):</p>
<blockquote>
<div><div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;autoimage\ngo\nquit\n&quot;</span><span class="w"> </span><span class="p">|</span><span class="w"> </span>cpptraj<span class="w"> </span>-p<span class="w"> </span>../1CRN_cg.prmtop<span class="w"> </span>-y<span class="w"> </span>1CRN_cg_md.nc<span class="w"> </span>-x<span class="w"> </span>1CRN_cg_md_pbc.nc<span class="w"> </span>--interactive
</pre></div>
</div>
</div></blockquote>
<p><strong>Load the processed trajectory in VMD:</strong></p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">vmd</span> <span class="o">../</span><span class="mi">1</span><span class="n">CRN_cg</span><span class="o">.</span><span class="n">prmtop</span> <span class="o">../</span><span class="mi">1</span><span class="n">CRN_cg</span><span class="o">.</span><span class="n">ncrst</span> <span class="mi">1</span><span class="n">CRN_cg_md</span><span class="o">.</span><span class="n">nc</span> <span class="o">-</span><span class="n">e</span> <span class="o">../</span><span class="n">sirah</span><span class="o">.</span><span class="n">amber</span><span class="o">/</span><span class="n">tools</span><span class="o">/</span><span class="n">sirah_vmdtk</span><span class="o">.</span><span class="n">tcl</span>
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The file <code class="docutils literal notranslate"><span class="pre">sirah_vmdtk.tcl</span></code> is a Tcl script that is part of SIRAH Tools and contains the macros to properly visualize the coarse-grained structures in VMD. Use the command <code class="docutils literal notranslate"><span class="pre">sirah_help</span></code> in the Tcl/Tk console of VMD to access the manual pages.</p>
</div>
</section>
</section>
<section id="lipid-bilayers-in-explicit-solvent">
<span id="tutorial-6"></span><h2>6. Lipid bilayers in explicit solvent<a class="headerlink" href="#lipid-bilayers-in-explicit-solvent" title="Link to this heading"></a></h2>
<p>This tutorial shows how to use the SIRAH force field to perform a coarse grained (CG) simulation of a
DMPC bilayer in explicit solvent (called WatFour, WT4). The main references for
this tutorial are: <a class="reference external" href="https://doi.org/10.1021/acs.jctc.9b00435">SIRAH Lipids</a>, and <a class="reference external" href="https://academic.oup.com/bioinformatics/article/32/10/1568/1743152">SIRAH Tools</a>.
We strongly advise you to read these articles before starting the tutorial.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>If you are not familiar with membrane stuff we strongly recommend you to first perform the <a class="reference external" href="http://ambermd.org/tutorials/advanced/tutorial16/">AMBER tutorial on lipids</a>.</p>
</div>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>Check <a class="reference internal" href="#download-amber"><span class="std std-ref">download</span></a> section for download and set up details before to start this tutorial.
Since this is <strong>tutorial 6</strong>, remember to replace <code class="docutils literal notranslate"><span class="pre">X.X</span></code> in your folder directory. The files corresponding to this tutorial can be found in: <code class="docutils literal notranslate"><span class="pre">sirah_[version].amber/tutorial/6/</span></code></p>
</div>
<section id="id35">
<h3>6.1. Build CG representations<a class="headerlink" href="#id35" title="Link to this heading"></a></h3>
<p>Map the atomistic structure of the preassembled DMPC bilayer to its CG representation:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>./sirah.amber/tools/CGCONV/cgconv.pl<span class="w"> </span>-i<span class="w"> </span>./sirah.amber/tutorial/6/DMPC64.pdb<span class="w"> </span>-o<span class="w"> </span>DMPC64_cg.pdb<span class="w"> </span>-a<span class="w"> </span>sirah.amber/tools/CGCONV/maps/tieleman_lipid.map
</pre></div>
</div>
<p>The input file <code class="docutils literal notranslate"><span class="pre">-i</span></code> DMPC64.pdb contains the atomistic representation of the DMPC bilayer, while the output <code class="docutils literal notranslate"><span class="pre">-o</span></code> DMPC64_cg.pdb is its SIRAH CG representation. The flag <code class="docutils literal notranslate"><span class="pre">-a</span></code> is a mapping scheme for lipids.</p>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>By default no mapping is applied to lipids, as there is no standard naming convention for them. So users are requested to append a MAP file from the list in <a class="reference internal" href="#table"><span class="std std-ref">Table 1</span></a>, by setting the flag <code class="docutils literal notranslate"><span class="pre">-a</span></code> in <strong>cgconv.pl</strong>. We recommend using <a class="reference external" href="https://m3g.github.io/packmol/">PACKMOL</a> for building the system. Reference building-block structures are provided at folder <code class="docutils literal notranslate"><span class="pre">sirah.amber/PDB/</span></code>, which agree with the mapping scheme in <code class="docutils literal notranslate"><span class="pre">sirah.amber/tools/CGCONV/maps/tieleman_lipid.map</span></code>. The provided DMPC bilayer contains 64 lipid molecules per leaflet distributed in a 6.4 * 6.4 nm surface, taking into account an approximate area per lipid of 0.64 nm<sup>2</sup> at 333 K  . The starting configuration was created with the input file <code class="docutils literal notranslate"><span class="pre">sirah.amber/tutorial/6/DMPC_bilayer.pkm</span></code>. See <a class="reference internal" href="FAQ.html"><span class="doc">FAQs</span></a> for cautions on mapping lipids to SIRAH and tips on using fragment-based topologies.</p>
</div>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>This an advanced usage of the script <strong>cgconv.pl</strong>, you can learn other capabilities from its help:
<code class="docutils literal notranslate"><span class="pre">./sirah.amber/tools/CGCONV/cgconv.pl</span> <span class="pre">-h</span></code></p>
</div>
<p>The input file <code class="docutils literal notranslate"><span class="pre">DMPC64.pdb</span></code> contains all the heavy atoms composing the lipids, while the output <code class="docutils literal notranslate"><span class="pre">DMPC64_cg.pdb</span></code> preserves a few of them. Please check both PDB and PQR structures using VMD:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>vmd<span class="w"> </span>-m<span class="w"> </span>sirah.amber/tutorial/6/DMPC64.pdb<span class="w"> </span>DMPC64_cg.pdb
</pre></div>
</div>
<p>From now on it is just normal AMBER stuff!</p>
</section>
<section id="id36">
<h3>6.2. Prepare LEaP input<a class="headerlink" href="#id36" title="Link to this heading"></a></h3>
<p>Use a text editor to create the file <code class="docutils literal notranslate"><span class="pre">gensystem.leap</span></code> including the following lines:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="gp"># </span>Load<span class="w"> </span>SIRAH<span class="w"> </span>force<span class="w"> </span>field
<span class="go">addPath ./sirah.amber</span>
<span class="go">source leaprc.sirah</span>

<span class="gp"># </span>Load<span class="w"> </span>model
<span class="go">Lipid = loadpdb DMPC64_cg.pdb</span>

<span class="gp"># </span>Add<span class="w"> </span>solvent,<span class="w"> </span>counterions<span class="w"> </span>and<span class="w"> </span><span class="m">0</span>.15M<span class="w"> </span>NaCl
<span class="gp"># </span>Tuned<span class="w"> </span>solute-solvent<span class="w"> </span>closeness<span class="w"> </span><span class="k">for</span><span class="w"> </span>best<span class="w"> </span>hydration
<span class="go">solvateBox Lipid WT4BOX {0 0 40} 0.7</span>
<span class="go">addIonsRand Lipid NaW 33 ClW 33</span>

<span class="gp"># </span>Save<span class="w"> </span>Parms
<span class="go">saveAmberParmNetcdf protein DMPC64_cg.prmtop DMPC64_cg.ncrst</span>

<span class="gp"># </span>EXIT
<span class="go">quit</span>
</pre></div>
</div>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<p>The available ionic species in SIRAH force field are: <code class="docutils literal notranslate"><span class="pre">Na⁺</span></code> (NaW), <code class="docutils literal notranslate"><span class="pre">K⁺</span></code> (KW) and <code class="docutils literal notranslate"><span class="pre">Cl⁻</span></code> (ClW). One ion pair (e.g. NaW-ClW) each 34 WT4 molecules renders a salt concentration of ~0.15M (see <a class="reference internal" href="#appendix"><span class="std std-ref">Appendix</span></a> for details).
Counterions were added according to <a class="reference external" href="https://pubs.acs.org/doi/10.1021/acs.jctc.9b00953">Machado et al.</a>.</p>
</div>
</section>
<section id="id38">
<h3>6.3. Run LEaP<a class="headerlink" href="#id38" title="Link to this heading"></a></h3>
<p>Run the LEaP application to generate the molecular topology and initial coordinate files:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>tleap<span class="w"> </span>-f<span class="w"> </span>gensystem.leap
</pre></div>
</div>
<div class="admonition caution">
<p class="admonition-title">Caution</p>
<p>Warning messages about long, triangular or square bonds in <code class="docutils literal notranslate"><span class="pre">leap.log</span></code> file are fine and expected due to the CG topology.</p>
</div>
<p>This should create a topology file <code class="docutils literal notranslate"><span class="pre">DMPC64_cg.prmtop</span></code> and a coordinate file <code class="docutils literal notranslate"><span class="pre">DMPC64_cg.ncrst</span></code>.</p>
<p>Use VMD to check how the CG model looks:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>vmd<span class="w"> </span>DMPC64_cg.prmtop<span class="w"> </span>DMPC64_cg.ncrst<span class="w"> </span>-e<span class="w"> </span>./sirah.amber/tools/sirah_vmdtk.tcl
</pre></div>
</div>
<p>By selecting +X, +Y and +Z periodic images from the <em>Periodic</em> tab in the <em>Graphical Representations</em> window you will see small vacuum slices at box boundaries. In the following step we will fix this issue by reducing the box dimensions a few angstroms. See <a class="reference internal" href="FAQ.html"><span class="doc">FAQs</span></a> for issues on membrane systems in AMBER.</p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>VMD assigns default radius to unknown atom types, the script <code class="docutils literal notranslate"><span class="pre">sirah_vmdtk.tcl</span></code> sets the right
ones. It also provides a kit of useful selection macros, coloring methods and backmapping utilities.
Use the command <code class="docutils literal notranslate"><span class="pre">sirah_help</span></code> in the Tcl/Tk console of VMD to access the manual pages.</p>
</div>
</section>
<section id="resize-the-box-with-cpptraj">
<h3>6.4. Resize the box with CPPTRAJ<a class="headerlink" href="#resize-the-box-with-cpptraj" title="Link to this heading"></a></h3>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>As PACKMOL does not consider periodicity while building up the system, increasing the XY box sides a few Angstroms may be required to avoid bad contacts between images.</p>
</div>
<p>Use a text editor to create the file <code class="docutils literal notranslate"><span class="pre">resize_box.cpptraj</span></code> including the following lines:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="gp"># </span>New<span class="w"> </span>box<span class="w"> </span>dimensions
<span class="go">box x 66 y 66 z 132</span>

<span class="gp"># </span>Amber<span class="w"> </span>NetCDF<span class="w"> </span>Restart<span class="w"> </span>generation
<span class="go">trajout DMPC64_cg_nb.ncrst</span>

<span class="gp"># </span>Do<span class="w"> </span>it!
<span class="go">go</span>

<span class="gp"># </span>EXIT
<span class="go">quit</span>
</pre></div>
</div>
<p>Run the CPPTRAJ application to adjust the size of the simulation box:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>cpptraj<span class="w"> </span>-p<span class="w"> </span>DMPC64_cg.prmtop<span class="w"> </span>-y<span class="w"> </span>DMPC64_cg.ncrst<span class="w"> </span>-i<span class="w"> </span>resize_box.cpptraj
</pre></div>
</div>
<p>Once again, use VMD to check the PBC images in the new box of the system:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>vmd<span class="w"> </span>DMPC64_cg.prmtop<span class="w"> </span>DMPC64_cg_nb.ncrst<span class="w"> </span>-e<span class="w"> </span>./sirah.amber/tools/sirah_vmdtk.tcl
</pre></div>
</div>
</section>
<section id="id39">
<h3>6.5. Run the simulation<a class="headerlink" href="#id39" title="Link to this heading"></a></h3>
<p>Make a new folder for the run:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>mkdir<span class="w"> </span>-p<span class="w"> </span>run<span class="p">;</span><span class="w"> </span><span class="nb">cd</span><span class="w"> </span>run
</pre></div>
</div>
<p>The folder <code class="docutils literal notranslate"><span class="pre">sirah.amber/tutorial/6/</span></code> contains typical input files for energy minimization
(<code class="docutils literal notranslate"><span class="pre">em_Lipid.in</span></code>), heating (<code class="docutils literal notranslate"><span class="pre">heat_Lipid.in</span></code>), equilibration (<code class="docutils literal notranslate"><span class="pre">eq_Lipid.in</span></code>) and production (<code class="docutils literal notranslate"><span class="pre">md_Lipid.in</span></code>) runs. Please check carefully the input flags.</p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<blockquote>
<div><p><strong>Some flags used in AMBER</strong></p>
</div></blockquote>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">-i</span></code>: Input file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-o</span></code>: Output file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-p</span></code>: Parameter/topology file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-c</span></code>: Coordinate file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-r</span></code>: Restart file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-x</span></code>: Trajectory file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">ref</span></code>: Reference file</p></li>
</ul>
</div>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>These input files are executed by the <strong>GPU</strong> implementation of <em>pmemd.cuda</em>, due to the system size we do not recommend the use of <strong>CPU</strong> implementations of AMBER.</p>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Other available implementations that could be used: <code class="docutils literal notranslate"><span class="pre">sander</span></code>  or <code class="docutils literal notranslate"><span class="pre">pmemd</span></code>, both <strong>CPU</strong> implementations of AMBER.</p>
</div>
<p><strong>Energy Minimization:</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/6/em_Lipid.in<span class="w"> </span>-p<span class="w"> </span>../DMPC64_cg.prmtop<span class="w"> </span>-c<span class="w"> </span>../DMPC64_cg_nb.ncrst<span class="w"> </span>-o<span class="w"> </span>DMPC64_cg_em.out<span class="w"> </span>-r<span class="w"> </span>DMPC64_cg_em.ncrst<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
<p><strong>Heating:</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/6/heat_Lipid.in<span class="w"> </span>-p<span class="w"> </span>../DMPC64_cg.prmtop<span class="w"> </span>-c<span class="w"> </span>DMPC64_cg_em.ncrst<span class="w"> </span>-o<span class="w"> </span>DMPC64_cg_eq_0.out<span class="w"> </span>-r<span class="w"> </span>DMPC64_cg_eq_0.ncrst<span class="w"> </span>-x<span class="w"> </span>DMPC64_cg_eq_0.nc<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>To avoid “<em>skinnb errors</em>” on GPU due to large box size fluctuations, the system must be equilibrated by several “short” runs using a large <em>skinnb</em> value. The number and length of the runs may vary according to the characteristic stabilization times of the system. For more information visit the <a class="reference external" href="http://ambermd.org/tutorials/advanced/tutorial16/">AMBER tutorial on lipids</a>.</p>
</div>
<p><strong>Periodic box equilibration in GPU code (500 ps x 9):</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="k">for</span><span class="w"> </span>i<span class="w"> </span><span class="k">in</span><span class="w"> </span><span class="k">$(</span>seq<span class="w"> </span><span class="m">1</span><span class="w"> </span><span class="m">9</span><span class="k">)</span>
<span class="k">do</span>
<span class="w">        </span><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;running equilibration </span><span class="nv">$i</span><span class="s2">&quot;</span>
<span class="w">        </span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span><span class="se">\</span>
<span class="w">        </span>-i<span class="w"> </span>../sirah.amber/tutorial/6/eq_Lipid.in<span class="w"> </span><span class="se">\</span>
<span class="w">        </span>-p<span class="w"> </span>../DMPC64_cg.prmtop<span class="w"> </span><span class="se">\</span>
<span class="w">        </span>-c<span class="w"> </span>DMPC64_cg_eq_<span class="k">$((</span><span class="nv">$i</span><span class="w"> </span><span class="o">-</span><span class="m">1</span><span class="k">))</span>.ncrst<span class="w"> </span><span class="se">\</span>
<span class="w">        </span>-o<span class="w"> </span>DMPC64_cg_eq_<span class="nv">$i</span>.out<span class="w"> </span><span class="se">\</span>
<span class="w">        </span>-r<span class="w"> </span>DMPC64_cg_eq_<span class="nv">$i</span>.ncrst<span class="w"> </span><span class="se">\</span>
<span class="w">        </span>-x<span class="w"> </span>DMPC64_cg_eq_<span class="nv">$i</span>.nc
<span class="k">done</span><span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
<p><strong>Production (1000ns):</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/6/md_Lipid.in<span class="w"> </span>-p<span class="w"> </span>../DMPC64_cg.prmtop<span class="w"> </span>-c<span class="w"> </span>DMPC64_cg_eq_9.ncrst<span class="w"> </span>-o<span class="w"> </span>DMPC64_cg_md.out<span class="w"> </span>-r<span class="w"> </span>DMPC64_cg_md.ncrst<span class="w"> </span>-x<span class="w"> </span>DMPC64_cg_md.nc<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
</section>
<section id="id40">
<h3>6.6. Visualizing the simulation<a class="headerlink" href="#id40" title="Link to this heading"></a></h3>
<p>That’s it! Now you can analyze the trajectory.</p>
<p>Process the output trajectory to account for the Periodic Boundary Conditions (PBC):</p>
<blockquote>
<div><div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;autoimage\ngo\nquit\n&quot;</span><span class="w"> </span><span class="p">|</span><span class="w"> </span>cpptraj<span class="w"> </span>-p<span class="w"> </span>../DMPC64_cg.prmtop<span class="w"> </span>-y<span class="w"> </span>DMPC64_cg_md.nc<span class="w"> </span>-x<span class="w"> </span>DMPC64_cg_md_pbc.nc<span class="w"> </span>--interactive
</pre></div>
</div>
</div></blockquote>
<p>Now you can check the simulation using VMD:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">vmd</span> <span class="o">../</span><span class="n">DMPC64_cg</span><span class="o">.</span><span class="n">prmtop</span> <span class="n">DMPC64_cg_md_pbc</span><span class="o">.</span><span class="n">nc</span> <span class="o">-</span><span class="n">e</span> <span class="o">../</span><span class="n">sirah</span><span class="o">.</span><span class="n">amber</span><span class="o">/</span><span class="n">tools</span><span class="o">/</span><span class="n">sirah_vmdtk</span><span class="o">.</span><span class="n">tcl</span>
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The file <code class="docutils literal notranslate"><span class="pre">sirah_vmdtk.tcl</span></code> is a Tcl script that is part of SIRAH Tools and contains the macros to properly visualize the coarse-grained structures in VMD.</p>
</div>
<p>You can also use CPPTRAJ to calculate the area per lipid:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>cpptraj<span class="w"> </span>-p<span class="w"> </span>../DMPC64_cg.prmtop<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/6/area_lipid.cpptraj
</pre></div>
</div>
<p>Use Grace to plot the results:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>xmgrace<span class="w"> </span>apl_DMPC64_310K.dat
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>To calculate the area per lipid, divide the membrane’s area by the DMPC molecules per leaflet:</p>
<blockquote>
<div><div class="math notranslate nohighlight">
\[\frac{Area}{Lipid} = \frac{Box(x) * Box(y)}{64}\]</div>
</div></blockquote>
</div>
<p>And density profiles and bilayer thickness:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>cpptraj<span class="w"> </span>-p<span class="w"> </span>../DMPC64_cg.prmtop<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/6/dens_profile.cpptraj
</pre></div>
</div>
<p>Use Grace to plot the results:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>xmgrace<span class="w"> </span>-nxy<span class="w"> </span>dens_profile_DMPC64_310K.dat
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The thickness of the bilayer is the distance between the two peaks corresponding to the position of phosphate beads (BFO) along the z-axis.</p>
</div>
</section>
</section>
<section id="membrane-proteins-in-explicit-solvent">
<span id="tutorial-7"></span><h2>7. Membrane proteins in explicit solvent<a class="headerlink" href="#membrane-proteins-in-explicit-solvent" title="Link to this heading"></a></h2>
<p>This tutorial shows how to use the SIRAH force field to perform a coarse grained (CG) simulation of a protein embedded in a lipid bilayer using explicit solvent (called WatFour, WT4). The main references for
this tutorial are: <a class="reference external" href="https://doi.org/10.1021/acs.jctc.9b00006">SIRAH 2.0</a>, <a class="reference external" href="https://doi.org/10.1021/acs.jctc.9b00435">SIRAH Lipids</a>, and <a class="reference external" href="https://academic.oup.com/bioinformatics/article/32/10/1568/1743152">SIRAH Tools</a>.
We strongly advise you to read these articles before starting the tutorial.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>We strongly advise you to read and complete <a class="reference internal" href="#tutorial-5"><span class="std std-ref">Tutorial 5</span></a> and <a class="reference internal" href="#tutorial-6"><span class="std std-ref">Tutorial 6</span></a> before starting.</p>
</div>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>Check <a class="reference internal" href="#download-amber"><span class="std std-ref">download</span></a> section for download and set up details before to start this tutorial.
Since this is <strong>tutorial 7</strong>, remember to replace <code class="docutils literal notranslate"><span class="pre">X.X</span></code> in your folder directory. The files corresponding to this tutorial can be found in: <code class="docutils literal notranslate"><span class="pre">sirah_[version].amber/tutorial/7/</span></code></p>
</div>
<section id="id41">
<h3>7.1. Build CG representations<a class="headerlink" href="#id41" title="Link to this heading"></a></h3>
<div class="admonition caution">
<p class="admonition-title">Caution</p>
<p>The mapping to CG requires the correct protonation state of each residue at a given pH. See <a class="reference internal" href="FAQ.html"><span class="doc">FAQs</span></a> and <a class="reference internal" href="#tutorial-5"><span class="std std-ref">Tutorial 5</span></a> for cautions while preparing and mapping atomistic proteins to SIRAH.</p>
</div>
<p>Map the atomistic structure of protein <a class="reference external" href="https://www.rcsb.org/structure/2KYV">2KYV</a> to its CG representation:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>./sirah.amber/tools/CGCONV/cgconv.pl<span class="w"> </span>-i<span class="w"> </span>-i<span class="w"> </span>sirah.amber/tutorial/7/2kyv.pqr<span class="w"> </span>-o<span class="w"> </span>2kyv_cg.pdb
</pre></div>
</div>
<p>The input file <code class="docutils literal notranslate"><span class="pre">-i</span></code> 2kyv.pqr contains the atomistic representation of <a class="reference external" href="https://www.rcsb.org/structure/2KYV">2KYV</a> structure at pH 7.0, while the output <code class="docutils literal notranslate"><span class="pre">-o</span></code> 2kyv_cg.pdb is its SIRAH CG representation.</p>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>If you already have an atomistic protein within a membrane, then you can simply map the entire system to SIRAH (this is highly recommended) and skip the step 2, however clipping the membrane patch may be required to set a correct solvation box (see bellow). By default no mapping is applied to lipids, as there is no standard naming convention for them. So users are requested to append a MAP file from the list in <a class="reference internal" href="#table"><span class="std std-ref">Table 1</span></a>, by setting the flag <code class="docutils literal notranslate"><span class="pre">-a</span></code> in <strong>cgconv.pl</strong>. We recommend using <a class="reference external" href="https://m3g.github.io/packmol/">PACKMOL</a> for building the system. Reference building-block structures are provided at folder sirah.amber/PDB/, which agree with the mapping scheme <code class="docutils literal notranslate"><span class="pre">sirah.amber/tools/CGCONV/maps/tieleman_lipid.map</span></code>. See <a class="reference internal" href="FAQ.html"><span class="doc">FAQs</span></a> for cautions on mapping lipids to SIRAH and tips on using fragment-based topologies.</p>
</div>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>This an advanced usage of the script <strong>cgconv.pl</strong>, you can learn other capabilities from its help:
<code class="docutils literal notranslate"><span class="pre">./sirah.amber/tools/CGCONV/cgconv.pl</span> <span class="pre">-h</span></code></p>
</div>
<section id="embed-the-protein-in-a-lipid-bilayer">
<h4>7.1.1. Embed the protein in a lipid bilayer<a class="headerlink" href="#embed-the-protein-in-a-lipid-bilayer" title="Link to this heading"></a></h4>
<p>We will show one possible way to do it by starting from a pre-equilibrated CG membrane patch.</p>
<p>In <code class="docutils literal notranslate"><span class="pre">sirah_[version].amber/tutorial/7/</span></code> you will find a pre-stabilized CG DMPC bilayer patch, concatenate it with the previously generated CG representation of the phospholamban (PLN) pentamer (PDB code: <a class="reference external" href="https://www.rcsb.org/structure/2KYV">2KYV</a>):</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>head<span class="w"> </span>-qn<span class="w"> </span>-1<span class="w"> </span>2kyv_cg.pdb<span class="w"> </span>./sirah.amber/tutorial/7/DMPC_cg.pdb<span class="w"> </span>&gt;<span class="w"> </span>2kyv_DMPC_cg_init.pdb
</pre></div>
</div>
<p>Luckily, we already oriented the protein inside the membrane. For setting up your own system you can go to <a class="reference external" href="https://opm.phar.umich.edu/">Orientations of Proteins in Membranes (OPM) database</a> and (if your structure is available) use the dummy atoms provided there to make them match with your membrane model (see <strong>Figure 1</strong>).</p>
<figure class="align-center" id="id55">
<a class="reference internal image-reference" href="_images/Tuto7.png"><img alt="_images/Tuto7.png" src="_images/Tuto7.png" style="width: 100%;" /></a>
<figcaption>
<p><span class="caption-text"><strong>Figure 1.</strong> Protein oriented inside the membrane from the OPM database with dummy atoms represented as orange spheres.</span><a class="headerlink" href="#id55" title="Link to this image"></a></p>
</figcaption>
</figure>
</section>
<section id="delete-close-contact-lipid-molecules">
<h4>7.1.2. Delete close contact lipid molecules<a class="headerlink" href="#delete-close-contact-lipid-molecules" title="Link to this heading"></a></h4>
<p>We need to use VMD to to delete lipid molecules in close contact with the protein. For a proper treatment and visualization of the system in VMD you must first generate the molecular topology and initial coordinate files.</p>
<p>Use a text editor to create the file <code class="docutils literal notranslate"><span class="pre">geninit.leap</span></code> including the following lines:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="gp"># </span>Load<span class="w"> </span>SIRAH<span class="w"> </span>force<span class="w"> </span>field
<span class="go">addPath ./sirah.amber</span>
<span class="go">source leaprc.sirah</span>

<span class="gp"># </span>Load<span class="w"> </span>model
<span class="go">ProtMem = loadpdb 2kyv_DMPC_cg_init.pdb</span>

<span class="gp"># </span>Save<span class="w"> </span>Parms
<span class="go">saveAmberParmNetcdf ProtMem 2kyv_DMPC_cg_init.prmtop 2kyv_DMPC_cg_init.ncrst</span>

<span class="gp"># </span>EXIT
<span class="go">quit</span>
</pre></div>
</div>
<p>Run the LEaP application to generate the molecular topology and initial coordinate files:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>tleap<span class="w"> </span>-f<span class="w"> </span>geninit.leap
</pre></div>
</div>
<div class="admonition caution">
<p class="admonition-title">Caution</p>
<p>Warning messages about long, triangular or square bonds in <code class="docutils literal notranslate"><span class="pre">leap.log</span></code> file are fine and expected due to the CG topology.</p>
</div>
<p>Now, open the files on VMD:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>vmd<span class="w"> </span>2kyv_DMPC_cg_init.prmtop<span class="w"> </span>2kyv_DMPC_cg_init.ncrst<span class="w"> </span>-e<span class="w"> </span>sirah.amber/tools/sirah_vmdtk.tcl
</pre></div>
</div>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>VMD assigns default radius to unknown atom types, the script <code class="docutils literal notranslate"><span class="pre">sirah_vmdtk.tcl</span></code> sets the right
ones. It also provides a kit of useful selection macros, coloring methods and backmapping utilities.
Use the command <code class="docutils literal notranslate"><span class="pre">sirah_help</span></code> in the Tcl/Tk console of VMD to access the manual pages.</p>
</div>
<p>In the VMD main window, select <em>Graphics</em> &gt; <em>Representations</em>. In the <em>Selected Atoms</em> text entry type:</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>not (same residue as (sirah_membrane within 3.5 of sirah_protein) or (sirah_membrane and x &lt; 5 or x &gt; 142 or y &lt; 2 or y &gt; 140))
</pre></div>
</div>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>In the first part of the selection, lipid molecules in close contact with the protein are removed. The second one is made to “trim” the membrane patch, deleting lipids with acyl chains located in the periodic boundary images. This is frequent when using pre-equilibrated membrane patches and is necessary to avoid clashes in the following steps.</p>
</div>
<p>To save the refined protein-membrane system, in the VMD main window click on <code class="docutils literal notranslate"><span class="pre">2kyv_DMPC_cg_init.prmtop</span></code>, then select <em>File</em> &gt; <em>Save Coordinates</em>. In the <em>Selected atoms</em> option choose the selection you have just created and Save as <code class="docutils literal notranslate"><span class="pre">2kyv_DMPC_cg.pdb</span></code>.</p>
<p>From now on it is just normal AMBER stuff!</p>
</section>
</section>
<section id="id42">
<h3>7.2. Prepare LEaP input<a class="headerlink" href="#id42" title="Link to this heading"></a></h3>
<p>Use a text editor to create the file <code class="docutils literal notranslate"><span class="pre">gensystem.leap</span></code> including the following lines:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="gp"># </span>Load<span class="w"> </span>SIRAH<span class="w"> </span>force<span class="w"> </span>field
<span class="go">addPath ./sirah.amber</span>
<span class="go">source leaprc.sirah</span>

<span class="gp"># </span>Load<span class="w"> </span>model
<span class="go">Lipid = loadpdb DMPC64_cg.pdb</span>

<span class="gp"># </span>Add<span class="w"> </span>solvent,<span class="w"> </span>counterions<span class="w"> </span>and<span class="w"> </span><span class="m">0</span>.15M<span class="w"> </span>NaCl
<span class="gp"># </span>Tuned<span class="w"> </span>solute-solvent<span class="w"> </span>closeness<span class="w"> </span><span class="k">for</span><span class="w"> </span>best<span class="w"> </span>hydration
<span class="go">solvateBox Lipid WT4BOX {0 0 40} 0.7</span>
<span class="go">addIonsRand Lipid NaW 33 ClW 33</span>

<span class="gp"># </span>Save<span class="w"> </span>Parms
<span class="go">saveAmberParmNetcdf protein DMPC64_cg.prmtop DMPC64_cg.ncrst</span>

<span class="gp"># </span>EXIT
<span class="go">quit</span>
</pre></div>
</div>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<p>The available ionic species in SIRAH force field are: <code class="docutils literal notranslate"><span class="pre">Na⁺</span></code> (NaW), <code class="docutils literal notranslate"><span class="pre">K⁺</span></code> (KW) and <code class="docutils literal notranslate"><span class="pre">Cl⁻</span></code> (ClW). One ion pair (e.g. NaW-ClW) each 34 WT4 molecules renders a salt concentration of ~0.15M (see <a class="reference internal" href="#appendix"><span class="std std-ref">Appendix</span></a> for details). Counterions were added according to <a class="reference external" href="https://pubs.acs.org/doi/10.1021/acs.jctc.9b00953">Machado et al.</a>.</p>
</div>
</section>
<section id="id44">
<h3>7.3. Run LEaP<a class="headerlink" href="#id44" title="Link to this heading"></a></h3>
<p>Run the LEaP application to generate the molecular topology and initial coordinate files:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>tleap<span class="w"> </span>-f<span class="w"> </span>gensystem.leap
</pre></div>
</div>
<div class="admonition caution">
<p class="admonition-title">Caution</p>
<p>Warning messages about long, triangular or square bonds in <code class="docutils literal notranslate"><span class="pre">leap.log</span></code> file are fine and expected due to the CG topology.</p>
</div>
<p>This should create a topology file <code class="docutils literal notranslate"><span class="pre">2kyv_DMPC_cg.prmtop</span></code> and a coordinate file <code class="docutils literal notranslate"><span class="pre">2kyv_DMPC_cg.ncrst</span></code>.</p>
<p>Use VMD to check how the CG model looks:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>vmd<span class="w"> </span>2kyv_DMPC_cg.prmtop<span class="w"> </span>2kyv_DMPC_cg.ncrst<span class="w"> </span>-e<span class="w"> </span>./sirah.amber/tools/sirah_vmdtk.tcl
</pre></div>
</div>
<p>By selecting +X, +Y and +Z periodic images from the <em>Periodic</em> tab in the <em>Graphical Representations</em> window you will see unwanted water near the
hydrophobic region of the membrane and small vacuum slices at box boundaries. In the following step we will fix these issues by deleting those water molecules and reducing the box dimensions a few angstroms. See <a class="reference internal" href="FAQ.html"><span class="doc">FAQs</span></a> for issues on membrane systems in AMBER. If you do not find your issue please start a discussion in our <a class="reference external" href="https://github.com/SIRAHFF/documentation/discussions">github discussion page F&amp;Q</a>.</p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>VMD assigns default radius to unknown atom types, the script <code class="docutils literal notranslate"><span class="pre">sirah_vmdtk.tcl</span></code> sets the right
ones. It also provides a kit of useful selection macros, coloring methods and backmapping utilities.
Use the command <code class="docutils literal notranslate"><span class="pre">sirah_help</span></code> in the Tcl/Tk console of VMD to access the manual pages.</p>
</div>
</section>
<section id="id45">
<h3>7.4. Resize the box with CPPTRAJ<a class="headerlink" href="#id45" title="Link to this heading"></a></h3>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>As PACKMOL does not consider periodicity while building up the system, increasing the XY box sides a few Angstroms may be required to avoid bad contacts between images.</p>
</div>
<p>Use a text editor to create the file <code class="docutils literal notranslate"><span class="pre">resize_box.cpptraj</span></code> including the following lines:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="gp"># </span>Set<span class="w"> </span>reference<span class="w"> </span>coordinate<span class="w"> </span>file
<span class="go">reference 2kyv_DMPC_cg.ncrst</span>

<span class="gp"># </span>Remove<span class="w"> </span>water<span class="w"> </span>using<span class="w"> </span>distance-based<span class="w"> </span>masks
<span class="go">strip :WT4&amp;(@BCT1,BCT2,BC13,BC23&lt;:12.0) parmout 2kyv_DMPC_cg.prmtop</span>

<span class="gp"># </span>New<span class="w"> </span>box<span class="w"> </span>dimensions
<span class="go">box x 128 y 128 z 114</span>

<span class="gp"># </span>Amber<span class="w"> </span>NetCDF<span class="w"> </span>Restart<span class="w"> </span>generation
<span class="go">trajout 2kyv_DMPC_cg_nb.ncrst</span>

<span class="gp"># </span>Do<span class="w"> </span>it!
<span class="go"> go</span>

<span class="gp"># </span>EXIT
<span class="go"> quit</span>
</pre></div>
</div>
<div class="admonition caution">
<p class="admonition-title">Caution</p>
<p>This is a critical step when preparing membrane systems to simulate with AMBER. In this case, the new box dimensions were set after some trial and error tests to allow for limited overlap between periodic box images. An excessive overlap may lead to important atom clashes an eventual system explosion during minimization/simulation, while insufficient overlap may impact the membrane cohesivity at PBC boundaries leading to pore formations or other issues.</p>
</div>
<p>Run the CPPTRAJ application application to adjust the size of the simulation box:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>cpptraj<span class="w"> </span>-p<span class="w"> </span>2kyv_DMPC_cg.prmtop<span class="w"> </span>-y<span class="w"> </span>2kyv_DMPC_cg.ncrst<span class="w"> </span>-i<span class="w"> </span>resize_box.cpptraj
</pre></div>
</div>
<p>Once again, use VMD to check the PBC images in the new box of the system:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>vmd<span class="w"> </span>2kyv_DMPC_cg.prmtop<span class="w"> </span>2kyv_DMPC_cg_nb.ncrst<span class="w"> </span>-e<span class="w"> </span>./sirah.amber/tools/sirah_vmdtk.tcl
</pre></div>
</div>
</section>
<section id="id46">
<h3>7.5. Run the simulation<a class="headerlink" href="#id46" title="Link to this heading"></a></h3>
<p>Make a new folder for the run:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>mkdir<span class="w"> </span>-p<span class="w"> </span>run<span class="p">;</span><span class="w"> </span><span class="nb">cd</span><span class="w"> </span>run
</pre></div>
</div>
<p>The folder <code class="docutils literal notranslate"><span class="pre">sirah.amber/tutorial/7/</span></code> contains typical input files for energy minimization
(<code class="docutils literal notranslate"><span class="pre">em1_Prot-Lip.in</span></code> and <code class="docutils literal notranslate"><span class="pre">em2_Prot-Lip.in</span></code>), heating (<code class="docutils literal notranslate"><span class="pre">heat_Prot-Lip.in</span></code>), equilibration (<code class="docutils literal notranslate"><span class="pre">eq_Prot-Lip.in</span></code>) and production (<code class="docutils literal notranslate"><span class="pre">md_Prot-Lip.in</span></code>) runs. Please check carefully the input flags.</p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<blockquote>
<div><p><strong>Some flags used in AMBER</strong></p>
</div></blockquote>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">-i</span></code>: Input file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-o</span></code>: Output file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-p</span></code>: Parameter/topology file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-c</span></code>: Coordinate file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-r</span></code>: Restart file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-x</span></code>: Trajectory file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">ref</span></code>: Reference file</p></li>
</ul>
</div>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>These input files are executed by the <strong>GPU</strong> implementation of <em>pmemd.cuda</em>, due to the system size we do not recommend the use of <strong>CPU</strong> implementations of AMBER.</p>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Other available implementations that could be used: <code class="docutils literal notranslate"><span class="pre">sander</span></code>  or <code class="docutils literal notranslate"><span class="pre">pmemd</span></code>, both <strong>CPU</strong> implementations of AMBER.</p>
</div>
<p><strong>Energy Minimization of side chains by restraining the backbone:</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/7/em1_Prot-Lip.in<span class="w"> </span>-p<span class="w"> </span>../2kyv_DMPC_cg.prmtop<span class="w"> </span>-c<span class="w"> </span>../2kyv_DMPC_cg_nb.ncrst<span class="w"> </span>-ref<span class="w"> </span>../2kyv_DMPC_cg_nb.ncrst<span class="w"> </span>-o<span class="w"> </span>2kyv_DMPC_cg_em_1.out<span class="w"> </span>-r<span class="w"> </span>2kyv_DMPC_cg_em_1.ncrst<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
<p><strong>Energy Minimization of of the whole system:</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/7/em2_Prot-Lip.in<span class="w"> </span>-p<span class="w"> </span>../2kyv_DMPC_cg.prmtop<span class="w"> </span>-c<span class="w"> </span>2kyv_DMPC_cg_em_1.ncrst<span class="w"> </span>-ref<span class="w"> </span>2kyv_DMPC_cg_em_1.ncrst<span class="w"> </span>-o<span class="w"> </span>2kyv_DMPC_cg_em_2.out<span class="w"> </span>-r<span class="w"> </span>2kyv_DMPC_cg_em_2.ncrst<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
<p><strong>Heating:</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/7/heat_Prot-Lip.in<span class="w"> </span>-p<span class="w"> </span>../2kyv_DMPC_cg.prmtop<span class="w"> </span>-c<span class="w"> </span>2kyv_DMPC_cg_em_2.ncrst<span class="w"> </span>-ref<span class="w"> </span>2kyv_DMPC_cg_em_2.ncrst<span class="w"> </span>-o<span class="w"> </span>2kyv_DMPC_cg_eq_0.out<span class="w"> </span>-r<span class="w"> </span>2kyv_DMPC_cg_eq_0.ncrst<span class="w"> </span>-x<span class="w"> </span>2kyv_DMPC_cg_eq_0.nc<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>To avoid “<em>skinnb errors</em>” on GPU due to large box size fluctuations, the system must be equilibrated by several “short” runs using a large <em>skinnb</em> value. The number and length of the runs may vary according to the characteristic stabilization times of the system. For more information visit the <a class="reference external" href="http://ambermd.org/tutorials/advanced/tutorial16/">AMBER tutorial on lipids</a>.</p>
</div>
<p><strong>Periodic box equilibration in GPU code (500 ps x 9):</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="k">for</span><span class="w"> </span>i<span class="w"> </span><span class="k">in</span><span class="w"> </span><span class="k">$(</span>seq<span class="w"> </span><span class="m">1</span><span class="w"> </span><span class="m">9</span><span class="k">)</span>
<span class="k">do</span>
<span class="w">        </span><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;running equilibration </span><span class="nv">$i</span><span class="s2">&quot;</span>
<span class="w">        </span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span><span class="se">\</span>
<span class="w">        </span>-i<span class="w"> </span>../sirah.amber/tutorial/7/heat_Prot-Lip.in<span class="w"> </span><span class="se">\</span>
<span class="w">        </span>-p<span class="w"> </span>../2kyv_DMPC_cg.prmtop<span class="w"> </span><span class="se">\</span>
<span class="w">        </span>-c<span class="w"> </span>2kyv_DMPC_cg_eq_<span class="k">$((</span><span class="nv">$i</span><span class="w"> </span><span class="o">-</span><span class="m">1</span><span class="k">))</span>.ncrst<span class="w"> </span><span class="se">\</span>
<span class="w">        </span>-ref<span class="w"> </span>2kyv_DMPC_cg_eq_<span class="k">$((</span><span class="nv">$i</span><span class="w"> </span><span class="o">-</span><span class="m">1</span><span class="k">))</span>.ncrst<span class="w"> </span><span class="se">\</span>
<span class="w">        </span>-o<span class="w"> </span>2kyv_DMPC_cg_eq_<span class="nv">$i</span>.out<span class="w"> </span><span class="se">\</span>
<span class="w">        </span>-r<span class="w"> </span>2kyv_DMPC_cg_eq_<span class="nv">$i</span>.ncrst<span class="w"> </span><span class="se">\</span>
<span class="w">        </span>-x<span class="w"> </span>2kyv_DMPC_cg_eq_<span class="nv">$i</span>.nc
<span class="k">done</span><span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
<p><strong>Production (1000ns):</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pmemd.cuda<span class="w"> </span>-O<span class="w"> </span>-i<span class="w"> </span>../sirah.amber/tutorial/7/md_Prot-Lip.in<span class="w"> </span>-p<span class="w"> </span>../2kyv_DMPC_cg.prmtop<span class="w"> </span>-c<span class="w"> </span>2kyv_DMPC_cg_eq_9.ncrst<span class="w"> </span>-o<span class="w"> </span>2kyv_DMPC_cg_md.out<span class="w"> </span>-r<span class="w"> </span>2kyv_DMPC_cg_md.ncrst<span class="w"> </span>-x<span class="w"> </span>2kyv_DMPC_cg_md.nc<span class="w"> </span><span class="p">&amp;</span>
</pre></div>
</div>
</section>
<section id="id47">
<h3>7.6. Visualizing the simulation<a class="headerlink" href="#id47" title="Link to this heading"></a></h3>
<p>That’s it! Now you can analyze the trajectory.</p>
<p>Process the output trajectory to account for the Periodic Boundary Conditions (PBC):</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;autoimage\ngo\nquit\n&quot;</span><span class="w"> </span><span class="p">|</span><span class="w"> </span>cpptraj<span class="w"> </span>-p<span class="w"> </span>../2kyv_DMPC_cg.prmtop<span class="w"> </span>-y<span class="w"> </span>2kyv_DMPC_cg_md.nc<span class="w"> </span>-x<span class="w"> </span>2kyv_DMPC_cg_md_pbc.nc<span class="w"> </span>--interactive
</pre></div>
</div>
<p>Now you can check the simulation using VMD:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">vmd</span> <span class="o">../</span><span class="mi">2</span><span class="n">kyv_DMPC_cg</span><span class="o">.</span><span class="n">prmtop</span> <span class="mi">2</span><span class="n">kyv_DMPC_cg_md_pbc</span><span class="o">.</span><span class="n">nc</span> <span class="o">-</span><span class="n">e</span> <span class="o">../</span><span class="n">sirah</span><span class="o">.</span><span class="n">amber</span><span class="o">/</span><span class="n">tools</span><span class="o">/</span><span class="n">sirah_vmdtk</span><span class="o">.</span><span class="n">tcl</span>
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The file <code class="docutils literal notranslate"><span class="pre">sirah_vmdtk.tcl</span></code> is a Tcl script that is part of SIRAH Tools and contains the macros to properly visualize the coarse-grained structures in VMD.</p>
</div>
</section>
</section>
<section id="appendix">
<span id="id48"></span><h2>Appendix<a class="headerlink" href="#appendix" title="Link to this heading"></a></h2>
<section id="calculating-ionic-concentrations">
<h3>Calculating ionic concentrations<a class="headerlink" href="#calculating-ionic-concentrations" title="Link to this heading"></a></h3>
<div class="math notranslate nohighlight">
\[\]</div>
<script type="text/javascript" src="http://localhost/mathjax/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>\[\rho_{\text{WT4}} = \rho_{\text{H}_2\text{O}} = 1000 \, \text{g/L} \]

 \[MW_{\text{H}_2\text{O}} = 18 \, \text{g/mol} \]

 \[1 \text{WT4} \sim 11 \text{H}_2\text{O} \]

 \[\text{M}=\frac{\text{mol}}{V} \quad ; \quad \text{n}={\text{N}_A} \quad ; \quad \rho = \frac{\text{m}}{V} \quad ; \quad \text{m} = \text{mol} \cdot \text{M} \cdot \text{W}\]

 \[\text{V} = \frac{\text{m}}{\rho} = \frac{\text{mol} \cdot \text{MW}_{\text{H}_2\text{O}}}{\rho} = \frac{\text{n}_{\text{H}_2\text{O}} \cdot \text{MW}_{\text{H}_2\text{O}}}{{\text{N}_A} \cdot \rho}\]


 \[\text{M}=\frac{\text{mol}}{V} = \frac{\text{n}_{\text{ion}}}{\text{N}_{\text{A}} \cdot \text{V}} = \frac{\text{n}_{\text{ion}}}{\text{N}_{\text{A}}} \frac{\text{N}_{\text{A}} \cdot \rho}{\text{n}_{\text{H}_2\text{O}} \cdot \text{MW}_{\text{H}_2\text{O}}} = \frac{\text{n}_{\text{ion}} \cdot 1000}{\text{n}_{\text{WT}_4} \cdot \text{(11)} \cdot \text{(18)}} \sim 5 \frac{\text{n}_{\text{ion}}}{\text{n}_{\text{WT}_4}}  \]<p>Number of <em>WT4</em> molecules per ion at 0.15M: <span class="raw-latex">\(\text{n}_{WT4} = 5 {{\text{n}_{\text{ion}}} \over \text{M}} = {5(1) \over 0.15} \sim 34 \)</span></p>
</section>
<section id="available-mapping-files">
<span id="table"></span><h3>Available mapping files<a class="headerlink" href="#available-mapping-files" title="Link to this heading"></a></h3>
<p>Table 1. Available mapping files (MAPs) at folder <code class="docutils literal notranslate"><span class="pre">sirah.amber/tools/CGCONV/maps/</span></code> for converting atomistic lipid structures to SIRAH models. <strong>Important!</strong> MAPs can not inter-convert different name conventions, e.g. amber_lipid.map won’t generate fragment-based residues from residue-based force fields. Due to possible nomenclature conflicts, users are advised to check and modify the MAPs as required.</p>
<table class="docutils align-default">
<tbody>
<tr class="row-odd"><td><p><strong>Map</strong></p></td>
<td><p><strong>Type</strong>*</p></td>
<td><p><strong>Compatibility</strong></p></td>
<td><p><strong>Source</strong></p></td>
</tr>
<tr class="row-even"><td><p>amber_lipid.map</p></td>
<td><p>F</p></td>
<td><p>AMBER Lipid11-17 force fields</p></td>
<td><div class="line-block">
<div class="line"><a class="reference external" href="http://ambermd.org/">AMBER</a></div>
<div class="line"><a class="reference external" href="https://software.acellera.com/htmd/index.html">HTMD</a></div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p>GAFF_lipid.map</p></td>
<td><p>R</p></td>
<td><p>AMBER GAFF force field</p></td>
<td><p><a class="reference external" href="https://lipidbook.org/">LipidBook</a></p></td>
</tr>
<tr class="row-even"><td><p>charmm_lipid.map</p></td>
<td><p>R</p></td>
<td><div class="line-block">
<div class="line">CHARMM 27/36 force field, and “CHARMM</div>
<div class="line">compatible” GAFF nomenclature</div>
</div>
</td>
<td><div class="line-block">
<div class="line"><a class="reference external" href="https://charmm-gui.org/">CHARMM-GUI</a></div>
<div class="line"><a class="reference external" href="https://www.gromacs.org/">GROMACS</a></div>
<div class="line"><a class="reference external" href="https://lipidbook.org/">LipidBook</a></div>
<div class="line"><a class="reference external" href="http://bioinf.modares.ac.ir/software/mb/">MemBuilder</a></div>
<div class="line"><a class="reference external" href="https://software.acellera.com/htmd/index.html">HTMD</a></div>
<div class="line"><a class="reference external" href="https://www.ks.uiuc.edu/Research/vmd/">VMD</a></div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p>slipids.map</p></td>
<td><p>R</p></td>
<td><p>Stockholm lipids force field</p></td>
<td><div class="line-block">
<div class="line"><a class="reference external" href="http://www.fos.su.se/~sasha/SLipids/About.html">SLIPIDS</a></div>
<div class="line"><a class="reference external" href="http://bioinf.modares.ac.ir/software/mb/">MemBuilder</a></div>
</div>
</td>
</tr>
<tr class="row-even"><td><p>OPLSA-AA_2014_lipid.map</p></td>
<td><p>R</p></td>
<td><p>All-atoms lipids for OPLS force field</p></td>
<td><div class="line-block">
<div class="line"><a class="reference external" href="https://doi.org/10.1021/jp5016627">Maciejewski</a></div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p>OPLSA-UA_lipid.map</p></td>
<td><p>R</p></td>
<td><p>United-atom lipids for OPLS force field</p></td>
<td><p><a class="reference external" href="https://lipidbook.org/">LipidBook</a></p></td>
</tr>
<tr class="row-even"><td><p>GROMOS43a1_lipid.map</p></td>
<td><p>R</p></td>
<td><div class="line-block">
<div class="line">United-atom lipids for GROMOS 43a1 and</div>
<div class="line">CKP force fields</div>
</div>
</td>
<td><div class="line-block">
<div class="line"><a class="reference external" href="https://lipidbook.org/">LipidBook</a></div>
<div class="line"><a class="reference external" href="http://bioinf.modares.ac.ir/software/mb/">MemBuilder</a></div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p>GROMOS43a1-s3_lipid.map</p></td>
<td><p>R</p></td>
<td><div class="line-block">
<div class="line">United-atom lipids for GROMOS 43a1-s3</div>
<div class="line">force field</div>
</div>
</td>
<td><div class="line-block">
<div class="line"><a class="reference external" href="https://www.gromacs.org/">GROMACS</a></div>
<div class="line"><a class="reference external" href="https://lipidbook.org/">LipidBook</a></div>
<div class="line"><a class="reference external" href="http://bioinf.modares.ac.ir/software/mb/">MemBuilder</a></div>
</div>
</td>
</tr>
<tr class="row-even"><td><p>GROMOS53a6_lipid.map</p></td>
<td><p>R</p></td>
<td><div class="line-block">
<div class="line">United-atom lipids for GROMOS 53a6</div>
<div class="line">force field</div>
</div>
</td>
<td><div class="line-block">
<div class="line"><a class="reference external" href="https://www.gromacs.org/">GROMACS</a></div>
<div class="line"><a class="reference external" href="https://lipidbook.org/">LipidBook</a></div>
<div class="line"><a class="reference external" href="http://bioinf.modares.ac.ir/software/mb/">MemBuilder</a></div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p>tieleman_lipid.map</p></td>
<td><p>R</p></td>
<td><div class="line-block">
<div class="line">Berger lipids as implemented by</div>
<div class="line">Tieleman et al. for GROMOS</div>
<div class="line">force fields.</div>
</div>
</td>
<td><div class="line-block">
<div class="line"><a class="reference external" href="https://doi.org/10.1021/ja0624321">Tieleman</a></div>
<div class="line"><a class="reference external" href="https://lipidbook.org/">LipidBook</a></div>
</div>
</td>
</tr>
</tbody>
</table>
<p>* Fragment-based (F) or Residue-based (R) topology.</p>
</section>
</section>
</section>
<section id="gromacs">
<h1>GROMACS<a class="headerlink" href="#gromacs" title="Link to this heading"></a></h1>
</section>
<section id="namd">
<span id="id50"></span><h1>NAMD<a class="headerlink" href="#namd" title="Link to this heading"></a></h1>
</section>
<section id="id51">
<span id="id52"></span><h1>SIRAH tools<a class="headerlink" href="#id51" title="Link to this heading"></a></h1>
<span class="target" id="id53"></span></section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="Developers.html" class="btn btn-neutral float-left" title="Developers" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2023, BA, SL, SM.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>